--- src/config_old.c	2013-01-20 16:54:29.505188612 +0100
+++ src/config.c	2013-01-20 19:12:11.645257675 +0100
@@ -675,6 +675,7 @@
 			}
 		}
 	}
+#ifdef ENABLE_BATTERY
 	else if (strcmp(key, "battery") == 0) {
 		if (new_config_file == 0) {
 			battery_enabled = atoi(value);
@@ -689,6 +690,7 @@
 			}
 		}
 	}
+#endif
 	else
 		fprintf(stderr, "tint2 : invalid option \"%s\",\n  upgrade tint2 or correct your config file\n", key);
 
