diff -rupN PyQwt-5.2.0/sip/qwt5qt3/qwt_picker_machine.sip PyQwt-5.2.0-new/sip/qwt5qt3/qwt_picker_machine.sip
--- PyQwt-5.2.0/sip/qwt5qt3/qwt_picker_machine.sip	2009-02-17 10:55:14.000000000 +0100
+++ PyQwt-5.2.0-new/sip/qwt5qt3/qwt_picker_machine.sip	2017-01-04 18:48:46.090764505 +0100
@@ -104,37 +104,37 @@ protected:
 %If (CXX_DYNAMIC_CAST)
 %ConvertToSubClassCode
     // Walk the inheritance tree depth first in alphabetical order
-#ifdef sipClass_QwtPickerClickPointMachine
+#ifdef sipType_QwtPickerClickPointMachine
     if (dynamic_cast<const QwtPickerClickPointMachine *>(sipCpp))
-        sipClass = sipClass_QwtPickerClickPointMachine;
+        sipType = sipType_QwtPickerClickPointMachine;
     else
 #endif
-#ifdef sipClass_QwtPickerClickRectMachine 
+#ifdef sipType_QwtPickerClickRectMachine
     if (dynamic_cast<const QwtPickerClickRectMachine *>(sipCpp))
-        sipClass = sipClass_QwtPickerClickRectMachine;
+        sipType = sipType_QwtPickerClickRectMachine;
     else
 #endif
-#ifdef sipClass_QwtPickerDragPointMachine
+#ifdef sipType_QwtPickerDragPointMachine
     if (dynamic_cast<const QwtPickerDragPointMachine *>(sipCpp))
-        sipClass = sipClass_QwtPickerDragPointMachine;
+        sipType = sipType_QwtPickerDragPointMachine;
     else
 #endif
-#ifdef sipClass_QwtPickerDragRectMachine 
+#ifdef sipType_QwtPickerDragRectMachine
     if (dynamic_cast<const QwtPickerDragRectMachine *>(sipCpp))
-        sipClass = sipClass_QwtPickerDragRectMachine;
+        sipType = sipType_QwtPickerDragRectMachine;
     else
 #endif
-#ifdef sipClass_QwtPickerPolygonMachine
+#ifdef sipType_QwtPickerPolygonMachine
     if (dynamic_cast<const QwtPickerPolygonMachine *>(sipCpp))
-        sipClass = sipClass_QwtPickerPolygonMachine;
+        sipType = sipType_QwtPickerPolygonMachine;
     else
 #endif
-#ifdef sipClass_QwtPickerMachine
+#ifdef sipType_QwtPickerMachine
     if (dynamic_cast<const QwtPickerMachine *>(sipCpp))
-        sipClass = sipClass_QwtPickerMachine;
+        sipType = sipType_QwtPickerMachine;
     else
 #endif
-        sipClass = 0;
+        sipType = 0;
 %End
 %End
 
