diff -rupN PyQwt-5.2.0/sip/qwt5qt3/qwt_plot_item.sip PyQwt-5.2.0-new/sip/qwt5qt3/qwt_plot_item.sip
--- PyQwt-5.2.0/sip/qwt5qt3/qwt_plot_item.sip	2009-02-17 10:55:14.000000000 +0100
+++ PyQwt-5.2.0-new/sip/qwt5qt3/qwt_plot_item.sip	2017-01-04 18:48:34.224862219 +0100
@@ -98,21 +98,21 @@ private:
 %ConvertToSubClassCode
     Py_BEGIN_ALLOW_THREADS
     switch (sipCpp->rtti()) {
-    case QwtPlotItem::Rtti_PlotItem: sipClass = sipClass_QwtPlotItem; break;
-    case QwtPlotItem::Rtti_PlotGrid: sipClass = sipClass_QwtPlotGrid; break; 
-#ifdef sipClass_QwtPlotScaleItem
-    case QwtPlotItem::Rtti_PlotScale: sipClass = sipClass_QwtPlotScaleItem; break; 
-#endif // sipClass_QwtPlotScaleItem
-    case QwtPlotItem::Rtti_PlotMarker: sipClass = sipClass_QwtPlotMarker; break; 
-    case QwtPlotItem::Rtti_PlotCurve: sipClass = sipClass_QwtPlotCurve; break;
-#ifdef sipClass_QwtPlotHistogram
-    case QwtPlotItem::Rtti_PlotHistogram: sipClass = sipClass_QwtPlotHistogram; break;
-#endif // sipClass_QwtPlotHistogram
-    case QwtPlotItem::Rtti_PlotSpectrogram: sipClass = sipClass_QwtPlotSpectrogram; break;
-#ifdef sipClass_QwtPlotSvgItem
-    case QwtPlotItem::Rtti_PlotSVG: sipClass = sipClass_QwtPlotSvgItem; break; 
-#endif // sipClass_QwtPlotSvgItem
-    default: sipClass = 0;
+    case QwtPlotItem::Rtti_PlotItem: sipType = sipType_QwtPlotItem; break;
+    case QwtPlotItem::Rtti_PlotGrid: sipType = sipType_QwtPlotGrid; break;
+#ifdef sipType_QwtPlotScaleItem
+    case QwtPlotItem::Rtti_PlotScale: sipType = sipType_QwtPlotScaleItem; break;
+#endif // sipType_QwtPlotScaleItem
+    case QwtPlotItem::Rtti_PlotMarker: sipType = sipType_QwtPlotMarker; break;
+    case QwtPlotItem::Rtti_PlotCurve: sipType = sipType_QwtPlotCurve; break;
+#ifdef sipType_QwtPlotHistogram
+    case QwtPlotItem::Rtti_PlotHistogram: sipType = sipType_QwtPlotHistogram; break;
+#endif // sipType_QwtPlotHistogram
+    case QwtPlotItem::Rtti_PlotSpectrogram: sipType = sipType_QwtPlotSpectrogram; break;
+#ifdef sipType_QwtPlotSvgItem
+    case QwtPlotItem::Rtti_PlotSVG: sipType = sipType_QwtPlotSvgItem; break;
+#endif // sipType_QwtPlotSvgItem
+    default: sipType = 0;
     }
     Py_END_ALLOW_THREADS
 %End // %ConvertToSubClassCode
@@ -190,21 +190,21 @@ private:
 %ConvertToSubClassCode
     Py_BEGIN_ALLOW_THREADS
     switch (sipCpp->rtti()) {
-    case QwtPlotItem::Rtti_PlotItem: sipClass = sipClass_QwtPlotItem; break;
-    case QwtPlotItem::Rtti_PlotGrid: sipClass = sipClass_QwtPlotGrid; break; 
-#ifdef sipClass_QwtPlotScaleItem
-    case QwtPlotItem::Rtti_PlotScale: sipClass = sipClass_QwtPlotScaleItem; break; 
-#endif // sipClass_QwtPlotScaleItem
-    case QwtPlotItem::Rtti_PlotMarker: sipClass = sipClass_QwtPlotMarker; break; 
-    case QwtPlotItem::Rtti_PlotCurve: sipClass = sipClass_QwtPlotCurve; break;
-#ifdef sipClass_QwtPlotHistogram
-    case QwtPlotItem::Rtti_PlotHistogram: sipClass = sipClass_QwtPlotHistogram; break;
-#endif // sipClass_QwtPlotHistogram
-    case QwtPlotItem::Rtti_PlotSpectrogram: sipClass = sipClass_QwtPlotSpectrogram; break;
-#ifdef sipClass_QwtPlotSvgItem
-    case QwtPlotItem::Rtti_PlotSVG: sipClass = sipClass_QwtPlotSvgItem; break; 
-#endif // sipClass_QwtPlotSvgItem
-    default: sipClass = 0;
+    case QwtPlotItem::Rtti_PlotItem: sipType = sipType_QwtPlotItem; break;
+    case QwtPlotItem::Rtti_PlotGrid: sipType = sipType_QwtPlotGrid; break;
+#ifdef sipType_QwtPlotScaleItem
+    case QwtPlotItem::Rtti_PlotScale: sipType = sipType_QwtPlotScaleItem; break;
+#endif // sipType_QwtPlotScaleItem
+    case QwtPlotItem::Rtti_PlotMarker: sipType = sipType_QwtPlotMarker; break;
+    case QwtPlotItem::Rtti_PlotCurve: sipType = sipType_QwtPlotCurve; break;
+#ifdef sipType_QwtPlotHistogram
+    case QwtPlotItem::Rtti_PlotHistogram: sipType = sipType_QwtPlotHistogram; break;
+#endif // sipType_QwtPlotHistogram
+    case QwtPlotItem::Rtti_PlotSpectrogram: sipType = sipType_QwtPlotSpectrogram; break;
+#ifdef sipType_QwtPlotSvgItem
+    case QwtPlotItem::Rtti_PlotSVG: sipType = sipType_QwtPlotSvgItem; break;
+#endif // sipType_QwtPlotSvgItem
+    default: sipType = 0;
     }
     Py_END_ALLOW_THREADS
 %End // %ConvertToSubClassCode
