diff -rupN PyQwt-5.2.0/sip/qwt5qt4/QwtMap.sip PyQwt-5.2.0-new/sip/qwt5qt4/QwtMap.sip
--- PyQwt-5.2.0/sip/qwt5qt4/QwtMap.sip	2009-03-29 19:39:25.000000000 +0200
+++ PyQwt-5.2.0-new/sip/qwt5qt4/QwtMap.sip	2017-01-04 19:26:45.684228520 +0100
@@ -49,13 +49,8 @@
 
         PyObject *kobj = PyFloat_FromDouble(i.key());
 
-#if SIP_VERSION < 0x040800
-        PyObject *tobj = sipConvertFromNewInstance(
-            t, sipClass_QString, sipTransferObj);
-#else
         PyObject *tobj = sipConvertFromNewType(
             t, sipType_QString, sipTransferObj);
-#endif
 
         if ((0 == kobj) || (0 == tobj) || (0 > PyDict_SetItem(d, kobj, tobj))) {
             Py_DECREF(d);
@@ -89,11 +84,7 @@
             return 0;
 
         while (PyDict_Next(sipPy, &i, &kobj, &tobj))
-#if SIP_VERSION < 0x040800
-            if (!sipCanConvertToInstance(tobj, sipClass_QString, SIP_NOT_NONE))
-#else
             if (!sipCanConvertToType(tobj, sipType_QString, SIP_NOT_NONE))
-#endif
                 return 0;
 
         return 1;
@@ -104,22 +95,12 @@
     while (PyDict_Next(sipPy, &i, &kobj, &tobj)) {
         int state;
         double k = PyFloat_AS_DOUBLE(kobj);
-#if SIP_VERSION < 0x040800
-        QString *t = reinterpret_cast<QString *>(
-            sipConvertToInstance(tobj, sipClass_QString, sipTransferObj,
-                                 SIP_NOT_NONE, &state, sipIsErr));
-#else
         QString *t = reinterpret_cast<QString *>(
             sipConvertToType(tobj, sipType_QString, sipTransferObj,
                              SIP_NOT_NONE, &state, sipIsErr));
-#endif
 
         if (*sipIsErr) {
-#if SIP_VERSION < 0x040800
-            sipReleaseInstance(t, sipClass_QString, state);
-#else
             sipReleaseType(t, sipType_QString, state);
-#endif
 
             delete qm;
             return 0;
@@ -127,11 +108,7 @@
 
         qm->insert(k, *t);
 
-#if SIP_VERSION < 0x040800
-        sipReleaseInstance(t, sipClass_QString, state);
-#else
         sipReleaseType(t, sipType_QString, state);
-#endif
     }
 
     *sipCppPtr = qm;
