diff -rupN PyQwt-5.2.0/sip/qwt5qt3/qwt_plot_layout.sip PyQwt-5.2.0-new/sip/qwt5qt3/qwt_plot_layout.sip
--- PyQwt-5.2.0/sip/qwt5qt3/qwt_plot_layout.sip	2009-02-17 10:55:14.000000000 +0100
+++ PyQwt-5.2.0-new/sip/qwt5qt3/qwt_plot_layout.sip	2017-01-04 18:48:53.740701500 +0100
@@ -84,10 +84,10 @@ sipRes = Py_BuildValue("i(iiii)", title,
 QRect scales[QwtPlot::axisCnt];
 sipCpp->sipProtect_alignScales(a0, *a1, scales);
 sipRes = sipBuildResult(0, "(BBBB)",
-                        new QRect(scales[0]), sipClass_QRect, 0,
-                        new QRect(scales[1]), sipClass_QRect, 0,
-                        new QRect(scales[2]), sipClass_QRect, 0,
-                        new QRect(scales[3]), sipClass_QRect, 0);
+                        new QRect(scales[0]), sipType_QRect, 0,
+                        new QRect(scales[1]), sipType_QRect, 0,
+                        new QRect(scales[2]), sipType_QRect, 0,
+                        new QRect(scales[3]), sipType_QRect, 0);
 %End
 
 }; // class QwtPlotLayout
