diff -rupN PyQwt-5.2.0/sip/qwt5qt4/qwt_abstract_slider.sip PyQwt-5.2.0-new/sip/qwt5qt4/qwt_abstract_slider.sip
--- PyQwt-5.2.0/sip/qwt5qt4/qwt_abstract_slider.sip	2009-03-29 19:39:25.000000000 +0200
+++ PyQwt-5.2.0-new/sip/qwt5qt4/qwt_abstract_slider.sip	2017-01-04 19:20:45.551874134 +0100
@@ -86,48 +86,48 @@ protected:
 %ConvertToSubClassCode
     static struct class_graph {
         const char *name;
-        sipWrapperType **type;
+        sipTypeDef **type;
         int yes, no;
     } graph[] = {
-        {sipName_QwtAbstractSlider, &sipClass_QwtAbstractSlider, 13, 1},
+        {sipName_QwtAbstractSlider, &sipType_QwtAbstractSlider, 13, 1},
 #if QWT_VERSION >= 0x050100
-        {sipName_QwtPanner, &sipClass_QwtPanner, 19, 2},
+        {sipName_QwtPanner, &sipType_QwtPanner, 19, 2},
 #else
         {0, 0, 19, 2},
 #endif
-        {sipName_QwtDynGridLayout, &sipClass_QwtDynGridLayout, -1, 3},
-        {sipName_QwtCounter, &sipClass_QwtCounter, -1, 4},
-        {sipName_QwtLegend, &sipClass_QwtLegend, -1, 5},
-        {sipName_QwtArrowButton, &sipClass_QwtArrowButton, -1, 6},
-        {sipName_QwtScaleWidget, &sipClass_QwtScaleWidget, -1, 7},
-        {sipName_QwtTextLabel, &sipClass_QwtTextLabel, 20, 8},
-        {sipName_QwtPlotCanvas, &sipClass_QwtPlotCanvas, -1, 9},
-        {sipName_QwtThermo, &sipClass_QwtThermo, -1, 10},
+        {sipName_QwtDynGridLayout, &sipType_QwtDynGridLayout, -1, 3},
+        {sipName_QwtCounter, &sipType_QwtCounter, -1, 4},
+        {sipName_QwtLegend, &sipType_QwtLegend, -1, 5},
+        {sipName_QwtArrowButton, &sipType_QwtArrowButton, -1, 6},
+        {sipName_QwtScaleWidget, &sipType_QwtScaleWidget, -1, 7},
+        {sipName_QwtTextLabel, &sipType_QwtTextLabel, 20, 8},
+        {sipName_QwtPlotCanvas, &sipType_QwtPlotCanvas, -1, 9},
+        {sipName_QwtThermo, &sipType_QwtThermo, -1, 10},
 #if QWT_VERSION >= 0x050100
-        {sipName_QwtMagnifier, &sipClass_QwtMagnifier, 21, 11},
+        {sipName_QwtMagnifier, &sipType_QwtMagnifier, 21, 11},
 #else
         {0, 0, 21, 11},
 #endif
-        {sipName_QwtPlot, &sipClass_QwtPlot, -1, 12},
-        {sipName_QwtPicker, &sipClass_QwtPicker, 22, -1},
-        {sipName_QwtDial, &sipClass_QwtDial, 17, 14},
-        {sipName_QwtSlider, &sipClass_QwtSlider, -1, 15},
-        {sipName_QwtWheel, &sipClass_QwtWheel, -1, 16},
-        {sipName_QwtKnob, &sipClass_QwtKnob, -1, -1},
-        {sipName_QwtAnalogClock, &sipClass_QwtAnalogClock, -1, 18},
-        {sipName_QwtCompass, &sipClass_QwtCompass, -1, -1},
-        {sipName_QwtPlotPanner, &sipClass_QwtPlotPanner, -1, -1},
-        {sipName_QwtLegendItem, &sipClass_QwtLegendItem, -1, -1},
-        {sipName_QwtPlotMagnifier, &sipClass_QwtPlotMagnifier, -1, -1},
-        {sipName_QwtPlotPicker, &sipClass_QwtPlotPicker, 23, -1},
-        {sipName_QwtPlotZoomer, &sipClass_QwtPlotZoomer, -1, -1},
+        {sipName_QwtPlot, &sipType_QwtPlot, -1, 12},
+        {sipName_QwtPicker, &sipType_QwtPicker, 22, -1},
+        {sipName_QwtDial, &sipType_QwtDial, 17, 14},
+        {sipName_QwtSlider, &sipType_QwtSlider, -1, 15},
+        {sipName_QwtWheel, &sipType_QwtWheel, -1, 16},
+        {sipName_QwtKnob, &sipType_QwtKnob, -1, -1},
+        {sipName_QwtAnalogClock, &sipType_QwtAnalogClock, -1, 18},
+        {sipName_QwtCompass, &sipType_QwtCompass, -1, -1},
+        {sipName_QwtPlotPanner, &sipType_QwtPlotPanner, -1, -1},
+        {sipName_QwtLegendItem, &sipType_QwtLegendItem, -1, -1},
+        {sipName_QwtPlotMagnifier, &sipType_QwtPlotMagnifier, -1, -1},
+        {sipName_QwtPlotPicker, &sipType_QwtPlotPicker, 23, -1},
+        {sipName_QwtPlotZoomer, &sipType_QwtPlotZoomer, -1, -1},
     };
     int i = 0;
-    sipClass = NULL;
+    sipType = NULL;
     do {
         struct class_graph *cg = &graph[i];
         if (cg->name != NULL && sipCpp->inherits(cg->name)) {
-            sipClass = *cg->type;
+            sipType = *cg->type;
             i = cg->yes;
         } else {
             i = cg->no;
