diff --git a/gl/stdinit_.h b/gl/stdint_.h

--- a/gl/stdint_.h	2012-12-26 15:40:09.771486357 -0500
+++ b/gl/stdint_.h	2012-12-26 15:41:45.409931190 -0500
@@ -62,7 +62,8 @@
      int{8,16,32,64}_t, uint{8,16,32,64}_t and __BIT_TYPES_DEFINED__.
      <inttypes.h> also defines intptr_t and uintptr_t.  */
 # define _GL_JUST_INCLUDE_ABSOLUTE_INTTYPES_H
-# include <inttypes.h>
+// This line breaks Clang/LLVM Builds.
+//# include <inttypes.h>
 # undef _GL_JUST_INCLUDE_ABSOLUTE_INTTYPES_H
 #elif @HAVE_SYS_INTTYPES_H@
   /* Solaris 7 <sys/inttypes.h> has the types except the *_fast*_t types, and
@@ -206,10 +207,12 @@
 
 /* 7.18.1.4. Integer types capable of holding object pointers */
 
-#undef intptr_t
-#undef uintptr_t
-#define intptr_t long int
-#define uintptr_t unsigned long int
+/* Comment this out, it breaks Clang/LLVM builds. */
+
+//#undef intptr_t
+//#undef uintptr_t
+//#define intptr_t long int
+//#define uintptr_t unsigned long int
 
 /* 7.18.1.5. Greatest-width integer types */
 
