qt4 doesn't provide different libraries for threading.
--- configure.in.orig	2010-01-12 16:40:06.935594142 +0100
+++ configure.in	2010-01-12 16:40:37.742311509 +0100
@@ -1124,7 +1124,7 @@
     AC_MSG_WARN(cannot find moc. QT parts disabled.)
     HAVE_QT=no
   fi
-  if test X"$HAVE_THREADS" = Xyes; then
+  if false && test X"$HAVE_THREADS" = Xyes; then
     AC_CHECK_LIB(qt-mt, open, qt_mt=yes, qt_mt=no)
     if test X"$qt_mt" = Xno; then
       AC_MSG_WARN(cannot find multi-threaded version of QT library. QT parts disabled.)
--- configure.orig	2010-01-12 16:40:11.555621774 +0100
+++ configure	2010-01-12 16:40:51.125633810 +0100
@@ -4197,7 +4197,7 @@
     echo "configure: warning: cannot find moc. QT parts disabled." 1>&2
     HAVE_QT=no
   fi
-  if test X"$HAVE_THREADS" = Xyes; then
+  if false && test X"$HAVE_THREADS" = Xyes; then
     echo $ac_n "checking for open in -lqt-mt""... $ac_c" 1>&6
 echo "configure:4189: checking for open in -lqt-mt" >&5
 ac_lib_var=`echo qt-mt'_'open | sed 'y%./+-%__p_%'`
