http://www.mico.org/pipermail/mico-devel/2009-April/010285.html

--- configure.in.orig	2009-04-02 12:53:03 +0200
+++ configure.in	2009-04-02 15:34:10 +0200
@@ -732,7 +732,9 @@
     # -mmillicode-long-calls
     CONF_LDFLAGS=-Wl,+s
     CONF_DLFLAGS=-Wl,-E
+    case $target in hppa*)
     CONF_SOEXT=sl
+    ;; esac
     CONF_OBJ_SIZE_LIMIT=yes
     ;;
   *freebsd2*|*freebsd3*)
--- configure.orig	2009-04-02 12:53:06 +0200
+++ configure	2009-04-02 15:31:39 +0200
@@ -3353,7 +3353,9 @@
     # -mmillicode-long-calls
     CONF_LDFLAGS=-Wl,+s
     CONF_DLFLAGS=-Wl,-E
+    case $target in hppa*)
     CONF_SOEXT=sl
+    ;; esac
     CONF_OBJ_SIZE_LIMIT=yes
     ;;
   *freebsd2*|*freebsd3*)
