Index: mediatomb-0.11.0/configure.ac
===================================================================
--- mediatomb-0.11.0.orig/configure.ac
+++ mediatomb-0.11.0/configure.ac
@@ -2340,9 +2340,14 @@ if test "x$FFMPEG_OK" = xyes; then
                 ]
         )
     else 
-        AC_CHECK_HEADER(ffmpeg/avformat.h,
-            [],
+        AC_CHECK_HEADERS([ffmpeg/avformat.h libavformat/avformat.h],
+            [ffmpeg_found=yes; break;], [],
             [
+                #define __STDC_CONSTANT_MACROS
+                #include <stdint.h>
+            ]
+        )
+	if test "x$ffmpeg_found" != "xyes"; then
                 CPPFLAGS="-I$SEARCH_DIR_HEADERS"
                 unset ac_cv_header_ffmpeg_avformat_h
                 AC_CHECK_HEADER($SEARCH_DIR_HEADERS/ffmpeg/avformat.h,
@@ -2357,12 +2362,7 @@ if test "x$FFMPEG_OK" = xyes; then
                         #include <stdint.h>
                     ]
                 )
-            ],
-            [
-                #define __STDC_CONSTANT_MACROS
-                #include <stdint.h>
-            ]
-        )
+	fi
     fi
 fi
  
@@ -2447,7 +2447,11 @@ if test "x$FFMPEG_OK" = xyes; then
             [
                 #define __STDC_CONSTANT_MACROS
                 #include <stdint.h>
+		#ifdef HAVE_LIBAVFORMAT_AVFORMAT_H
+		#include <libavformat/avformat.h>
+		#else
                 #include <ffmpeg/avformat.h>
+		#endif
             ],
             [
                 av_register_all();
@@ -2462,7 +2466,11 @@ if test "x$FFMPEG_OK" = xyes; then
                     #define __STDC_CONSTANT_MACROS
                     #include <stdint.h>
                     extern "C" {
-                        #include <ffmpeg/avformat.h>
+                        #ifdef HAVE_LIBAVFORMAT_AVFORMAT_H
+			#include <libavformat/avformat.h>
+			#else
+                	#include <ffmpeg/avformat.h>
+			#endif
                     }
                 ],
                 [
Index: mediatomb-0.11.0/src/metadata/ffmpeg_handler.cc
===================================================================
--- mediatomb-0.11.0.orig/src/metadata/ffmpeg_handler.cc
+++ mediatomb-0.11.0/src/metadata/ffmpeg_handler.cc
@@ -56,7 +56,11 @@ extern "C" 
 {
 #endif
 
-#include <ffmpeg/avformat.h>
+#ifdef HAVE_LIBAVFORMAT_AVFORMAT_H
+# include <libavformat/avformat.h>
+#else
+# include <ffmpeg/avformat.h>
+#endif
 
 #ifdef FFMPEG_NEEDS_EXTERN_C
 } // extern "C"
