diff --git a/configure.ac b/configure.ac
index fbaa376..a2561c9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1007,7 +1007,7 @@ xyesno)
     else
         # should check these...
         X11_INCLUDES="$X11_INCLUDES $X_CFLAGS"
-        if test "x$HAVE_XF86VIDMODE" == xyes; then
+        if test "x$HAVE_XF86VIDMODE" = xyes; then
            GL_LIB_DEPS="$X_LIBS -lX11 -lXext -lXxf86vm -lXdamage -lXfixes"
 	else
            GL_LIB_DEPS="$X_LIBS -lX11 -lXext -lXdamage -lXfixes"
