diff --git a/configure.ac b/configure.ac
index 21a1986..94b2247 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1777,7 +1777,6 @@ if test "x$with_gallium_drivers" != x; then
         xr300)
             HAVE_GALLIUM_R300=yes
             PKG_CHECK_MODULES([RADEON], [libdrm_radeon >= $LIBDRM_RADEON_REQUIRED])
-            gallium_require_llvm "Gallium R300"
             GALLIUM_DRIVERS_DIRS="$GALLIUM_DRIVERS_DIRS r300"
             gallium_check_st "radeon/drm" "dri-r300" "" "" "xvmc-r300" "vdpau-r300"
             ;;
