--- frontend/common/src/SoundPlayer.cc.orig	2010-06-16 10:49:11.000000000 +0200
+++ frontend/common/src/SoundPlayer.cc	2010-06-16 10:54:04.000000000 +0200
@@ -334,8 +334,11 @@
 void
 SoundPlayer::init()
 {
-  driver->init();
-  register_sound_events();
+  if (driver != NULL)
+  {
+    driver->init();
+    register_sound_events();
+  }
 }
 
 void
@@ -405,6 +408,10 @@
 void
 SoundPlayer::sync_settings()
 {
+  if (driver == NULL)
+    {
+      return;
+    }
   if (driver != NULL)
     {
       for (unsigned int i = 0; i < sizeof(sound_registry)/sizeof(sound_registry[0]); i++)
