--- gcc-3.2/gcc/config/i386/i386.c.athlon-alignment	2002-09-20 13:58:03.000000000 +0200
+++ gcc-3.2/gcc/config/i386/i386.c	2002-09-20 14:03:59.000000000 +0200
@@ -857,7 +857,7 @@ override_options ()
       {&pentium_cost, 0, 0, 16, 7, 16, 7, 16, 1},
       {&pentiumpro_cost, 0, 0, 16, 15, 16, 7, 16, 1},
       {&k6_cost, 0, 0, 32, 7, 32, 7, 32, 1},
-      {&athlon_cost, 0, 0, 16, 7, 64, 7, 16, 1},
+      {&athlon_cost, 0, 0, 16, 7, 16, 7, 16, 1},
       {&pentium4_cost, 0, 0, 0, 0, 0, 0, 0, 1}
     };
 
