--- encfs/BlockFileIO.cpp
+++ encfs/BlockFileIO.cpp
@@ -20,6 +20,7 @@
 #include "MemoryPool.h"
 
 #include <rlog/rlog.h>
+#include <cstring>
 
 #include "i18n.h"
 
--- encfs/BlockNameIO.cpp
+++ encfs/BlockNameIO.cpp
@@ -23,6 +23,7 @@
 #include <rlog/rlog.h>
 #include <rlog/Error.h>
 #include <rlog/RLogChannel.h>
+#include <cstring>
 
 #include "i18n.h"
 
--- encfs/CipherFileIO.cpp
+++ encfs/CipherFileIO.cpp
@@ -23,7 +23,7 @@
 #include <rlog/rlog.h>
 
 #include <fcntl.h>
-#include <errno.h>
+#include <cerrno>
 
 /*
     - Version 2:0 adds support for a per-file initialization vector with a
--- encfs/CipherV3.cpp
+++ encfs/CipherV3.cpp
@@ -22,7 +22,7 @@
 
 #include "../config.h"
 
-#include <string.h>
+#include <cstring>
 
 #define NO_DES
 
--- encfs/Config.cpp
+++ encfs/Config.cpp
@@ -23,6 +23,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <cstring>
 
 
 using namespace std;
--- encfs/ConfigVar.cpp
+++ encfs/ConfigVar.cpp
@@ -17,6 +17,7 @@
 
 #include "ConfigVar.h"
 #include <rlog/rlog.h>
+#include <cstring>
 
 using namespace rlog;
 
--- encfs/DirNode.cpp
+++ encfs/DirNode.cpp
@@ -22,9 +22,10 @@
 
 #include <sys/stat.h>
 #include <sys/types.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
+#include <cerrno>
+#include <cstring>
+#include <cstdio>
+#include <cstdlib>
 #include <pthread.h>
 #include <unistd.h>
 #ifdef linux
--- encfs/FileNode.cpp
+++ encfs/FileNode.cpp
@@ -19,7 +19,8 @@
 // of sys/stat.h or other system headers (to be safe)
 #include "encfs.h"
 
-#include <errno.h>
+#include <cstring>
+#include <cerrno>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <fcntl.h>
--- encfs/FileUtils.cpp
+++ encfs/FileUtils.cpp
@@ -45,9 +45,10 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <ctype.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <errno.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cerrno>
+#include <cstring>
 
 #include <iostream>
 #include <sstream>
--- encfs/MACFileIO.cpp
+++ encfs/MACFileIO.cpp
@@ -22,6 +22,7 @@
 #include <rlog/rlog.h>
 #include <rlog/Error.h>
 #include <rlog/RLogChannel.h>
+#include <cstring>
 
 #include "i18n.h"
 
--- encfs/MACFileIOCompat.cpp
+++ encfs/MACFileIOCompat.cpp
@@ -21,6 +21,7 @@
 
 #include <rlog/rlog.h>
 #include <rlog/Error.h>
+#include <cstring>
 
 #include "i18n.h"
 
--- encfs/MemoryPool.cpp
+++ encfs/MemoryPool.cpp
@@ -19,8 +19,8 @@
 #include "MemoryPool.h"
 #include <rlog/rlog.h>
 
-#include <stdlib.h>
-#include <string.h>
+#include <cstdlib>
+#include <cstring>
 
 #include "config.h"
 #include <pthread.h>
--- encfs/NameIO.cpp
+++ encfs/NameIO.cpp
@@ -26,6 +26,7 @@
 // for static build.  Need to reference the modules which are registered at
 // run-time, to ensure that the linker doesn't optimize them away.
 #include <iostream>
+#include <cstring>
 #include "BlockNameIO.h"
 #include "StreamNameIO.h"
 #include "NullNameIO.h"
--- encfs/NullCipher.cpp
+++ encfs/NullCipher.cpp
@@ -21,6 +21,7 @@
 #include "Interface.h"
 
 #include <rlog/rlog.h>
+#include <cstring>
 
 using namespace std;
 using namespace rel;
--- encfs/NullNameIO.cpp
+++ encfs/NullNameIO.cpp
@@ -19,6 +19,7 @@
 
 #include "Cipher.h"
 #include "base64.h"
+#include <cstring>
 
 using namespace rel;
 
--- encfs/OpaqueValue.cpp
+++ encfs/OpaqueValue.cpp
@@ -20,7 +20,7 @@
 #include "PtrImpl.h"
 #include <rlog/rlog.h>
 
-#include <stdlib.h>
+#include <cstdlib>
 
 using namespace std;
 using namespace rel;
--- encfs/Ptr.cpp
+++ encfs/Ptr.cpp
@@ -20,7 +20,7 @@
 
 #include "Ptr.h"
 
-#include <stdlib.h>
+#include <cstdlib>
 
 using namespace rel;
 
--- encfs/RawFileIO.cpp
+++ encfs/RawFileIO.cpp
@@ -28,7 +28,8 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 
-#include <errno.h>
+#include <cerrno>
+#include <cstring>
 
 using namespace std;
 
--- encfs/SSL_Cipher.cpp
+++ encfs/SSL_Cipher.cpp
@@ -31,7 +31,7 @@
 #include "MemoryPool.h"
 #include "Mutex.h"
 
-#include <string.h>
+#include <cstring>
 
 #include <sys/mman.h>
 
--- encfs/StreamNameIO.cpp
+++ encfs/StreamNameIO.cpp
@@ -22,6 +22,7 @@
 
 #include <rlog/rlog.h>
 #include <rlog/Error.h>
+#include <cstring>
 
 #include "i18n.h"
 
--- encfs/encfs.cpp
+++ encfs/encfs.cpp
@@ -17,12 +17,12 @@
 
 #include "encfs.h"
 
-#include <stdio.h>
-#include <string.h>
+#include <cstdio>
+#include <cstring>
 #include <unistd.h>
 #include <fcntl.h>
 #include <dirent.h>
-#include <errno.h>
+#include <cerrno>
 #include <sys/statvfs.h>
 
 #include <sys/types.h>
--- encfs/encfs.h
+++ encfs/encfs.h
@@ -28,7 +28,7 @@
 #endif
 
 #ifndef linux
-#include <errno.h>
+#include <cerrno>
 
 static __inline int setfsuid(uid_t uid)
 {
--- encfs/main.cpp
+++ encfs/main.cpp
@@ -25,10 +25,10 @@
 #include <sstream>
 
 #include <assert.h>
-#include <stdio.h>
+#include <cstdio>
 #include <unistd.h>
 #include <sys/time.h>
-#include <errno.h>
+#include <cerrno>
 
 #include <getopt.h>
 
--- encfs/readpassphrase.cpp
+++ encfs/readpassphrase.cpp
@@ -36,14 +36,14 @@
 #ifndef HAVE_READPASSPHRASE
 
 #include <signal.h>
-#include <stdio.h>
-#include <errno.h>
+#include <cstdio>
+#include <cerrno>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
 #include <paths.h>
-#include <string.h>
+#include <cstring>
 #include <ctype.h>
 
 #include <termios.h>
--- encfs/test.cpp
+++ encfs/test.cpp
@@ -22,7 +22,7 @@
 
 #include <iostream>
 
-#include <stdlib.h>
+#include <cstdlib>
 
 #include "Cipher.h"
 #include "DirNode.h"
--- intl/autosprintf.cpp
+++ intl/autosprintf.cpp
@@ -17,7 +17,7 @@
    Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
    USA.  */
 
-/* Tell glibc's <stdio.h> to provide a prototype for vasprintf().
+/* Tell glibc's <cstdio> to provide a prototype for vasprintf().
    This must come before <config.h> because <config.h> may include
    <features.h>, and once <features.h> has been included, it's too late.  */
 #ifndef _GNU_SOURCE
@@ -28,10 +28,10 @@
 #include "autosprintf.h"
 
 #include <stdarg.h>
-#include <stdlib.h>
-#include <string.h>
+#include <cstdlib>
+#include <cstring>
 //#include "lib-asprintf.h"
-#include <stdio.h>
+#include <cstdio>
 
 namespace gnu
 {
