From 8c4cdd5304929d19a27f876eb9cb5bbb67a4d59f Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Mon, 29 Nov 2010 21:44:03 +0000
Subject: capplet: Give the dialog a reasonable default height

https://bugzilla.gnome.org/show_bug.cgi?id=635891
---
diff --git a/capplet/gsm-properties-dialog.c b/capplet/gsm-properties-dialog.c
index 4432b09..8b51169 100644
--- a/capplet/gsm-properties-dialog.c
+++ b/capplet/gsm-properties-dialog.c
@@ -784,6 +784,7 @@ gsm_properties_dialog_init (GsmPropertiesDialog *dialog)
                                                      "main-notebook"));
         gtk_box_pack_start (GTK_BOX (content_area), widget, TRUE, TRUE, 0);
 
+        gtk_window_set_default_size (GTK_WINDOW (dialog), -1, 450);
         gtk_window_set_resizable (GTK_WINDOW (dialog), TRUE);
         gtk_container_set_border_width (GTK_CONTAINER (dialog), 6);
         gtk_box_set_spacing (GTK_BOX (content_area), 2);
--
cgit v0.9.0.2
