--- a/modules/text_renderer/freetype.c
+++ b/modules/text_renderer/freetype.c
@@ -67,7 +67,7 @@
 # define DEFAULT_FONT_FILE "/usr/share/fonts/nokia/nosnb.ttf"
 # define DEFAULT_FAMILY "Nokia Sans Bold"
 #else
-# define DEFAULT_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
-# define DEFAULT_FAMILY "Serif Bold"
+# define DEFAULT_FONT_FILE "/usr/share/fonts/dejavu/DejaVuSans.ttf"
+# define DEFAULT_FAMILY "DejaVu Sans"
 #endif
 
