diff -durN vlc-2.1.5/modules/audio_output/oss.c vlc-2.1.5-patched/modules/audio_output/oss.c
--- vlc-2.1.5/modules/audio_output/oss.c	2014-02-27 14:31:53.000000000 +0100
+++ vlc-2.1.5-patched/modules/audio_output/oss.c	2014-08-17 10:01:38.197923781 +0200
@@ -32,11 +32,7 @@
 #include <sys/types.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
-#ifdef HAVE_SOUNDCARD_H
-# include <soundcard.h>
-#else
-# include <sys/soundcard.h>
-#endif
+#include "/usr/lib/oss/include/sys/soundcard.h"
 
 #ifndef SNDCTL_DSP_HALT
 # define SNDCTL_DSP_HALT SNDCTL_DSP_RESET
@@ -209,6 +205,21 @@
     }
     aout_FormatPrepare (fmt);
 
+    /* Select timing */
+    unsigned bytes;
+    if (spdif)
+        bytes = AOUT_SPDIF_SIZE;
+    else
+        bytes = fmt->i_rate / (CLOCK_FREQ / AOUT_MIN_PREPARE_TIME)
+                * fmt->i_bytes_per_frame;
+    if (unlikely(bytes < 16))
+        bytes = 16;
+
+    int frag = (AOUT_MAX_ADVANCE_TIME / AOUT_MIN_PREPARE_TIME) << 16
+             | (32 - clz32(bytes - 1));
+    if (ioctl (fd, SNDCTL_DSP_SETFRAGMENT, &frag) < 0)
+        msg_Err (aout, "cannot set 0x%08x fragment: %m", frag);
+
     sys->fd = fd;
     VolumeSync (aout);
     sys->starting = true;
