diff -uNr a/src/posix/linux_specific.c b/src/posix/linux_specific.c
--- a/src/posix/linux_specific.c	2011-12-08 20:00:28.000000000 +0200
+++ b/src/posix/linux_specific.c	2013-08-02 19:47:52.823858758 +0200
@@ -83,14 +83,14 @@
     unsigned refs;
 } once = { VLC_STATIC_MUTEX, 0 };
 
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
 # include <gnu/libc-version.h>
 # include <stdlib.h>
 #endif
 
 void system_Init (void)
 {
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
     const char *glcv = gnu_get_libc_version ();
 
     /* gettext in glibc 2.5-2.7 is not thread-safe. LibVLC keeps crashing,
