diff -urN modsecurity/config modsecurity.fix/config
--- modsecurity/config	2014-04-15 06:44:04.000000000 -0600
+++ modsecurity.fix/config	2014-06-15 22:37:09.770275079 -0600
@@ -1,11 +1,11 @@
 #!/bin/sh
 
 CFLAGS="$CFLAGS \
-     -I/usr/include/apr-1.0  \
-     -I/usr/include/apr-1.0 -I/usr/include \
+     -I/usr/include/apr-1 \
+     -I/usr/include/apr-1 -I/usr/include \
     -I/usr/include/apache2 \
     -I/usr/include/libxml2 \
-    -DWITH_LUA -I/usr/include/lua5.1   \
+    -DWITH_LUA \
     -DWITH_PCRE_STUDY -DMODSEC_PCRE_MATCH_LIMIT=1500 -DMODSEC_PCRE_MATCH_LIMIT_RECURSION=1500  -DREQUEST_EARLY          \
      \
     -DWITH_YAJL -I/usr/include/yajl  "
@@ -17,7 +17,7 @@
     -I/usr/include/apache2 \
     -L/usr/lib/x86_64-linux-gnu -lcurl \
     -lxml2 \
-    -llua5.1   \
+    -llua   \
     -lpcre \
     -L/usr/lib  -lcap \
     -lyajl  "
