diff --git a/Makefile.top b/Makefile.top
index 9e67a71..64a7b6c 100644
--- a/Makefile.top
+++ b/Makefile.top
@@ -244,7 +244,7 @@ stlport\
 sysui\
 xpdf\
 
-export gb_TAILBUILDTARGET=all slowcheck
+export gb_TAILBUILDTARGET=all
 
 define gbuild_module_rules
 .PHONY: $(1) $(1).all $(1).build $(1).check $(1).clean $(1).subsequentcheck $(1).deliver
@@ -546,7 +546,7 @@ findunusedcode:
               | grep -v ^WSObject \
               > unusedcode.easy
 
-check: dev-install subsequentcheck
+check: unitcheck slowcheck dev-install subsequentcheck
 
 subsequentcheck :| $(if $(filter-out subsequentcheck,$(MAKECMDGOALS)),dev-install)
 	$(GNUMAKE) -j $(CHECK_PARALLELISM) $(GMAKE_OPTIONS) -f Makefile.post $@
diff --git a/solenv/bin/build.pl b/solenv/bin/build.pl
index c8e62ee..11a9069 100755
--- a/solenv/bin/build.pl
+++ b/solenv/bin/build.pl
@@ -1902,7 +1902,7 @@ sub run_job {
             return 1 if (! $path =~ /prj$/ );
             mkpath("$workdir/Logs");
             my $gbuild_flags = '-j' . $ENV{PARALLELISM};
-            my $gbuild_target = 'all slowcheck';
+            my $gbuild_target = 'all';
             if ($registered_name =~ /tail_build\/prj$/ )
             {
                 $gbuild_target = $ENV{gb_TAILBUILDTARGET};
diff --git a/solenv/gbuild/Module.mk b/solenv/gbuild/Module.mk
index 97f1a81..c7ae7f4 100644
--- a/solenv/gbuild/Module.mk
+++ b/solenv/gbuild/Module.mk
@@ -139,7 +139,7 @@ gb_Module_DEVINSTALLHINT=
 debugrun :| build
 endif
 
-all : build unitcheck
+all : build
 	$(info $(gb_Module_BUILDHINT))
 
 build : 
