diff -urN commoncpp2-1.8.1/configure.ac commoncpp2-1.8.1-patched/configure.ac
--- commoncpp2-1.8.1/configure.ac	2010-10-31 18:10:07.000000000 -0400
+++ commoncpp2-1.8.1-patched/configure.ac	2014-07-23 05:53:30.301876037 -0400
@@ -342,7 +342,7 @@
     MODULE_FLAGS="-dynamic -bundle -undefined suppress -flat_namespace -read_only_relocs suppress"
     ;;
 linux*)
-    BASE_LIB="../src/libccgnu2.la"
+    BASE_LIB="../src/libccgnu2.so"
     ;;
 esac
 
