http://bugs.gentoo.org/246747

--- cracklib/python/Makefile.in
+++ cracklib/python/Makefile.in
@@ -57,7 +57,7 @@
 am__installdirs = "$(DESTDIR)$(pyexecdir)" "$(DESTDIR)$(pythondir)"
 pyexecLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(pyexec_LTLIBRARIES)
-_cracklibmodule_la_LIBADD =
+_cracklibmodule_la_LIBADD = -lpython$(PYTHON_VERSION)
 _cracklibmodule_la_SOURCES = _cracklibmodule.c
 _cracklibmodule_la_OBJECTS = _cracklibmodule.lo
 _cracklibmodule_la_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
