diff -Nuar --exclude '*.orig' LVM2.2.02.36.orig/doc/example.conf LVM2.2.02.36/doc/example.conf
--- LVM2.2.02.36.orig/doc/example.conf	2008-04-10 11:50:37.000000000 -0700
+++ LVM2.2.02.36/doc/example.conf	2008-04-29 23:45:08.165059223 -0700
@@ -50,7 +50,9 @@
 
 
     # By default we accept every block device:
-    filter = [ "a/.*/" ]
+    # Gentoo: we exclude /dev/nbd by default, because it makes a lot of kernel
+    # noise when you probed while not available.
+    filter = [ "r|/dev/nbd.*|", "a/.*/" ]
 
     # Exclude the cdrom drive
     # filter = [ "r|/dev/cdrom|" ]
@@ -86,7 +88,7 @@
     # If sysfs is mounted (2.6 kernels) restrict device scanning to 
     # the block devices it believes are valid.
     # 1 enables; 0 disables.
-    sysfs_scan = 1	
+    sysfs_scan = 1
 
     # By default, LVM2 will ignore devices used as components of
     # software RAID (md) devices by looking for md superblocks.
@@ -219,12 +221,14 @@
     # e.g. vgscan.lvm1 and they will stop working after you start using
     # the new lvm2 on-disk metadata format.
     # The default value is set when the tools are built.
-    # fallback_to_lvm1 = 0
+    # Gentoo: the LVM tools are a seperate package.
+    fallback_to_lvm1 = 0
 
     # The default metadata format that commands should use - "lvm1" or "lvm2".
     # The command line override is -M1 or -M2.
     # Defaults to "lvm1" if compiled in, else "lvm2".
-    # format = "lvm1"
+    # Gentoo: default to LVM2 format
+    format = "lvm2"
 
     # Location of proc filesystem
     proc = "/proc"
@@ -351,12 +355,12 @@
 
 # Metadata settings
 #
-# metadata {
+metadata {
     # Default number of copies of metadata to hold on each PV.  0, 1 or 2.
     # You might want to override it from the command line with 0 
     # when running pvcreate on new PVs which are to be added to large VGs.
-
-    # pvmetadatacopies = 1
+    # Gentoo: enabled for data safety
+    pvmetadatacopies = 2
 
     # Approximate default size of on-disk metadata areas in sectors.
     # You should increase this if you have large volume groups or
@@ -378,11 +382,11 @@
     # the supplied toolset to make changes (e.g. vgcfgrestore).
 
     # dirs = [ "/etc/lvm/metadata", "/mnt/disk2/lvm/metadata2" ]
-#}
+}
 
 # Event daemon
-#
-# dmeventd {
+# Gentoo: uncommented, since we provide the dmeventd init script
+dmeventd {
     # mirror_library is the library used when monitoring a mirror device.
     #
     # "libdevmapper-event-lvm2mirror.so" attempts to recover from
@@ -390,7 +394,7 @@
     # reconfigures a mirror as necessary. If no mirror library is
     # provided, mirrors are not monitored through dmeventd.
 
-    # mirror_library = "libdevmapper-event-lvm2mirror.so"
+    mirror_library = "libdevmapper-event-lvm2mirror.so"
 
     # snapshot_library is the library used when monitoring a snapshot device.
     #
@@ -399,6 +403,6 @@
     # snapshot exceedes 80%. The warning is repeated when 85%, 90% and
     # 95% of the snapshot are filled.
 
-    # snapshot_library = "libdevmapper-event-lvm2snapshot.so"
-#}
+    snapshot_library = "libdevmapper-event-lvm2snapshot.so"
+}
 
