diff -Nuar LVM2.2.02.56.orig/daemons/dmeventd/Makefile.in LVM2.2.02.56/daemons/dmeventd/Makefile.in
--- LVM2.2.02.56.orig/daemons/dmeventd/Makefile.in	2009-11-13 12:48:01.000000000 +0000
+++ LVM2.2.02.56/daemons/dmeventd/Makefile.in	2009-11-29 23:50:22.873898621 +0000
@@ -60,7 +60,7 @@
 	$(RM) -f $@
 	$(LN_S) $(LIB_SHARED) $@
 
-dmeventd: $(LIB_SHARED) $(VERSIONED_SHLIB) dmeventd.o
+dmeventd: $(LIB_SHARED) $(LIB_STATIC) $(VERSIONED_SHLIB) dmeventd.o
 	$(CC) $(CFLAGS) $(LDFLAGS) -L. -o $@ dmeventd.o \
 	$(LVMLIBS) $(LIBS) -rdynamic
 
