diff -Nuar LVM2.2.02.51.orig/tools/Makefile.in LVM2.2.02.51/tools/Makefile.in
--- LVM2.2.02.51.orig/tools/Makefile.in	2009-11-09 21:31:21.785012413 +0000
+++ LVM2.2.02.51/tools/Makefile.in	2009-11-09 21:40:50.157475714 +0000
@@ -109,13 +109,13 @@
 	$(CC) $(CFLAGS) $(LDFLAGS) \
 	      -L$(top_srcdir)/libdm \
 		  -o $@ dmsetup.o \
-		  -ldevmapper $(LIB_PTHREAD)
+		  -ldevmapper $(LIBS) $(LIB_PTHREAD)
 
 dmsetup.static: dmsetup.o $(interfacedir)/libdevmapper.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -static \
 	      -L$(interfacedir) \
 		  -o $@ dmsetup.o \
-		  -ldevmapper $(LIB_PTHREAD)
+		  -ldevmapper $(LIBS) $(LIB_PTHREAD)
 
 all: device-mapper
 
