--- daemons/dmeventd/Makefile.in	2009-03-22 04:01:26.754802410 -0400
+++ daemons/dmeventd/Makefile.in.new	2009-03-22 04:01:14.693689634 -0400
@@ -42,7 +42,7 @@
 LDFLAGS += -ldl -ldevmapper -lpthread
 CLDFLAGS += -ldl -ldevmapper -lpthread
 
-dmeventd: $(LIB_SHARED) $(VERSIONED_SHLIB) dmeventd.o
+dmeventd: $(LIB_SHARED) $(LIB_STATIC) $(VERSIONED_SHLIB) dmeventd.o
 	$(CC) -o $@ dmeventd.o $(CFLAGS) $(LDFLAGS) \
 	-L. -ldevmapper-event $(LIBS) -rdynamic
 
