diff -urN LVM2.2.02.63/liblvm/liblvm2app.pc.in LVM2.2.02.63-patched/liblvm/liblvm2app.pc.in
--- LVM2.2.02.63/liblvm/liblvm2app.pc.in	2010-04-19 00:36:16.000000000 -0600
+++ LVM2.2.02.63-patched/liblvm/liblvm2app.pc.in	2009-07-22 19:41:53.000000000 -0600
@@ -8,4 +8,4 @@
 Version: @LVM_MAJOR@.@LVM_LIBAPI@
 Cflags: -I${includedir} 
 Libs: -L${libdir} -llvm2app
-Libs.private: -ldevmapper
+Libs.private: -ldevmapper -ldl
diff -urN LVM2.2.02.63/liblvm/Makefile.in LVM2.2.02.63-patched/liblvm/Makefile.in
--- LVM2.2.02.63/liblvm/Makefile.in	2010-04-19 00:37:24.000000000 -0600
+++ LVM2.2.02.63-patched/liblvm/Makefile.in	2010-04-09 15:42:51.000000000 -0600
@@ -36,7 +36,7 @@
 
 include $(top_builddir)/make.tmpl
 
-LIBS += $(LVMINTERNAL_LIBS) -ldevmapper
+LIBS += $(LVMINTERNAL_LIBS) -ldevmapper -ldl
 
 ifeq ("@DMEVENTD@", "yes")
   LIBS += -ldevmapper-event
