--- orig/deadbeef.desktop.in	2012-10-18 13:58:43.000000000 +0400
+++ fixed/deadbeef.desktop.in	2012-10-18 13:58:08.000000000 +0400
@@ -12,29 +12,3 @@
 Terminal=false
 Type=Application
 Icon=deadbeef
-
-X-Ayatana-Desktop-Shortcuts=Play;Pause;Stop;Next;Prev
-[Play Shortcut Group]
-Name=Play
-Exec=deadbeef --play
-TargetEnvironment=Unity
-
-[Pause Shortcut Group]
-Name=Pause
-Exec=deadbeef --pause
-TargetEnvironment=Unity
-
-[Stop Shortcut Group]
-Name=Stop
-Exec=deadbeef --stop
-TargetEnvironment=Unity
-
-[Next Shortcut Group]
-Name=Next
-Exec=deadbeef --next
-TargetEnvironment=Unity
-
-[Prev Shortcut Group]
-Name=Prev
-Exec=deadbeef --prev
-TargetEnvironment=Unity
