diff -ur xvba-driver-0.8.0.orig/src/shaders/pso2h.py xvba-driver-0.8.0/src/shaders/pso2h.py
--- xvba-driver-0.8.0.orig/src/shaders/pso2h.py	2013-01-30 22:42:11.509420966 -0500
+++ xvba-driver-0.8.0/src/shaders/pso2h.py	2013-01-30 23:01:12.113423626 -0500
@@ -46,7 +46,7 @@
 
 def define(format, filename):
   path, file = os.path.split(filename)
-  return format % string.upper(file.replace('.', '_').capitalize())
+  return format % str(file.replace('.', '_').capitalize()).upper()
 
 class PSO:
   def __init__(self, filename=None, name=None):
@@ -65,7 +65,7 @@
       # skip comments
       if line.startswith('#'):
         continue
-      line = string.strip(line)
+      line = str(line).strip()
       line += '\\n'
       size += len(line) - 1;
       body += "  \"%s\",\n" % line
@@ -79,7 +79,7 @@
                               __HEADER__))
 
 def usage():
-  print __doc__
+  print(__doc__)
 
 def main(argv):
   opt_shader = None
@@ -107,11 +107,11 @@
 
   #input validation
   if not opt_shader:
-    print "error: you must supply a shader file.\n"
+    print("error: you must supply a shader file.\n")
     usage()
     sys.exit(1)
   if not os.access(opt_shader, os.F_OK):
-    print opt_shader + ": file not found"
+    print(opt_shader + ": file not found")
     sys.exit(1)
   file, ext = os.path.splitext(opt_shader)
   if not opt_header:
