diff -urN xvba-driver-0.8.0/src/xvba_dump.c xvba-driver-0.8.0-fixed/src/xvba_dump.c
--- xvba-driver-0.8.0/src/xvba_dump.c	2012-11-23 13:29:10.000000000 +0000
+++ xvba-driver-0.8.0-fixed/src/xvba_dump.c	2014-03-02 05:38:37.455775710 +0000
@@ -150,8 +150,6 @@
         _(VAEncSequenceParameterBufferType);
         _(VAEncPictureParameterBufferType);
         _(VAEncSliceParameterBufferType);
-        _(VAEncH264VUIBufferType);
-        _(VAEncH264SEIBufferType);
 #endif
 #undef _
     }
