--- a/trunk/libgphoto2/configure.ac
+++ b/trunk/libgphoto2/configure.ac
@@ -273,22 +273,14 @@
 GP_CONFIG_MSG([JPEG mangling support],[${libjpeg_msg}])
 
 dnl ---------------------------------------------------------------------------
-dnl check for libxml22
-dnl ---------------------------------------------------------------------------
-LIBXML2=""
-libxml2_msg="no"
-AC_SUBST(LIBXML2)
-AC_ARG_WITH([libxml2], AS_HELP_STRING([--without-libxml2], [Build without libxml2 library (default: no)]))
-AS_IF([test "x$with_libxml2" != "xno"], [
-	AC_CHECK_LIB(xml2,xmlFirstElementChild,[
-		AC_CHECK_HEADER(libxml/parser.h,[
-			AC_DEFINE(HAVE_LIBXML2,1,[define if we found LIBXML2 and its headers])
-			LIBXML2="-lxml2"
-			libxml2_msg="yes"
-		])
-	])
-])
-GP_CONFIG_MSG([LIBXML2 to support Olympus E remote control],[${libxml2_msg}])
+dnl check for libxml2
+dnl ---------------------------------------------------------------------------
+GP_CONFIG_MSG([libxml2 support])
+GP_CHECK_LIBRARY([LIBXML2],[libxml-2.0],[>= 2.0],
+                [libxml/parser.h],[xmlFirstElementChild],[],[],
+                [default-on],
+                [http://xmlsoft.org]
+)
