http://bugs.gentoo.org/336598

--- libgphoto2_port/m4/gp-libltdl.m4
+++ libgphoto2_port/m4/gp-libltdl.m4
@@ -23,7 +23,7 @@
 
 Please make sure that the proper development package is installed
 (libltdl-dev, libtool-ltdl-devel, etc.)
-])dnl
+])[]dnl
 ])dnl
 ])
 else
--- m4m/gp-libltdl.m4
+++ m4m/gp-libltdl.m4
@@ -23,7 +23,7 @@
 
 Please make sure that the proper development package is installed
 (libltdl-dev, libtool-ltdl-devel, etc.)
-])dnl
+])[]dnl
 ])dnl
 ])
 else
