Fix sandbox issue.

Signed-off-by: Lars Wendler <polynomial-c@gentoo.org>

--- gawk-4.1.1/extension/Makefile.am	2014-02-20 19:42:25.000000000 +0100
+++ gawk-4.1.1/extension/Makefile.am	2014-02-20 19:42:25.000000000 +0100
@@ -100,7 +100,7 @@
 
 install-data-hook:
 	for i in $(pkgextension_LTLIBRARIES) ; do \
-		$(RM) $(pkgextensiondir)/$$i ; \
+		$(RM) $(DESTDIR)$(pkgextensiondir)/$$i ; \
 	done
 
 # Keep the uninstall check working:
--- gawk-4.1.1/extension/Makefile.in	2014-04-08 18:13:03.000000000 +0200
+++ gawk-4.1.1/extension/Makefile.in	2014-04-08 18:13:03.000000000 +0200
@@ -1231,7 +1231,7 @@
 
 install-data-hook:
 	for i in $(pkgextension_LTLIBRARIES) ; do \
-		$(RM) $(pkgextensiondir)/$$i ; \
+		$(RM) $(DESTDIR)$(pkgextensiondir)/$$i ; \
 	done
 
 # Keep the uninstall check working:
