commit 495f586d03ba6974cc46b933ccabe2424c42eb0a
Author: William Pitcock <nenolod@dereferenced.org>
Date:   Sat May 12 17:54:47 2012 -0500

    stdinc: remove alloca.h (issue #19)

diff --git a/stdinc.h b/stdinc.h
index 0262c5b..f010f8d 100644
--- a/stdinc.h
+++ b/stdinc.h
@@ -32,9 +32,7 @@
 #include <string.h>
 #include <libgen.h>
 
-#ifndef _WIN32
-# include <alloca.h>
-#else
+#ifdef _WIN32
 # define WIN32_LEAN_AND_MEAN
 # include <windows.h>
 # include <malloc.h>
