--- gcc-3.4.3-ssp/gcc/gcc.c	2004-11-14 21:17:58.585438344 -0500
+++ gcc-3.4.3-ssp-libssp/gcc/gcc.c	2004-11-21 13:28:43.699379520 -0500
@@ -711,7 +711,17 @@
 static const char *link_gcc_c_sequence_spec = LINK_GCC_C_SEQUENCE_SPEC;
 static const char *asm_spec = ASM_SPEC;
 static const char *asm_final_spec = ASM_FINAL_SPEC;
+
+#if defined(_LIBSSP_PROVIDES_SSP_)
+#if defined(EFAULT_PIE_SSP) || defined(EFAULT_SSP)
+static const char *link_spec = LINK_SPEC " %{!fno-stack-protector|!fno-stack-protector-all:-lssp %{static: -lc}}";
+#else
+static const char *link_spec = LINK_SPEC " %{fstack-protector|fstack-protector-all:-lssp %{static: -lc}}";
+#endif // defined(EFAULT_PIE_SSP) || defined(EFAULT_SSP)
+#else 
 static const char *link_spec = LINK_SPEC;
+#endif // defined(_LIBSSP_PROVIDES_SSP_)
+
 static const char *lib_spec = LIB_SPEC;
 static const char *libgcc_spec = LIBGCC_SPEC;
 static const char *endfile_spec = ENDFILE_SPEC;
diff -Nru gcc-3.4.3-ssp/gcc/libgcc-std.ver gcc-3.4.3-ssp-libssp/gcc/libgcc-std.ver
--- gcc-3.4.3-ssp/gcc/libgcc-std.ver	2004-11-14 21:18:00.004222656 -0500
+++ gcc-3.4.3-ssp-libssp/gcc/libgcc-std.ver	2004-11-15 19:22:11.802713352 -0500
@@ -175,7 +175,7 @@
   _Unwind_SjLj_ForcedUnwind
   _Unwind_SjLj_Resume
 
-%if !defined(_LIBC_PROVIDES_SSP_)
+%if !defined(_LIBC_PROVIDES_SSP_) && !defined(_LIBSSP_PROVIDES_SSP_)
   # stack smash handler symbols
   __guard
   __stack_smash_handler
diff -Nru gcc-3.4.3-ssp/gcc/libgcc2.c gcc-3.4.3-ssp-libssp/gcc/libgcc2.c
--- gcc-3.4.3-ssp/gcc/libgcc2.c	2004-11-14 21:18:00.004222656 -0500
+++ gcc-3.4.3-ssp-libssp/gcc/libgcc2.c	2004-11-15 19:24:58.428382400 -0500
@@ -1680,7 +1680,7 @@
 
 
 #ifdef L_stack_smash_handler
-#ifndef _LIBC_PROVIDES_SSP_
+#if !defined(_LIBC_PROVIDES_SSP_) && !defined(_LIBSSP_PROVIDES_SSP_)
 #include <stdio.h>
 #include <string.h>
 #include <fcntl.h>
@@ -1797,5 +1797,5 @@
 #endif
   _exit (127);
 }
-#endif /* _LIBC_PROVIDES_SSP_ */
+#endif /* _LIBC_PROVIDES_SSP_ && _LIBSSP_PROVIDES_SSP_ */
 #endif /* L_stack_smash_handler */
