diff -urN libtorrent-rasterbar-0.16.18/bindings/python/Makefile.am libtorrent-rasterbar-0.16.18-patched/bindings/python/Makefile.am
--- libtorrent-rasterbar-0.16.18/bindings/python/Makefile.am	2013-05-11 15:01:08.000000000 -0400
+++ libtorrent-rasterbar-0.16.18-patched/bindings/python/Makefile.am	2014-10-23 09:39:06.632219413 -0400
@@ -8,6 +8,7 @@
   make_torrent.py           \
   src/alert.cpp             \
   src/big_number.cpp        \
+  src/bytes.hpp             \
   src/converters.cpp        \
   src/create_torrent.cpp    \
   src/datetime.cpp          \
diff -urN libtorrent-rasterbar-0.16.18/bindings/python/src/bytes.hpp libtorrent-rasterbar-0.16.18-patched/bindings/python/src/bytes.hpp
--- libtorrent-rasterbar-0.16.18/bindings/python/src/bytes.hpp	1969-12-31 19:00:00.000000000 -0500
+++ libtorrent-rasterbar-0.16.18-patched/bindings/python/src/bytes.hpp	2014-10-23 09:38:12.087216727 -0400
@@ -0,0 +1,18 @@
+// Copyright Arvid Norberg 2006-2013. Use, modification and distribution is
+// subject to the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BYTES_HPP
+#define BYTES_HPP
+
+#include <string>
+
+struct bytes
+{
+    bytes(std::string const& s): arr(s) {}
+    bytes() {}
+    std::string arr;
+};
+
+#endif
+
