--- a/appargs.cpp
+++ b/appargs.cpp
@@ -19,6 +19,7 @@
  * 
  */
 
+#include <string.h>
 
 #include "strtok.h"
 
--- a/icd-main.cpp
+++ b/icd-main.cpp
@@ -23,6 +23,7 @@
 #include <errno.h>
 #include <sys/poll.h>
 #include <sys/stat.h>
+#include <string.h>
 
 #include "inotify-cxx.h"
 #include "appinst.h"
--- a/ict-main.cpp
+++ b/ict-main.cpp
@@ -25,6 +25,7 @@
 #include <fcntl.h>
 #include <stdlib.h>
 #include <limits.h>
+#include <string.h>
 
 #include "inotify-cxx.h"
 #include "appargs.h"
--- a/incroncfg.cpp
+++ b/incroncfg.cpp
@@ -17,6 +17,8 @@
 #include <fstream>
 #include <sstream>
 
+#include <string.h>
+
 #include "incroncfg.h"
 
 
--- a/usertable.cpp
+++ b/usertable.cpp
@@ -25,6 +25,7 @@
 #include <grp.h>
 #include <stdlib.h>
 #include <sys/stat.h>
+#include <string.h>
 
 #include "usertable.h"
 #include "incroncfg.h"
