diff -ur incron-0.5.8.orig/appargs.cpp incron-0.5.8/appargs.cpp
--- incron-0.5.8.orig/appargs.cpp	2008-12-29 18:31:09.000000000 +0200
+++ incron-0.5.8/appargs.cpp	2009-07-30 14:40:39.000000000 +0300
@@ -19,6 +19,7 @@
  * 
  */
 
+#include <stdio.h>
 
 #include <cstring>
 
diff -ur incron-0.5.8.orig/icd-main.cpp incron-0.5.8/icd-main.cpp
--- incron-0.5.8.orig/icd-main.cpp	2008-12-29 18:31:09.000000000 +0200
+++ incron-0.5.8/icd-main.cpp	2009-07-30 14:38:36.000000000 +0300
@@ -13,6 +13,7 @@
  *  
  */
 
+#include <stdio.h>
 #include <map>
 #include <signal.h>
 #include <wait.h>
diff -ur incron-0.5.8.orig/incroncfg.cpp incron-0.5.8/incroncfg.cpp
--- incron-0.5.8.orig/incroncfg.cpp	2008-12-29 18:31:09.000000000 +0200
+++ incron-0.5.8/incroncfg.cpp	2009-07-30 14:41:31.000000000 +0300
@@ -181,7 +181,7 @@
 
 bool IncronCfg::IsComment(const char* s)
 {
-  char* sx = strchr(s, '#');
+  const char* sx = strchr(s, '#');
   if (sx == NULL)
     return false;
     
diff -ur incron-0.5.8.orig/inotify-cxx.cpp incron-0.5.8/inotify-cxx.cpp
--- incron-0.5.8.orig/inotify-cxx.cpp	2008-12-29 18:31:09.000000000 +0200
+++ incron-0.5.8/inotify-cxx.cpp	2009-07-30 14:39:06.000000000 +0300
@@ -19,7 +19,7 @@
  * 
  */
  
-
+#include <stdio.h>
 #include <errno.h>
 #include <unistd.h>
 #include <fcntl.h>
diff -ur incron-0.5.8.orig/usertable.cpp incron-0.5.8/usertable.cpp
--- incron-0.5.8.orig/usertable.cpp	2008-12-29 18:31:09.000000000 +0200
+++ incron-0.5.8/usertable.cpp	2009-07-30 14:38:50.000000000 +0300
@@ -16,7 +16,7 @@
  * 
  */
 
-
+#include <stdio.h>
 #include <pwd.h>
 #include <syslog.h>
 #include <errno.h>
