--- src/acpustatus.cc	2014-09-04 20:37:29.093962217 +0400
+++ src/acpustatus.cc	2014-09-04 20:39:15.847297073 +0400
@@ -25,17 +25,12 @@
 #include "sysdep.h"
 #include "default.h"
 
-#if defined(linux)
-//#include <linux/kernel.h>
-#include <sys/sysinfo.h>
-#endif
 #if defined(sun) && defined(SVR4)
 #include <sys/loadavg.h>
 #endif
 #ifdef HAVE_KSTAT_H
 #include <sys/resource.h>
 #include <kstat.h>
-#include <sys/sysinfo.h>
 #endif
 
 #ifdef HAVE_SYS_PARAM_H
@@ -56,6 +51,10 @@
 #include <dirent.h>
 #include "intl.h"
 
+#if defined(linux) || defined(HAVE_KSTAT_H)
+#include <sys/sysinfo.h>
+#endif
+
 #if (defined(linux) || defined(HAVE_KSTAT_H)) || defined(HAVE_SYSCTL_CP_TIME)
 
 extern ref<YPixmap> taskbackPixmap;
