--- orig/icewm-1.2.21/src/acpustatus.cc	2005-05-31 14:23:48 -0500
+++ work/icewm-1.2.21/src/acpustatus.cc	2005-08-03 18:29:29 -0500
@@ -25,17 +25,12 @@
 #include "sysdep.h"
 #include "default.h"
 
-#if defined(linux)
-//#include <linux/kernel.h>
-#include <sys/sysinfo.h>
-#endif
 #if defined(sun) && defined(SVR4)
 #include <sys/loadavg.h>
 #endif
 #ifdef HAVE_KSTAT_H
 #include <sys/resource.h>
 #include <kstat.h>
-#include <sys/sysinfo.h>
 #endif
 
 #ifdef HAVE_SYS_PARAM_H
@@ -54,6 +49,10 @@
 #include <sched.h>
 #endif
 
+#if defined(linux) || defined(HAVE_KSTAT_H)
+#include <sys/sysinfo.h>
+#endif
+
 #include "intl.h"
 
 #if (defined(linux) || defined(HAVE_KSTAT_H)) || defined(HAVE_SYSCTL_CP_TIME)
