--- icewm-1.3.7/src/decorate.cc.old	2012-06-10 10:59:22.496484163 +0300
+++ icewm-1.3.7/src/decorate.cc	2012-06-10 11:00:05.453078780 +0300
@@ -320,6 +320,7 @@
     YWindow::configure(r);
 
     performLayout();
+	sendConfigure();
 }
 
 void YFrameWindow::performLayout()
@@ -331,8 +332,6 @@
     layoutShape();
     if (affectsWorkArea())
         manager->updateWorkArea();
-
-    sendConfigure();
 }
 
 void YFrameWindow::layoutTitleBar() {
