--- ipmimonitoring/src/ipmimonitoring/Makefile.am.orig	2008-04-02 16:54:29.000000000 +0200
+++ ipmimonitoring/src/ipmimonitoring/Makefile.am	2008-04-02 16:55:00.000000000 +0200
@@ -23,8 +23,8 @@
 	../../../common/src/libllnlcommon.la \
 	../libipmimonitoring/libipmimonitoring.la \
 	../../../common/src/libpstdout.la \
-	../../../ipmidetect/src/libipmidetect/libipmidetect.la \
 	../../../common/src/libhostrange.la \
+	../../../ipmidetect/src/libipmidetect/libipmidetect.la \
 	../../../common/src/libtoolcommon.la \
 	../../../common/src/libcbuf.la
 
