diff -urN linux-2.6.32/Makefile linux-2.6.32.gcc//Makefile
--- linux-2.6.32/Makefile	2011-12-27 01:18:30.858772823 +0000
+++ linux-2.6.32.gcc//Makefile	2011-12-27 01:19:47.321762282 +0000
@@ -219,8 +219,8 @@
 	  else if [ -x /bin/bash ]; then echo /bin/bash; \
 	  else echo sh; fi ; fi)
 
-HOSTCC       = gcc
-HOSTCXX      = g++
+HOSTCC       = gcc-4.4.5
+HOSTCXX      = g++-4.4.5
 HOSTCFLAGS   = -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 -fomit-frame-pointer
 HOSTCXXFLAGS = -O2
 
@@ -306,7 +306,7 @@
 
 AS		= $(CROSS_COMPILE)as
 LD		= $(CROSS_COMPILE)ld
-CC		= $(CROSS_COMPILE)gcc
+CC		= $(CROSS_COMPILE)gcc-4.4.5
 CPP		= $(CC) -E
 AR		= $(CROSS_COMPILE)ar
 NM		= $(CROSS_COMPILE)nm
