Fix parallel make problems.
Upstream changesets 4547, 4552.
--- gentpl.py
+++ gentpl.py
@@ -487,7 +487,7 @@
 def manpage():
     r  = "if COND_MAN_PAGES\n"
     r += gvar_add("man_MANS", "[+ name +].[+ mansection +]\n")
-    r += rule("[+ name +].[+ mansection +]", "[+ name +]", """
+    r += rule("[+ name +].[+ mansection +]", "[+ name +] grub-mkconfig_lib", """
 chmod a+x [+ name +]
 PATH=$(builddir):$$PATH pkgdatadir=$(builddir) $(HELP2MAN) --section=[+ mansection +] -i $(top_srcdir)/docs/man/[+ name +].h2m -o $@ [+ name +]
 """)
--- grub-core/Makefile.am
+++ grub-core/Makefile.am
@@ -63,7 +63,7 @@
 rs_decoder.S: $(srcdir)/lib/reed_solomon.c
 	$(TARGET_CC) $(TARGET_CPPFLAGS) $(TARGET_CFLAGS) -Os -I$(top_builddir) -S -DSTANDALONE -o $@ $< -g0 -mregparm=3 -ffreestanding
 
-kern/i386/pc/startup.S: $(builddir)/rs_decoder.S
+boot/i386/pc/startup_raw.S: $(builddir)/rs_decoder.S
 boot/mips/loongson/fwstart.S: $(builddir)/sm712_start.S
 
 CLEANFILES += grub_script.yy.c grub_script.yy.h
