diff --git a/genlop b/genlop
index 51b6f35..9ae5010 100755
--- a/genlop
+++ b/genlop
@@ -982,6 +982,21 @@ sub rsync() {
 					print "     rsync'ed at >>> ", colored((scalar localtime "$1"), $COLORS{'green'}),"\n";
 				}
 			}
+			
+			if ($_ =~ m/^(.*?)\: \>\>\> Git pull in .*successful$/) {
+				if ($date_found) {
+					if (datecompare($1) <= 0) {
+						next;
+					}
+				}
+
+				if ($gmt_found) {
+					print "     Git pulled at >>> ", colored((scalar gmtime "$1"), $COLORS{'green'}),"\n";
+				}
+				else {
+					print "     Git pulled at >>> ", colored((scalar localtime "$1"), $COLORS{'green'}),"\n";
+				}
+			}
 		}
 		close($handle);
 	}
