--- a/pym/portage/repository/config.py	2016-06-01 19:54:34.000000000 +0000
+++ b/pym/portage/repository/config.py	2017-03-27 22:46:37.816400937 +0000
@@ -80,7 +80,7 @@
 		'find_invalid_path_char', 'force', 'format', 'local_config', 'location',
 		'main_repo', 'manifest_hashes', 'masters', 'missing_repo_name',
 		'name', 'portage1_profiles', 'portage1_profiles_compat', 'priority',
-		'profile_formats', 'sign_commit', 'sign_manifest',
+		'profile_formats', 'sign_commit', 'sign_manifest', 'sync_branch',
 		'sync_depth', 'sync_hooks_only_on_change',
 		'sync_type', 'sync_umask', 'sync_uri', 'sync_user', 'thin_manifest',
 		'update_changelog', '_eapis_banned', '_eapis_deprecated',
@@ -174,6 +174,12 @@
 
 		self.module_specific_options = {}
 
+		#allow repos.conf to specify which branch to sync/use
+		sync_branch = repo_opts.get('sync-branch')
+		if sync_branch is not None:
+			sync_branch = sync_branch.strip()
+		self.sync_branch = sync_branch or None
+
 		# Not implemented.
 		format = repo_opts.get('format')
 		if format is not None:
--- a/pym/portage/sync/modules/git/git.py	2017-03-27 23:49:48.196282923 +0000
+++ b/pym/portage/sync/modules/git/git.py	2017-03-27 23:54:42.624273756 +0000
@@ -56,6 +56,8 @@
 			git_cmd_opts += " --depth %d" % self.repo.sync_depth
 		if self.repo.module_specific_options.get('sync-git-clone-extra-opts'):
 			git_cmd_opts += " %s" % self.repo.module_specific_options['sync-git-clone-extra-opts']
+		if self.repo.sync_branch is not None:
+			git_cmd_opts += " -b %s" % self.repo.sync_branch
 		git_cmd = "%s clone%s %s ." % (self.bin_command, git_cmd_opts,
 			portage._shell_quote(sync_uri))
 		writemsg_level(git_cmd + "\n")
