diff -urN LVM2.2.02.65/make.tmpl.in LVM2.2.02.65-patched//make.tmpl.in
--- LVM2.2.02.65/make.tmpl.in	2010-05-14 07:32:36.000000000 -0600
+++ LVM2.2.02.65-patched//make.tmpl.in	2010-05-20 08:39:57.000000000 -0600
@@ -302,7 +302,7 @@
 install_lib_shared: $(LIB_SHARED)
 	$(INSTALL_PROGRAM) -D $< $(libdir)/$(<F).$(LIB_VERSION)
 	$(INSTALL) -d $(usrlibdir)
-	$(LN_S) -f $(USRLIB_RELPATH)$(<F).$(LIB_VERSION) $(usrlibdir)/$(<F)
+	$(LN_S) -f $(USRLIB_RELPATH)$(<F).$(LIB_VERSION) $(<F)
 
 # FIXME: plugins are installed to subdirs 
 #        and for compatibility links in libdir are created
diff -urN LVM2.2.02.65/tools/Makefile.in LVM2.2.02.65-patched//tools/Makefile.in
--- LVM2.2.02.65/tools/Makefile.in	2010-05-11 02:34:38.000000000 -0600
+++ LVM2.2.02.65-patched//tools/Makefile.in	2010-05-20 08:41:43.000000000 -0600
@@ -172,7 +172,7 @@
 install_cmdlib_dynamic: liblvm2cmd.$(LIB_SUFFIX)
 	$(INSTALL_PROGRAM) -D $< $(libdir)/$(<F).$(LIB_VERSION)
 	$(INSTALL) -d $(usrlibdir)
-	$(LN_S) -f $(USRLIB_RELPATH)$(<F).$(LIB_VERSION) $(usrlibdir)/$(<F)
+	$(LN_S) -f $(USRLIB_RELPATH)$(<F).$(LIB_VERSION) $(<F)
 
 install_cmdlib_static: liblvm2cmd-static.a
 	$(INSTALL_DATA) -D $< $(usrlibdir)/liblvm2cmd.a
