Index: lvm2-2.02.39/lib/device/dev-md.c
===================================================================
--- lvm2-2.02.39.orig/lib/device/dev-md.c	2009-02-23 21:35:16.000000000 +0000
+++ lvm2-2.02.39/lib/device/dev-md.c	2009-02-23 21:34:02.000000000 +0000
@@ -93,7 +93,7 @@
 	if (size < MD_RESERVED_SECTORS * 2)
 		return 0;
 
-	if (!dev_open(dev)) {
+	if (!dev_open_flags(dev, O_RDONLY, 1, 0)) {
 		stack;
 		return -1;
 	}
Index: lvm2-2.02.39/lib/device/device.c
===================================================================
--- lvm2-2.02.39.orig/lib/device/device.c	2009-02-23 21:35:19.000000000 +0000
+++ lvm2-2.02.39/lib/device/device.c	2009-02-23 21:34:02.000000000 +0000
@@ -57,7 +57,7 @@
 	uint16_t *part_magic;
 	struct partition *part;
 
-	if (!dev_open(dev)) {
+	if (!dev_open_flags(dev, O_RDONLY, 1, 0)) {
 		stack;
 		return -1;
 	}
Index: lvm2-2.02.39/lib/format1/disk-rep.c
===================================================================
--- lvm2-2.02.39.orig/lib/format1/disk-rep.c	2009-02-23 21:35:24.000000000 +0000
+++ lvm2-2.02.39/lib/format1/disk-rep.c	2009-02-23 21:34:02.000000000 +0000
@@ -415,7 +415,7 @@
 {
 	struct disk_list *dl;
 
-	if (!dev_open(dev))
+	if (!dev_open_flags(dev, O_RDONLY, 1, 0))
 		return_NULL;
 
 	dl = __read_disk(fmt, dev, mem, vg_name);
Index: lvm2-2.02.39/lib/format_pool/disk_rep.c
===================================================================
--- lvm2-2.02.39.orig/lib/format_pool/disk_rep.c	2009-02-23 21:35:29.000000000 +0000
+++ lvm2-2.02.39/lib/format_pool/disk_rep.c	2009-02-23 21:34:02.000000000 +0000
@@ -348,7 +348,7 @@
 {
 	struct pool_list *pl;
 
-	if (!dev_open(dev))
+	if (!dev_open_flags(dev, O_RDONLY, 1, 0))
 		return_NULL;
 
 	if (!(pl = dm_pool_zalloc(mem, sizeof(*pl)))) {
Index: lvm2-2.02.39/lib/format_text/format-text.c
===================================================================
--- lvm2-2.02.39.orig/lib/format_text/format-text.c	2009-02-23 21:35:33.000000000 +0000
+++ lvm2-2.02.39/lib/format_text/format-text.c	2009-02-23 21:34:02.000000000 +0000
@@ -157,7 +157,7 @@
 		  PRIu64, mdac->area.start, mdac->area.size);
 	area = &mdac->area;
 
-	if (!dev_open(area->dev))
+	if (!dev_open_flags(area->dev, O_RDONLY, 1, 0))
 		return_0;
 
 	if (!(mdah = _raw_read_mda_header(fmt, area)))
@@ -415,7 +415,7 @@
 	int noprecommit = 0;
 	struct mda_header *mdah;
 
-	if (!dev_open(dev_area->dev))
+	if (!dev_open_flags(dev_area->dev, O_RDONLY, 1, 0))
 		return_0;
 
 	if (!(mdah = _raw_read_mda_header(fid->fmt, dev_area)))
@@ -442,7 +442,7 @@
 	char *desc;
 	uint32_t wrap = 0;
 
-	if (!dev_open(area->dev))
+	if (!dev_open_flags(area->dev, O_RDONLY, 1, 0))
 		return_NULL;
 
 	if (!(mdah = _raw_read_mda_header(fid->fmt, area)))
@@ -1032,7 +1032,7 @@
 	if (mda_free_sectors)
 		*mda_free_sectors = ((dev_area->size - MDA_HEADER_SIZE) / 2) >> SECTOR_SHIFT;
 
-	if (!dev_open(dev_area->dev))
+	if (!dev_open_flags(dev_area->dev, O_RDONLY, 1, 0))
 		return_NULL;
 
 	if (!(mdah = _raw_read_mda_header(fmt, dev_area)))
Index: lvm2-2.02.39/lib/label/label.c
===================================================================
--- lvm2-2.02.39.orig/lib/label/label.c	2009-02-23 21:35:38.000000000 +0000
+++ lvm2-2.02.39/lib/label/label.c	2009-02-23 21:34:02.000000000 +0000
@@ -272,7 +272,7 @@
 		return 1;
 	}
 
-	if (!dev_open(dev)) {
+	if (!dev_open_flags(dev, O_RDONLY, 1, 0)) {
 		stack;
 
 		if ((info = info_from_pvid(dev->pvid, 0)))
@@ -351,7 +351,7 @@
 	struct lvmcache_info *info;
 	int r = 0;
 
-	if (!dev_open(dev)) {
+	if (!dev_open_flags(dev, O_RDONLY, 1, 0)) {
 		if ((info = info_from_pvid(dev->pvid, 0)))
 			lvmcache_update_vgname_and_id(info, info->fmt->orphan_vg_name,
 						      info->fmt->orphan_vg_name,
