diff -r -u collectd-4.10.1.orig/configure.in collectd-4.10.1/configure.in
--- collectd-4.10.1.orig/configure.in	2010-07-09 12:01:59.000000000 +0200
+++ collectd-4.10.1/configure.in	2010-08-27 22:27:57.010271693 +0200
@@ -1626,7 +1626,7 @@
 # Check for the iptc_init symbol in the library.
 if test "x$with_libiptc" = "xyes" && test "x$with_own_libiptc" = "xno"
 then
-	AC_CHECK_LIB(iptc, iptc_init,
+	AC_SEARCH_LIBS(iptc_init, [iptc ip4tc],
 	[
 		AC_DEFINE(HAVE_LIBIPTC, 1, [Define to 1 if you have the iptc library (-liptc).])
 	],
