diff -Nuar udev-init-scripts-26/init.d/udev-mount udev-init-scripts-26-fix/init.d/udev-mount
--- udev-init-scripts-26/init.d/udev-mount	2013-04-15 15:49:33.000000000 +0000
+++ udev-init-scripts-26-fix/init.d/udev-mount	2014-02-04 16:10:28.339393688 +0000
@@ -1,5 +1,4 @@
 #!/sbin/runscript
-# Copyright 1999-2013 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
 description="mount devtmpfs on /dev"
@@ -7,7 +6,7 @@
 depend()
 {
 	provide dev-mount
-	keyword -vserver -lxc
+	keyword -vserver -lxc -openvz
 }
 
 mount_dev_directory()
