diff -Nuar -Nuar NetworkManager-0.9.8.2/configure.ac NetworkManager-0.9.8.2-fix/configure.ac
--- NetworkManager-0.9.8.2/configure.ac	2013-06-07 20:16:35.000000000 +0000
+++ NetworkManager-0.9.8.2-fix/configure.ac	2013-10-18 04:18:48.071205376 +0000
@@ -490,7 +490,7 @@
 
 # DHCP client support
 AC_ARG_WITH([dhclient], AS_HELP_STRING([--with-dhclient=yes|no|path], [Enable dhclient 4.x support]))
-AC_ARG_WITH([dhcpcd], AS_HELP_STRING([--with-dhcpcd=yes|no|path], [Enable dhcpcd 4.x support]))
+AC_ARG_WITH([dhcpcd], AS_HELP_STRING([--with-dhcpcd=yes|no|path], [Enable dhcpcd 6.x support]))
 # Default to "yes"
 AS_IF([test -z "$with_dhclient"], with_dhclient=yes)
 AS_IF([test -z "$with_dhcpcd"], with_dhcpcd=yes)
@@ -507,8 +507,8 @@
 if test "$with_dhcpcd" = "yes"; then
 	AC_PATH_PROGS(with_dhcpcd, dhcpcd, no, /sbin:/usr/sbin:/usr/local/sbin)
 	if test "$with_dhcpcd" != "no"; then
-		if ! $with_dhcpcd --version 2>&1 | grep -q "^dhcpcd [[45]]\."; then
-			AC_MSG_WARN([Cannot use dhcpcd, version 4.x or 5.x is required])
+		if ! $with_dhcpcd --version 2>&1 | grep -q "^dhcpcd [[56]]\."; then
+			AC_MSG_WARN([Cannot use dhcpcd, version 5.x or 6.x is required])
 			with_dhcpcd=no
 		fi
 	fi
