From d1c2e454c9a5d7413ae5b6fe05c2c1168baf0f31 Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint@balintreczey.hu>
Date: Thu, 6 Mar 2014 13:19:34 +0100
Subject: [PATCH 1/2] Don't enter ffmpeg dir when using external ffmpeg/libav

---
 lib/Makefile.in |    2 ++
 1 file changed, 2 insertions(+)

--- a/lib/Makefile.in
+++ b/lib/Makefile.in
@@ -87,12 +87,14 @@
 endif
 
 ffmpeg:
+ifneq (@USE_EXTERNAL_FFMPEG@,1)
 	$(MAKE) -C $@
 ifeq ($(findstring osx,$(ARCH)), osx)
 	-$(AR) d ffmpeg/libavcodec/libavcodec.a log2_tab.o
 	-$(AR) d ffmpeg/libavformat/libavformat.a log2_tab.o
 	-$(AR) d ffmpeg/libswresample/libswresample.a log2_tab.o
 endif
+endif
 
 clean:
 	rm -f $(addprefix $(SYSDIR)/, $(LIBS))
