From e7fc1bf22a8ee7e744883eaa0c8f2910d51b4208 Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint@balintreczey.hu>
Date: Sun, 30 Mar 2014 14:54:07 +0200
Subject: [PATCH] Fix av_stream_get_r_frame_rate Libav hack accessor

---
 lib/xbmc-libav-hacks/accessors.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/lib/xbmc-libav-hacks/accessors.c b/lib/xbmc-libav-hacks/accessors.c
index 417bf32..969d576 100644
--- a/lib/xbmc-libav-hacks/accessors.c
+++ b/lib/xbmc-libav-hacks/accessors.c
@@ -28,8 +28,5 @@ AVDictionary *av_frame_get_metadata (const AVFrame *frame)
 
 AVRational av_stream_get_r_frame_rate(const AVStream *s)
 {
-    AVRational zero;
-    zero.num = 0;
-    zero.den = 1;
-    return zero;
+    return s->avg_frame_rate;
 }
-- 
1.7.10.4

