From 5451bad08317e8f36d9541980f773ceeb7262e80 Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint@balintreczey.hu>
Date: Fri, 7 Mar 2014 00:17:40 +0100
Subject: [PATCH] Disble static ffmpeg when using external ffmpeg/libav

---
 configure.in |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.in b/configure.in
index 815e2db..553ac70 100644
--- a/configure.in
+++ b/configure.in
@@ -874,6 +874,9 @@ elif test "$use_arch" = "arm"; then
     fi
   fi
 fi
+if test "$use_static_ffmpeg" = "yes" -a "$use_external_ffmpeg" = "yes"; then
+  use_static_ffmpeg=no
+fi
 if test "$use_static_ffmpeg" = "yes"; then
   USE_STATIC_FFMPEG=1
   AC_DEFINE([USE_STATIC_FFMPEG], [1], [link ffmpeg statically])
--
1.7.10.4
