From dc85016024fef825a8fffcd9881a7a0bd86912af Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint@balintreczey.hu>
Date: Fri, 16 May 2014 12:07:51 +0200
Subject: [PATCH] Enable using external ffmpeg in ./configure

---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index 4c73ff1..31b02a7 100644
--- a/configure.in
+++ b/configure.in
@@ -600,7 +600,7 @@ PKG_PROG_PKG_CONFIG
 MAKE="${MAKE:-make}"
 OBJDUMP="${OBJDUMP:-objdump}"
 
-use_external_ffmpeg=no
+use_external_ffmpeg=yes
 use_static_ffmpeg=no
 
 # ffmpeg needs the output of uname -s (e.x. linux, darwin) for the target_os
-- 
1.9.1

