From bf558f775574b2243f1da21b345e8ae252b500d7 Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint@balintreczey.hu>
Date: Thu, 6 Mar 2014 16:37:39 +0100
Subject: [PATCH 2/2] Fix compilation with libav 10 beta1

---
 lib/xbmc-libav-hacks/libav_hacks.h |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/xbmc-libav-hacks/libav_hacks.h b/lib/xbmc-libav-hacks/libav_hacks.h
index f81e29e..a315c05 100644
--- a/lib/xbmc-libav-hacks/libav_hacks.h
+++ b/lib/xbmc-libav-hacks/libav_hacks.h
@@ -35,6 +35,11 @@
 #error "Your libav version is too old. Please update to libav-10 or git master."
 #endif

+// libavcodec
+
+// AVCodecContext does not have pkt_timebase member
+#define pkt_timebase time_base
+
 // libavutil

 #define AVFRAME_IN_LAVU
--
1.7.10.4
