diff --git a/xbmc/utils/MathUtils.h b/xbmc/utils/MathUtils.h
index 96af9f4..76dacc8 100644
--- a/xbmc/utils/MathUtils.h
+++ b/xbmc/utils/MathUtils.h
@@ -34,6 +34,7 @@
 
 #if defined(__ppc__) || \
     defined(__powerpc__) || \
+   (defined(__VFP_FP__) && defined(__SOFTFP__)) || \
    (defined(TARGET_DARWIN_IOS) && defined(__llvm__)) || \
    (defined(TARGET_ANDROID) && defined(__arm__)) || \
     defined(TARGET_RASPBERRY_PI)
@@ -42,6 +43,7 @@
 
 #if defined(__ppc__) || \
     defined(__powerpc__) || \
+   (defined(__VFP_FP__) && defined(__SOFTFP__)) || \
    (defined(TARGET_DARWIN) && defined(__llvm__)) || \
    (defined(TARGET_ANDROID) && defined(__arm__)) || \
     defined(TARGET_RASPBERRY_PI)
