diff --git a/xbmc/visualizations/XBMCProjectM/libprojectM/Renderer.hpp b/xbmc/visualizations/XBMCProjectM/libprojectM/Renderer.hpp
index 7d1bfdf..3e7b11b 100644
--- a/xbmc/visualizations/XBMCProjectM/libprojectM/Renderer.hpp
+++ b/xbmc/visualizations/XBMCProjectM/libprojectM/Renderer.hpp
@@ -25,7 +25,7 @@
 #include <FTGLPixmapFont.h>
 #include <FTGLExtrdFont.h>
 #else
-#include <FTGL/FTGL.h>
+#include <FTGL/ftgl.h>
 #include <FTGL/FTGLPixmapFont.h>
 #include <FTGL/FTGLExtrdFont.h>
 #endif
