diff -urN ruby-1.9.1-p129/lib/mkmf.rb ruby-1.9.1-p129-fixed/lib/mkmf.rb
--- ruby-1.9.1-p129/lib/mkmf.rb	2009-07-24 23:00:40.000000000 -0600
+++ ruby-1.9.1-p129-fixed/lib/mkmf.rb	2009-07-24 23:02:25.000000000 -0600
@@ -1703,7 +1703,7 @@
       for f in files
 	dest = "#{dir}/#{File.basename(f)}"
 	mfile.print("install-rb#{sfx}: #{dest}\n")
-        mfile.print("#{dest}: #{f}\n")
+        mfile.print("#{dest}: #{f} #{dir}\n")
         mfile.print("\t$(#{$extout ? 'COPY' : 'INSTALL_DATA'}) ")
 	mfile.print("#{fseprepl[f]} $(@D#{sep})\n")
 	if defined?($installed_list) and !$extout
