diff -Naur php-5.2.10.old/ext/pdo_dblib/dblib_driver.c php-5.2.10/ext/pdo_dblib/dblib_driver.c
--- php-5.2.10.old/ext/pdo_dblib/dblib_driver.c	2009-03-20 23:14:17.000000000 +0100
+++ php-5.2.10/ext/pdo_dblib/dblib_driver.c	2009-09-25 00:19:31.000000000 +0200
@@ -230,9 +230,9 @@
 		goto cleanup;
 	}
 
-	if (DBSETOPT(H->link, DBTEXTLIMIT, "2147483647") == FAIL) {
-		goto cleanup;
-	}
+//	if (DBSETOPT(H->link, DBTEXTLIMIT, "2147483647") == FAIL) {
+//		goto cleanup;
+//	}
 
 	if (vars[3].optval && FAIL == dbuse(H->link, vars[3].optval)) {
 		goto cleanup;
