diff -Nurp genkernel-3.4.10.906.orig/gen_bootloader.sh genkernel-3.4.10.906/gen_bootloader.sh
--- genkernel-3.4.10.906.orig/gen_bootloader.sh	2009-08-05 15:19:47.000000000 +0200
+++ genkernel-3.4.10.906/gen_bootloader.sh	2009-08-11 18:40:03.292176371 +0200
@@ -62,7 +62,7 @@ set_bootloader_grub() {
 
 			# Add grub configuration to grub.conf	
 			echo "# Genkernel generated entry, see GRUB documentation for details" >> ${GRUB_CONF}
-			echo "title=Gentoo Linux ($KV)" >> ${GRUB_CONF}
+			echo "title=Sabayon Linux ($KV)" >> ${GRUB_CONF}
 			if [ "${BUILD_INITRD}" -eq '0' ]
 			then
 				echo -e "\tkernel /kernel-${KNAME}-${ARCH}-${KV} root=${GRUB_ROOTFS}" >> ${GRUB_CONF}
@@ -139,7 +139,7 @@ set_bootloader_grub_duplicate_default() 
 	sed -n 1,$((${startline}-1))p "${GRUB_CONF}" > "${GRUB_CONF_TMP}"
 
 	# Put in our title
-	echo "title=Gentoo Linux (${KV})" >> "${GRUB_CONF_TMP}"
+	echo "title=Sabayon Linux (${KV})" >> "${GRUB_CONF_TMP}"
 
 	# Pass the default entry (minus the title) through to the replacement function and pipe the output to GRUB_CONF_TMP
 	sed -n $((${startline}+1)),$((${stopline}-1))p "${GRUB_CONF}" | set_bootloader_grub_duplicate_default_replace_kernel_initrd >> "${GRUB_CONF_TMP}"
diff -Nurp genkernel-3.4.10.906.orig/gen_cmdline.sh genkernel-3.4.10.906/gen_cmdline.sh
--- genkernel-3.4.10.906.orig/gen_cmdline.sh	2009-08-05 15:19:47.000000000 +0200
+++ genkernel-3.4.10.906/gen_cmdline.sh	2009-08-11 18:40:03.292176371 +0200
@@ -2,7 +2,7 @@
 # $Id: 20380ac6f16d8c9be4dcfbfd6bc1e0b0ba1979b3 $
 
 longusage() {
-  echo "Gentoo Linux Genkernel ${GK_V}"
+  echo "Sabayon/Gentoo Linux Genkernel ${GK_V}"
   echo "Usage: "
   echo "  genkernel [options] action"
   echo
@@ -136,7 +136,7 @@ longusage() {
 }
 
 usage() {
-  echo "Gentoo Linux Genkernel ${GK_V}"
+  echo "Sabayon/Gentoo Linux Genkernel ${GK_V}"
   echo "Usage: "
   echo "	genkernel [options] all"
   echo
diff -Nurp genkernel-3.4.10.906.orig/genkernel genkernel-3.4.10.906/genkernel
--- genkernel-3.4.10.906.orig/genkernel	2009-08-05 15:22:28.000000000 +0200
+++ genkernel-3.4.10.906/genkernel	2009-08-11 18:40:03.292176371 +0200
@@ -93,7 +93,7 @@ then
 fi
 
 clear_log
-NORMAL=${GOOD} print_info 1 "Gentoo Linux Genkernel; Version ${GK_V}${NORMAL}"
+NORMAL=${GOOD} print_info 1 "Sabayon/Gentoo Linux Genkernel; Version ${GK_V}${NORMAL}"
 print_info 1 "Running with options: ${Options}"
 echo
 
diff -Nurp genkernel-3.4.10.906.orig/netboot/linuxrc.x genkernel-3.4.10.906/netboot/linuxrc.x
--- genkernel-3.4.10.906.orig/netboot/linuxrc.x	2008-12-24 17:17:49.000000000 +0100
+++ genkernel-3.4.10.906/netboot/linuxrc.x	2009-08-11 18:40:03.293490689 +0200
@@ -249,8 +249,8 @@ SubGenius() {
 GenMotd() {
 	echo -e ""										> /etc/motd
 	echo -e ""										>> /etc/motd
-	echo -e "Gentoo Linux; http://www.gentoo.org/"						>> /etc/motd
-	echo -e " Copyright 2001-${CPYYEAR} Gentoo Foundation; Distributed under the GPL"	>> /etc/motd
+	echo -e "Sabayon Linux; http://www.sabayon.org/"					>> /etc/motd
+	echo -e " Copyright 2004-${CPYYEAR} Sabayon Linux; Distributed under the GPL"		>> /etc/motd
 	echo -e ""										>> /etc/motd
 	echo -e " Gentoo/${MYARCH} Netboot for ${MACHTYPE} Systems"				>> /etc/motd
 	echo -e " ${BUILDDATE}"									>> /etc/motd
