diff -Nurp genkernel-3.4.10.906.orig/defaults/linuxrc genkernel-3.4.10.906/defaults/linuxrc
--- genkernel-3.4.10.906.orig/defaults/linuxrc	2009-08-12 14:28:26.919151873 +0200
+++ genkernel-3.4.10.906/defaults/linuxrc	2009-08-12 14:30:39.632271882 +0200
@@ -132,6 +132,7 @@ do
 		# Redirect output to a specific tty
 		CONSOLE\=*|console\=*)
 			CONSOLE=`parse_opt "${x}"`
+			CONSOLE=`basename ${CONSOLE}`
 #			exec >${CONSOLE} <${CONSOLE} 2>&1
 		;;
 		# /dev/md
