--- lastfmsubmitd.orig	2006-09-21 16:04:18.755100865 +0200
+++ lastfmsubmitd	2006-09-21 16:05:49.094220050 +0200
@@ -146,10 +146,12 @@
             signal.alarm(0)
         # XXX: handshaketimeouterror?
         except IOError, e:
             signal.alarm(0)
             raise HandshakeError(e)
+        except Exception, e:
+            raise HandshakeError(e)
 
         if status_line == 'UPTODATE':
             self.read_submit_url(response)
         elif status_line.startswith('UPDATE'):
             try:
