diff -ur mpdscribble-0.18.orig/configure.ac mpdscribble-0.18/configure.ac
--- mpdscribble-0.18.orig/configure.ac	2009-07-08 20:18:18.000000000 +0300
+++ mpdscribble-0.18/configure.ac	2009-07-29 12:18:40.000000000 +0300
@@ -42,11 +42,11 @@
 dnl
 
 PKG_CHECK_MODULES([libglib],
-	[glib-2.0 >= 2.16],
+	[glib-2.0 >= 2.16 gthread-2.0],
 	[glib216=yes], [glib216=no])
 if test "x$glib216" != "xyes"; then
 	PKG_CHECK_MODULES([libglib],
-		[glib-2.0 >= 2.6],,
+		[glib-2.0 >= 2.6 gthread-2.0],,
 		[AC_MSG_ERROR([glib 2.6 is required])])
 
 	# use libgcrypt's MD5 routines with GLib < 2.16
