commit f6af85bef74ae6d1df1e811db7274829e50f10a9
Author: hasufell <hasufell@gentoo.org>
Date:   Fri Jul 19 18:37:21 2013 +0200

    unbundle colamd

diff --git a/extern/CMakeLists.txt b/extern/CMakeLists.txt
index f6de873..2b40c22 100644
--- a/extern/CMakeLists.txt
+++ b/extern/CMakeLists.txt
@@ -26,7 +26,6 @@
 # Otherwise we get warnings here that we cant fix in external projects
 remove_strict_flags()
 
-add_subdirectory(colamd)
 add_subdirectory(rangetree)
 add_subdirectory(wcwidth)
 add_subdirectory(libmv)
diff --git a/source/blenderplayer/CMakeLists.txt b/source/blenderplayer/CMakeLists.txt
index d91319d..bc3baee 100644
--- a/source/blenderplayer/CMakeLists.txt
+++ b/source/blenderplayer/CMakeLists.txt
@@ -170,8 +170,6 @@ endif()
 		list(APPEND BLENDER_SORTED_LIBS extern_ceres)
 	endif()
 
-	list(APPEND BLENDER_SORTED_LIBS extern_colamd)
-
 	if(WITH_MOD_BOOLEAN)
 		list(APPEND BLENDER_SORTED_LIBS bf_intern_bsp)
 		list(APPEND BLENDER_SORTED_LIBS bf_intern_moto)
@@ -214,7 +212,7 @@ endif()
 		list(SORT REM_MSG)
 		message(STATUS "Player Skipping: (${REM_MSG})")
 	endif()
-	target_link_libraries(blenderplayer ${BLENDER_SORTED_LIBS})
+	target_link_libraries(blenderplayer ${BLENDER_SORTED_LIBS} colamd)
 
 	unset(SEARCHLIB)
 	unset(SORTLIB)
diff --git a/source/creator/CMakeLists.txt b/source/creator/CMakeLists.txt
index 8ff4ec4..c84d944 100644
--- a/source/creator/CMakeLists.txt
+++ b/source/creator/CMakeLists.txt
@@ -919,7 +919,6 @@ endif()
 		bf_intern_smoke
 		extern_minilzo
 		extern_lzma
-		extern_colamd
 		ge_logic_ketsji
 		extern_recastnavigation
 		ge_logic
@@ -1033,7 +1032,7 @@ endif()
 		list(SORT REM_MSG)
 		message(STATUS "Blender Skipping: (${REM_MSG})")
 	endif()
-	target_link_libraries(blender ${BLENDER_SORTED_LIBS})
+	target_link_libraries(blender ${BLENDER_SORTED_LIBS} colamd)
 
 	unset(SEARCHLIB)
 	unset(SORTLIB)
