commit 26e853d31931a8bb08695aa98ae53b263236d407
Author: hasufell <hasufell@gentoo.org>
Date:   Fri Jul 19 18:40:23 2013 +0200

    unbundle glog/gflags

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b1a7830..a30831c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,6 +106,14 @@ set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin CACHE INTERNAL "" FORCE)
 set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib CACHE INTERNAL "" FORCE)
 
 #-----------------------------------------------------------------------------
+# Check for some modules
+if(UNIX)
+  find_package(PkgConfig)
+  pkg_check_modules(_PC_GFLAGS libgflags)
+  pkg_check_modules(_PC_GLOG libglog)
+endif()
+
+#-----------------------------------------------------------------------------
 # Set default config options
 
 get_blender_version()
diff --git a/extern/libmv/CMakeLists.txt b/extern/libmv/CMakeLists.txt
index 2c9a219..7061e04 100644
--- a/extern/libmv/CMakeLists.txt
+++ b/extern/libmv/CMakeLists.txt
@@ -47,8 +47,6 @@
 	)
 
 	list(APPEND INC
-		third_party/gflags
-		third_party/glog/src
 		third_party/ceres/include
 		../../intern/guardedalloc
 	)
@@ -208,7 +206,10 @@
 	)
 endif()
 
+include_directories(${_PC_GFLAGS_INCLUDE_DIRS} ${_PC_GLOG_INCLUDE_DIRS})
 blender_add_lib(extern_libmv "${SRC}" "${INC}" "${INC_SYS}")
+target_link_libraries(extern_libmv ${_PC_GFLAGS_LIBRARIES} ${_PC_GLOG_LIBRARIES})
+
 
 if(WITH_LIBMV)
 	add_subdirectory(third_party)
@@ -299,5 +300,5 @@
 	set(GLOG_INC_SYS
 	)
 
-	blender_add_lib(extern_glog "${GLOG_SRC}" "${GLOG_INC}" "${GLOG_INC_SYS}")
+	# blender_add_lib(extern_glog "${GLOG_SRC}" "${GLOG_INC}" "${GLOG_INC_SYS}")
 endif()
