--- LVM2.2.02.64.orig/udev/10-dm.rules.in	2010-04-28 15:37:37.000000000 +0200
+++ LVM2.2.02.64/udev/10-dm.rules.in	2010-05-26 17:40:48.708056437 +0200
@@ -99,7 +95,7 @@ ENV{DM_SUSPENDED}=="Suspended", ENV{DM_S
 # possible future changes.
 ENV{DM_UDEV_RULES_VSN}="1"
 
-ENV{DM_UDEV_DISABLE_DM_RULES_FLAG}!="1", ENV{DM_NAME}=="?*", SYMLINK+="(DM_DIR)/$env{DM_NAME}"
+ENV{DM_UDEV_DISABLE_DM_RULES_FLAG}!="1", ENV{DM_NAME}=="?*", NAME="mapper/$env{DM_NAME}", SYMLINK+="$kernel"
 
 # We have to ignore further rule application for inappropriate events
 # and devices. But still send the notification if cookie exists.
