diff -urN net-snmp-5.7.1/agent/mibgroup/if-mib/data_access/interface_linux.c net-snmp-5.7.1.new/agent/mibgroup/if-mib/data_access/interface_linux.c
--- net-snmp-5.7.1/agent/mibgroup/if-mib/data_access/interface_linux.c	2011-09-27 22:53:47.000000000 -0600
+++ net-snmp-5.7.1.new/agent/mibgroup/if-mib/data_access/interface_linux.c	2011-12-16 10:58:50.036810111 -0700
@@ -19,6 +19,11 @@
 #ifdef HAVE_PCI_LOOKUP_NAME
 #include <pci/pci.h>
 static struct pci_access *pci_access;
+static int pci_init_ok = 1;
+static void pci_error(char *msg, ...)
+{ 
+  pci_init_ok = 0;
+}
 #endif
 
 #ifdef HAVE_LINUX_ETHTOOL_H
@@ -105,7 +110,6 @@
 int netsnmp_prefix_listen(void);
 #endif
 
-
 void
 netsnmp_arch_interface_init(void)
 {
@@ -145,10 +149,13 @@
 #endif
 
 #ifdef HAVE_PCI_LOOKUP_NAME
     pci_access = pci_alloc();
-    if (pci_access)
+    pci_access->error = pci_error;
+    if (pci_access) {
 	pci_init(pci_access);
-    else
+    	if (!pci_init_ok)
+		pci_access = NULL;
+    } else
 	snmp_log(LOG_ERR, "Unable to create pci access method\n");
 #endif
 }
