diff -ur flac-1.2.1.orig/configure.in flac-1.2.1/configure.in
--- flac-1.2.1.orig/configure.in	2007-09-13 18:48:42.000000000 +0300
+++ flac-1.2.1/configure.in	2007-09-20 19:26:05.000000000 +0300
@@ -301,7 +301,7 @@
 	CPPFLAGS="-DNDEBUG $CPPFLAGS"
 	if test "x$GCC" = xyes; then
 		CPPFLAGS="-DFLaC__INLINE=__inline__ $CPPFLAGS"
-		CFLAGS="-O3 -funroll-loops -finline-functions -Wall -W -Winline $CFLAGS"
+		CFLAGS="$CFLAGS"
 	fi
 fi
 
