Index: flac-1.2.1/configure.in
===================================================================
--- flac-1.2.1.orig/configure.in
+++ flac-1.2.1/configure.in
@@ -255,6 +255,12 @@ AC_DEFINE(FLAC__HAS_OGG)
 AH_TEMPLATE(FLAC__HAS_OGG, [define if you have the ogg library])
 fi
 
+dnl Build examples?
+AC_ARG_ENABLE(examples,
+       AS_HELP_STRING([--disable-examples], [Don't build examples during make]),,
+       enable_examples="yes")
+AM_CONDITIONAL(EXAMPLES, [test "$enable_examples" = "yes"])
+
 dnl check for i18n(internationalization); these are from libiconv/gettext
 AM_ICONV
 AM_LANGINFO_CODESET
Index: flac-1.2.1/Makefile.am
===================================================================
--- flac-1.2.1.orig/Makefile.am
+++ flac-1.2.1/Makefile.am
@@ -30,7 +30,11 @@
 
 AUTOMAKE_OPTIONS = foreign 1.7
 
-SUBDIRS = doc include m4 man src examples test build obj
+SUBDIRS = doc include m4 man src test build obj
+
+if EXAMPLES
+SUBDIRS += examples
+endif
 
 DISTCLEANFILES = libtool-disable-static
 
