--- distribute_setup.py
+++ distribute_setup.py
@@ -78,11 +78,11 @@
     except ImportError:
         return do_download()
     try:
-        pkg_resources.require("distribute>="+version); return
+        pkg_resources.require("setuptools>="+version); return
     except pkg_resources.VersionConflict, e:
         if was_imported:
             print >>sys.stderr, (
-            "The required version of distribute (>=%s) is not available, and\n"
+            "The required version of setuptools (>=%s) is not available, and\n"
             "can't be installed while this script is running. Please install\n"
             " a more recent version first, using 'easy_install -U distribute'."
             "\n\n(Currently using %r)"
@@ -116,7 +116,7 @@
             if delay:
                 log.warn("""
 ---------------------------------------------------------------------------
-This script requires distribute version %s to run (even to display
+This script requires setuptools version %s to run (even to display
 help).  I will attempt to download it for you (from
 %s), but
 you may need to enable firewall access for this script first.
--- setup.py
+++ setup.py
@@ -21,7 +21,7 @@
     before_install()
 
 dist = setup(
-    name="distribute",
+    name="setuptools",
     version=VERSION,
     description="Download, build, install, upgrade, and uninstall Python "
         "packages -- easily!",
--- setuptools/package_index.py
+++ setuptools/package_index.py
@@ -142,7 +142,7 @@
                 yield urlparse.urljoin(url, htmldecode(match.group(1)))
 
 user_agent = "Python-urllib/%s distribute/%s" % (
-    urllib2.__version__, require('distribute')[0].version
+    urllib2.__version__, require('setuptools')[0].version
 )
 
 
