--- distribute_setup.py
+++ distribute_setup.py
@@ -174,12 +174,14 @@
     return open(path).read() == content
 
 def _rename_path(path):
+    return
     new_name = path + '.OLD.%s' % time.time()
     log.warn('Renaming %s into %s' % (path, new_name))
     os.rename(path, new_name)
     return new_name
 
 def _remove_flat_installation(placeholder):
+    return
     if not os.path.isdir(placeholder):
         log.warn('Unkown installation at %s' % placeholder)
         return False
@@ -212,6 +214,7 @@
     return True
 
 def after_install(dist):
+    return
     log.warn('After install bootstrap.')
     placeholder = dist.get_command_obj('install').install_purelib
     if not os.path.exists(placeholder):
@@ -256,10 +259,12 @@
     return True
 
 def before_install():
+    return
     log.warn('Before install bootstrap.')
     fake_setuptools()
 
 def fake_setuptools():
+    return
     log.warn('Scanning installed packages')
     try:
         import pkg_resources
@@ -324,52 +329,6 @@
 
 def main(argv, version=DEFAULT_VERSION):
     """Install or upgrade setuptools and EasyInstall"""
-    # let's deactivate any existing setuptools installation first
-    fake_setuptools()
-    try:
-        import setuptools
-        # we need to check if the installed setuptools
-        # is from Distribute or from setuptools
-        if not hasattr(setuptools, '_distribute'):
-            # now we are ready to install distribute
-            raise ImportError
-    except ImportError:
-        egg = None
-        try:
-            egg = download_setuptools(version, delay=0)
-            sys.path.insert(0, egg)
-            import setuptools
-            if not hasattr(setuptools, '_distribute'):
-                placeholder = os.path.split(os.path.dirname(setuptools.__file__))[0]
-                if not placeholder.endswith('.egg'):
-                    res = _remove_flat_installation(placeholder)
-                    if res:
-                        _relaunch()
-                print >> sys.stderr, (
-                "The patch didn't work, Setuptools is still active.\n"
-                "Possible reason: your have a system-wide setuptools installed "
-                "and you are in a virtualenv.\n"
-                "If you are inside a virtualenv, make sure you used the --no-site-packages option"
-                )
-                sys.exit(2)
-            dist = _easy_install(argv, egg)
-            after_install(dist)
-            return
-            #from setuptools.command import easy_install
-            #try:
-            #    return easy_install.main(list(argv)+['-v']+[egg])
-            #except DistutilsError:
-            #    return sys.exit(2)
-        finally:
-            if egg and os.path.exists(egg):
-                os.unlink(egg)
-    else:
-        if setuptools.__version__ == '0.0.1':
-            print >>sys.stderr, (
-            "You have an obsolete version of setuptools installed.  Please\n"
-            "remove it from your system entirely before rerunning this script."
-            )
-            sys.exit(2)
 
     req = "distribute>="+version
     import pkg_resources
