--- Configure.orig	2006-06-02 13:14:22.000000000 -0500
+++ Configure	2006-06-02 13:07:03.000000000 -0500
@@ -2967,7 +2967,7 @@
 : Try to determine whether config.sh was made on this system
 case "$config_sh" in
 '')
-myuname=`$uname -a 2>/dev/null`
+myuname=`$uname -a | $sed -e "s/'//" 2>/dev/null`
 $test -z "$myuname" && myuname=`hostname 2>/dev/null`
 # tr '[A-Z]' '[a-z]' would not work in EBCDIC
 # because the A-Z/a-z are not consecutive.
