diff -rup xmlrpc-c-1.06.03/examples/cpp/sample_add_client_complex.cpp xmlrpc-c-devel/examples/cpp/sample_add_client_complex.cpp
--- xmlrpc-c-1.06.03/examples/cpp/sample_add_client_complex.cpp	2005-11-06 21:28:00.000000000 +0100
+++ xmlrpc-c-devel/examples/cpp/sample_add_client_complex.cpp	2008-01-17 16:42:40.000000000 +0100
@@ -15,6 +15,7 @@
 =============================================================================*/
 
 #include <cassert>
+#include <cstdlib>
 #include <string>
 #include <iostream>
 #include <xmlrpc-c/girerr.hpp>
diff -rup xmlrpc-c-1.06.03/examples/cpp/xmlrpc_sample_add_client.cpp xmlrpc-c-devel/examples/cpp/xmlrpc_sample_add_client.cpp
--- xmlrpc-c-1.06.03/examples/cpp/xmlrpc_sample_add_client.cpp	2005-06-05 02:01:35.000000000 +0200
+++ xmlrpc-c-devel/examples/cpp/xmlrpc_sample_add_client.cpp	2008-01-17 16:42:04.000000000 +0100
@@ -1,4 +1,5 @@
 #include <string>
+#include <cstdlib>
 #include <iostream>
 #include <xmlrpc-c/girerr.hpp>
 #include <xmlrpc-c/base.hpp>
diff -rup xmlrpc-c-1.06.03/include/xmlrpc-c/oldcppwrapper.hpp xmlrpc-c-devel/include/xmlrpc-c/oldcppwrapper.hpp
--- xmlrpc-c-1.06.03/include/xmlrpc-c/oldcppwrapper.hpp	2008-01-17 16:43:30.000000000 +0100
+++ xmlrpc-c-devel/include/xmlrpc-c/oldcppwrapper.hpp	2008-01-17 16:29:31.000000000 +0100
@@ -26,6 +26,7 @@
 
 
 #include <string>
+#include <cstdlib>
 #include <xmlrpc-c/base.h>
 #include <xmlrpc-c/client.h>
 #include <xmlrpc-c/server.h>
diff -rup xmlrpc-c-1.06.03/src/cpp/client_simple.cpp xmlrpc-c-devel/src/cpp/client_simple.cpp
--- xmlrpc-c-1.06.03/src/cpp/client_simple.cpp	2008-01-17 16:43:30.000000000 +0100
+++ xmlrpc-c-devel/src/cpp/client_simple.cpp	2008-01-17 16:27:18.000000000 +0100
@@ -1,4 +1,5 @@
 #include <string>
+#include <cstring>
 
 #include "xmlrpc-c/girerr.hpp"
 using girerr::error;
diff -rup xmlrpc-c-1.06.03/src/cpp/server_abyss.cpp xmlrpc-c-devel/src/cpp/server_abyss.cpp
--- xmlrpc-c-1.06.03/src/cpp/server_abyss.cpp	2008-01-17 16:43:30.000000000 +0100
+++ xmlrpc-c-devel/src/cpp/server_abyss.cpp	2008-01-17 16:26:23.000000000 +0100
@@ -1,4 +1,5 @@
 #include <cassert>
+#include <cstdlib>
 #include <string>
 #include <memory>
 #include <signal.h>
diff -rup xmlrpc-c-1.06.03/src/cpp/value.cpp xmlrpc-c-devel/src/cpp/value.cpp
--- xmlrpc-c-1.06.03/src/cpp/value.cpp	2008-01-17 16:43:30.000000000 +0100
+++ xmlrpc-c-devel/src/cpp/value.cpp	2008-01-17 16:25:45.000000000 +0100
@@ -31,6 +31,7 @@
 #include <string>
 #include <vector>
 #include <time.h>
+#include <cstdlib>
 
 #include "xmlrpc-c/girerr.hpp"
 using girerr::error;
diff -rup xmlrpc-c-1.06.03/src/cpp/XmlRpcCpp.cpp xmlrpc-c-devel/src/cpp/XmlRpcCpp.cpp
--- xmlrpc-c-1.06.03/src/cpp/XmlRpcCpp.cpp	2008-01-17 16:43:30.000000000 +0100
+++ xmlrpc-c-devel/src/cpp/XmlRpcCpp.cpp	2008-01-17 16:24:46.000000000 +0100
@@ -25,6 +25,8 @@
 
 
 #include <string>
+#include <cstdlib>
+#include <memory.h>
 #include "xmlrpc-c/oldcppwrapper.hpp"
 
 using std::string;
diff -rup xmlrpc-c-1.06.03/tools/xml-rpc-api2cpp/xml-rpc-api2cpp.cpp xmlrpc-c-devel/tools/xml-rpc-api2cpp/xml-rpc-api2cpp.cpp
--- xmlrpc-c-1.06.03/tools/xml-rpc-api2cpp/xml-rpc-api2cpp.cpp	2008-01-17 16:43:30.000000000 +0100
+++ xmlrpc-c-devel/tools/xml-rpc-api2cpp/xml-rpc-api2cpp.cpp	2008-01-17 16:28:34.000000000 +0100
@@ -1,5 +1,6 @@
 #include <iostream>
 #include <stdexcept>
+#include <cstdlib>
 
 #include "xmlrpc-c/oldcppwrapper.hpp"
 
