Index: xmlrpc-c-1.06.02/lib/abyss/src/thread_fork.c
===================================================================
--- xmlrpc-c-1.06.02.orig/lib/abyss/src/thread_fork.c
+++ xmlrpc-c-1.06.02/lib/abyss/src/thread_fork.c
@@ -281,11 +281,6 @@ ThreadForks(void) {
 
 
 
-void
-ThreadUpdateStatus(TThread * const threadP) {
-    /* Necessary only for primitive Unix forking */
-}
-
 
 /*********************************************************************
 ** Mutex
