diff -NrU5 xmlrpc-c-1.06.27.orig/src/cpp/test/server_abyss.cpp xmlrpc-c-1.06.27/src/cpp/test/server_abyss.cpp
--- xmlrpc-c-1.06.27.orig/src/cpp/test/server_abyss.cpp	2008-05-31 01:21:14.000000000 +0200
+++ xmlrpc-c-1.06.27/src/cpp/test/server_abyss.cpp	2008-05-31 01:21:43.000000000 +0200
@@ -12,10 +12,11 @@
 #include <iostream>
 #include <vector>
 #include <sstream>
 #include <memory>
 #include <time.h>
+#include <cstring>
 
 #include "xmlrpc-c/girerr.hpp"
 using girerr::error;
 using girerr::throwf;
 #include "xmlrpc-c/base.hpp"
diff -NrU5 xmlrpc-c-1.06.27.orig/src/cpp/test/test.cpp xmlrpc-c-1.06.27/src/cpp/test/test.cpp
--- xmlrpc-c-1.06.27.orig/src/cpp/test/test.cpp	2008-05-31 01:21:14.000000000 +0200
+++ xmlrpc-c-1.06.27/src/cpp/test/test.cpp	2008-05-31 01:21:20.000000000 +0200
@@ -3,10 +3,11 @@
 #include <iomanip>
 #include <vector>
 #include <sstream>
 #include <memory>
 #include <time.h>
+#include <cstring>
 
 #include "xmlrpc-c/girerr.hpp"
 using girerr::error;
 #include "transport_config.h"
 #include "xmlrpc-c/base.hpp"
