Index: xmlrpc-c-1.06.09/src/Makefile
===================================================================
--- xmlrpc-c-1.06.09.orig/src/Makefile
+++ xmlrpc-c-1.06.09/src/Makefile
@@ -48,6 +48,7 @@ endif
 
 ifeq ($(ENABLE_LIBXML2_BACKEND),yes)
   LIBXML_INCLUDES = $(shell xml2-config --cflags)
+  LIBXML_LIBS = $(shell xml2-config --libs)
 else
   LIBXML_INCLUDES = -I$(SRCDIR)/lib/expat/xmlparse
 endif
@@ -95,7 +96,7 @@ LIBXMLRPC_OBJS = \
 	xmlrpc_utf8.lo \
 	xmlrpc_authcookie.lo \
 
-LIBXMLRPC_LIBDEP = -lxmlrpc_util
+LIBXMLRPC_LIBDEP = -lxmlrpc_util $(LIBXML_LIBS)
 
 LIB_OBJS = $(LIBXMLRPC_CLIENT_OBJS) $(LIBXMLRPC_SERVER_OBJS)
 
diff -Naur xmlrpc-c-1.06.09.orig/examples/cpp/Makefile xmlrpc-c-1.06.09/examples/cpp/Makefile
--- xmlrpc-c-1.06.09.orig/examples/cpp/Makefile	2006-01-09 00:36:06.000000000 +0100
+++ xmlrpc-c-1.06.09/examples/cpp/Makefile	2008-03-02 00:15:42.000000000 +0100
@@ -63,13 +63,13 @@
 all: $(PROGS)
 
 $(SERVERPROGS_ABYSS):%:%.o
-	$(CXXLD) -o $@ $(LDFLAGS) $^ $(LDADD_SERVER_ABYSS)
+	$(CXXLD) -o $@ $^ $(LDADD_SERVER_ABYSS) $(LDFLAGS)
 
 $(LEGACY_CLIENTPROGS):%:%.o
-	$(CXXLD) -o $@ $(LDFLAGS) $^ $(LDADD_LEGACY_CLIENT)
+	$(CXXLD) -o $@ $^ $(LDADD_LEGACY_CLIENT) $(LDFLAGS)
 
 $(CLIENTPROGS):%:%.o
-	$(CXXLD) -o $@ $(LDFLAGS) $^ $(LDADD_CLIENT)
+	$(CXXLD) -o $@ $^ $(LDADD_CLIENT) $(LDFLAGS)
 
 %.o:%.cpp
 	$(CXX) -c $(INCLUDES) $(CXXFLAGS) $<
diff -Naur xmlrpc-c-1.06.09.orig/examples/Makefile xmlrpc-c-1.06.09/examples/Makefile
--- xmlrpc-c-1.06.09.orig/examples/Makefile	2005-11-20 22:39:57.000000000 +0100
+++ xmlrpc-c-1.06.09/examples/Makefile	2008-03-02 00:15:04.000000000 +0100
@@ -81,16 +81,16 @@
 	$(MAKE) -C $(dir $@) $(notdir $@)
 
 $(CLIENTPROGS):%:%.o
-	$(CCLD) -o $@ $(LDFLAGS) $^ $(LDADD_CLIENT)
+	$(CCLD) -o $@ $^ $(LDADD_CLIENT) $(LDFLAGS)
 
 $(SERVERPROGS_CGI):%.cgi:%_cgi.o
-	$(CCLD) -o $@ $(LDFLAGS) $^ $(LDADD_SERVER_CGI)
+	$(CCLD) -o $@ $^ $(LDADD_SERVER_CGI) $(LDFLAGS)
 
 $(SERVERPROGS_ABYSS):%:%.o
-	$(CCLD) -o $@ $(LDFLAGS) $^ $(LDADD_SERVER_ABYSS)
+	$(CCLD) -o $@ $^ $(LDADD_SERVER_ABYSS) $(LDFLAGS)
 
 gen_sample_add_xml:%:%.o
-	$(CCLD) -o $@ $(LDFLAGS) $^ $(LDADD_BASE)
+	$(CCLD) -o $@ $^ $(LDADD_BASE) $(LDFLAGS)
 
 %.o:%.c
 	$(CC) -c $(INCLUDES) $(CFLAGS) $<
