Index: xmlrpc-c-1.06.02/tools/Makefile.common
===================================================================
--- xmlrpc-c-1.06.02.orig/tools/Makefile.common
+++ xmlrpc-c-1.06.02/tools/Makefile.common
@@ -1,6 +1,8 @@
 # -*-makefile-*-    <-- an Emacs control
 
-CLIENT_LDLIBS = -L$(BUILDDIR)/src/.libs -lxmlrpc_client -lxmlrpc
+CLIENT_LDLIBS = -L$(BUILDDIR)/lib/libutil/.libs -lxmlrpc_util
+
+CLIENT_LDLIBS += -L$(BUILDDIR)/src/.libs -lxmlrpc_client -lxmlrpc
 
 ifeq ($(MUST_BUILD_LIBWWW_CLIENT),yes)
   CLIENT_LDLIBS += $(shell libwww-config --libs)
@@ -14,7 +16,6 @@ endif
 
 CLIENT_LDLIBS += $(LDLIBS_XML)
 
-CLIENT_LDLIBS += -L$(BUILDDIR)/lib/libutil/.libs -lxmlrpc_util
 
 
 UTIL_DIR = $(BUILDDIR)/lib/util
