--- xmlrpc-c-1.06.27.orig/GNUmakefile
+++ xmlrpc-c-1.06.27/GNUmakefile
@@ -100,13 +100,13 @@
 # Install old names of header files for backward compatibility
 	cd $(DESTDIR)$(HEADERINST_DIR); \
 	  rm -f xmlrpc.h xmlrpc_client.h xmlrpc_server.h xmlrpc_cgi.h \
-                xmlrpc_server_abyss.h xmlrpc_server_w32httpsys.h \
+                xmlrpc_abyss.h xmlrpc_server_w32httpsys.h \
 	        XmlRpcCpp.h; \
 	  $(LN_S) xmlrpc-c/oldxmlrpc.h         xmlrpc.h; \
 	  $(LN_S) xmlrpc-c/client.h            xmlrpc_client.h; \
 	  $(LN_S) xmlrpc-c/server.h            xmlrpc_server.h; \
 	  $(LN_S) xmlrpc-c/server_cgi.h        xmlrpc_cgi.h; \
-	  $(LN_S) xmlrpc-c/server_abyss.h      xmlrpc_server_abyss.h; \
+	  $(LN_S) xmlrpc-c/server_abyss.h      xmlrpc_abyss.h; \
 	  $(LN_S) xmlrpc-c/server_w32httpsys.h xmlrpc_server_w32httpsys.h; \
 	  $(LN_S) xmlrpc-c/oldcppwrapper.hpp   XmlRpcCpp.h ;\
 