Index: xmlrpc-c-1.05/src/cpp/Makefile
===================================================================
--- xmlrpc-c-1.05.orig/src/cpp/Makefile
+++ xmlrpc-c-1.05/src/cpp/Makefile
@@ -135,7 +135,7 @@ libxmlrpc_client++.a: $(LIBXMLRPC_CLIENT
 #  RULES TO COMPILE OBJECT MODULES FOR LIBRARIES
 #-----------------------------------------------------------------------------
 
-CXXFLAGS = $(CXXFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD)
+CXXFLAGS = $(CXXFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD) -fPIC
 
 XmlRpcCpp.o:%.o:%.cpp
 	$(CXX) -c $(BASIC_INCLUDES) $(CXXFLAGS) $<
