--- xmlrpc-c-1.06.09/tools/Makefile.common.orig	2006-07-30 05:07:28.000000000 +0200
+++ xmlrpc-c-1.06.09/tools/Makefile.common	2008-02-29 11:57:06.000000000 +0100
@@ -1,6 +1,7 @@
 # -*-makefile-*-    <-- an Emacs control
 
 CLIENT_LDLIBS = -L$(BUILDDIR)/src/.libs -lxmlrpc_client -lxmlrpc
+CLIENT_LDLIBS += -L$(BUILDDIR)/lib/libutil/.libs -lxmlrpc_util
 
 ifeq ($(MUST_BUILD_LIBWWW_CLIENT),yes)
   CLIENT_LDLIBS += $(shell libwww-config --libs)
@@ -14,8 +15,6 @@
 
 CLIENT_LDLIBS += $(LDLIBS_XML)
 
-CLIENT_LDLIBS += -L$(BUILDDIR)/lib/libutil/.libs -lxmlrpc_util
-
 
 UTIL_DIR = $(BUILDDIR)/lib/util
 
