Index: xmlrpc-c/lib/curl_transport/xmlrpc_curl_transport.c
===================================================================
--- xmlrpc-c.orig/lib/curl_transport/xmlrpc_curl_transport.c
+++ xmlrpc-c/lib/curl_transport/xmlrpc_curl_transport.c
@@ -1214,7 +1214,7 @@ setupCurlSession(xmlrpc_env *           
             curl_easy_setopt(curlSessionP, CURLOPT_SSLENGINE,
                              curlSetupP->sslEngine);
         if (curlSetupP->sslEngineDefault)
-            curl_easy_setopt(curlSessionP, CURLOPT_SSLENGINE_DEFAULT);
+            curl_easy_setopt(curlSessionP, CURLOPT_SSLENGINE_DEFAULT, 1);
         if (curlSetupP->sslVersion != XMLRPC_SSLVERSION_DEFAULT)
             curl_easy_setopt(curlSessionP, CURLOPT_SSLVERSION,
                              curlSetupP->sslVersion);

