diff -urN crda-1.1.3/Makefile crda-1.1.3-fixed/Makefile
--- crda-1.1.3/Makefile	2013-01-27 19:47:00.408013408 -0700
+++ crda-1.1.3-fixed/Makefile	2013-01-27 19:50:03.996013960 -0700
@@ -100,11 +100,6 @@
 	$(NQ)
 	$(Q) exit 1
 
-keys-%.c: utils/key2pub.py $(wildcard $(PUBKEY_DIR)/*.pem)
-	$(NQ) '  GEN ' $@
-	$(NQ) '  Trusted pubkeys:' $(wildcard $(PUBKEY_DIR)/*.pem)
-	$(Q)./utils/key2pub.py --$* $(wildcard $(PUBKEY_DIR)/*.pem) $@
-
 %.o: %.c regdb.h reglib.h
 	$(NQ) '  CC  ' $@
 	$(Q)$(CC) -c $(CPPFLAGS) $(CFLAGS) -o $@ $<
