--- a/LVM2.2.02.98/make.tmpl.in	2013-07-31 18:36:16.313390582 -0400
+++ b/LVM2.2.02.98/make.tmpl.in	2013-07-31 18:35:26.524389673 -0400
@@ -32,7 +32,7 @@
 
 LIBS = @LIBS@
 # Extra libraries always linked with static binaries
-STATIC_LIBS = $(SELINUX_LIBS) $(UDEV_LIBS)
+STATIC_LIBS = $(SELINUX_LIBS) $(UDEV_STATIC_LIBS)
 DEFS += @DEFS@
 CFLAGS += @CFLAGS@
 CLDFLAGS += @CLDFLAGS@
@@ -47,6 +47,7 @@
 READLINE_LIBS = @READLINE_LIBS@
 SELINUX_LIBS = @SELINUX_LIBS@
 UDEV_LIBS = @UDEV_LIBS@
+UDEV_STATIC_LIBS = @UDEV_STATIC_LIBS@
 TESTING = @TESTING@
 
 # Setup directory variables
--- a/LVM2.2.02.98/configure.in	2013-07-31 18:36:16.291390582 -0400
+++ b/LVM2.2.02.98/configure.in	2013-07-31 18:32:18.849386248 -0400
@@ -930,9 +930,11 @@
 	if  test x$PKGCONFIG_INIT != x1; then
 		pkg_config_init
 	fi
	PKG_CHECK_MODULES(UDEV, libudev >= 143, [UDEV_PC="libudev"])
+	UDEV_STATIC_LIBS=`$PKG_CONFIG --static --libs libudev`
 	AC_DEFINE([UDEV_SYNC_SUPPORT], 1, [Define to 1 to enable synchronisation with udev processing.])
 fi
+AC_SUBST(UDEV_STATIC_LIBS)
 
 dnl -- Enable udev rules
 AC_MSG_CHECKING(whether to enable installation of udev rules required for synchronisation)

