--- mail/em-account-editor.c~	2010-12-27 15:57:03.000000000 +0100
+++ mail/em-account-editor.c	2011-03-27 17:57:40.000000000 +0200
@@ -68,7 +68,7 @@
 #include "mail-ops.h"
 #include "mail-mt.h"
 
-#if defined (HAVE_NSS)
+#if defined (HAVE_NSS) && defined (ENABLE_SMIME)
 #include "smime/gui/e-cert-selector.h"
 #endif
 
@@ -1202,7 +1202,7 @@
 	return folder;
 }
 
-#if defined (HAVE_NSS)
+#if defined (HAVE_NSS) && defined (ENABLE_SMIME)
 static void
 smime_changed (EMAccountEditor *emae)
 {
@@ -3072,7 +3072,7 @@
 emae_security_page (EConfig *ec, EConfigItem *item, GtkWidget *parent, GtkWidget *old, gpointer data)
 {
 	EMAccountEditor *emae = data;
-#if defined (HAVE_NSS)
+#if defined (HAVE_NSS) && defined (ENABLE_SMIME)
 	EMAccountEditorPrivate *priv = emae->priv;
 #endif
 	GtkWidget *w;
@@ -3092,7 +3092,7 @@
 	emae_account_toggle (emae, "pgp_no_imip_sign", E_ACCOUNT_PGP_NO_IMIP_SIGN, builder);
 	emae_account_toggle (emae, "pgp_always_trust", E_ACCOUNT_PGP_ALWAYS_TRUST, builder);
 
-#if defined (HAVE_NSS)
+#if defined (HAVE_NSS) && defined (ENABLE_SMIME)
 	/* TODO: this should handle its entry separately? */
 	priv->smime_sign_key = emae_account_entry (emae, "smime_sign_key", E_ACCOUNT_SMIME_SIGN_KEY, builder);
 	priv->smime_sign_key_select = (GtkButton *)e_builder_get_widget (builder, "smime_sign_key_select");
--- mail/em-format-html-display.c~	2010-10-14 23:18:47.000000000 +0200
+++ mail/em-format-html-display.c	2011-03-27 17:58:58.000000000 +0200
@@ -50,7 +50,7 @@
 #include <e-util/e-dialog-utils.h>
 #include <e-util/e-icon-factory.h>
 
-#ifdef HAVE_NSS
+#if defined (HAVE_NSS) && defined (ENABLE_SMIME)
 #include "certificate-viewer.h"
 #include "e-cert-db.h"
 #endif
@@ -171,7 +171,7 @@
 	po->widget = NULL;
 }
 
-#ifdef HAVE_NSS
+#if defined (HAVE_NSS) && defined (ENABLE_SMIME)
 static void
 efhd_xpkcs7mime_viewcert_clicked (GtkWidget *button,
                                   struct _smime_pobject *po)
--- modules/addressbook/e-book-shell-backend.c~	2010-11-10 19:55:36.000000000 +0100
+++ modules/addressbook/e-book-shell-backend.c	2011-03-27 17:59:56.000000000 +0200
@@ -317,6 +317,7 @@
 		autocompletion_config_new,
 		200);
 
+#ifdef ENABLE_SMIME
 	preferences_window = e_shell_get_preferences_window (shell);
 	e_preferences_window_add_page (
 		E_PREFERENCES_WINDOW (preferences_window),
@@ -325,6 +326,7 @@
 		_("Certificates"),
 		certificate_manager_config_new,
 		700);
+#endif
 
 	return FALSE;
 }
