diff --git a/addressbook/gui/contact-editor/Makefile.am b/addressbook/gui/contact-editor/Makefile.am
index 486d71e..763808b 100644
--- a/addressbook/gui/contact-editor/Makefile.am
+++ b/addressbook/gui/contact-editor/Makefile.am
@@ -33,7 +33,7 @@ libecontacteditor_la_LIBADD =						\
 	$(top_builddir)/addressbook/gui/widgets/libeabwidgets.la	\
 	$(top_builddir)/addressbook/gui/merging/libeabbookmerging.la	\
 	$(top_builddir)/addressbook/printing/libecontactprint.la	\
-	$(top_builddir)/widgets/menus/.libs/libmenus.la	\
+	$(top_builddir)/widgets/menus/libmenus.la	\
 	$(EVOLUTION_ADDRESSBOOK_LIBS)					\
 	$(GNOME_PLATFORM_LIBS)
 
