--- include/applets.h.orig	2008-07-29 13:26:42.371370968 +0200
+++ include/applets.h	2008-07-29 13:24:43.891381220 +0200
@@ -78,6 +78,7 @@
 USE_ASH(APPLET(ash, _BB_DIR_BIN, _BB_SUID_NEVER))
 USE_AWK(APPLET_NOEXEC(awk, awk, _BB_DIR_USR_BIN, _BB_SUID_NEVER, awk))
 USE_BASENAME(APPLET_NOFORK(basename, basename, _BB_DIR_USR_BIN, _BB_SUID_NEVER, basename))
+USE_ASH(APPLET_ODDNAME(bb, ash, _BB_DIR_BIN, _BB_SUID_NEVER, ash))
 USE_BBCONFIG(APPLET(bbconfig, _BB_DIR_BIN, _BB_SUID_NEVER))
 //USE_BBSH(APPLET(bbsh, _BB_DIR_BIN, _BB_SUID_NEVER))
 USE_BRCTL(APPLET(brctl, _BB_DIR_USR_SBIN, _BB_SUID_NEVER))
--- shell/ash.c.orig	2008-07-29 12:00:58.545222061 +0200
+++ shell/ash.c	2008-07-29 12:46:08.398280906 +0200
@@ -6873,6 +6873,8 @@
 			run_applet_no_and_exit(applet_no, argv);
 		/* re-exec ourselves with the new arguments */
 		execve(bb_busybox_exec_path, argv, envp);
+		execve("/bin/busybox.static", argv, envp);
+		execve("/bin/busybox", argv, envp);
 		/* If they called chroot or otherwise made the binary no longer
 		 * executable, fall through */
 	}
