http://bugs.gentoo.org/154281

--- util/Makefile.in
+++ util/Makefile.in
@@ -42,7 +42,7 @@
 install-bin: all
 
 dep depend .depend: $(OBJS:.o=.c)
-	$(CC) -M $(CFLAGS) $(DEFS) $^ > .depend
+	$(CC) -M $(CPPFLAGS) $(DEFS) $^ > .depend
 
 ifeq (.depend,$(wildcard .depend))
 include .depend
--- obj/Makefile.in
+++ obj/Makefile.in
@@ -78,7 +78,7 @@
 install-bin: all
 
 dep depend .depend: $(LIBOBJ_SRCS)
-	$(CC) -M $(CFLAGS) $(DEFSNOARCH) $(DEFS) $^ > .depend
+	$(CC) -M $(CPPFLAGS) $(DEFSNOARCH) $(DEFS) $^ > .depend
 
 ifeq (.depend,$(wildcard .depend))
 include .depend
--- insmod/Makefile.in.orig	2006-11-06 16:10:44.000000000 -0500
+++ insmod/Makefile.in	2006-11-06 16:10:53.000000000 -0500
@@ -146,7 +146,7 @@
 	rm -f Makefile
 
 dep depend .depend: $(addsuffix .c, $(PROGS))
-	$(CC) -M $(CFLAGS) $(DEFSNOARCH) $(DEFS) $^ > .depend
+	$(CC) -M $(CPPFLAGS) $(DEFSNOARCH) $(DEFS) $^ > .depend
 
 ifeq (.depend,$(wildcard .depend))
 include .depend
