diff --git a/docs/examples/certinfo.c b/docs/examples/certinfo.c
index 02558d9..e1463ec 100644
--- a/docs/examples/certinfo.c
+++ b/docs/examples/certinfo.c
@@ -22,7 +22,6 @@
 #include <stdio.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 static size_t wrfu(void *ptr,  size_t  size,  size_t  nmemb,  void *stream)
diff --git a/docs/examples/chkspeed.c b/docs/examples/chkspeed.c
index fbcb1f7..efcf472 100644
--- a/docs/examples/chkspeed.c
+++ b/docs/examples/chkspeed.c
@@ -35,7 +35,6 @@
 #include <time.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #define URL_BASE "http://speedtest.your.domain/"
diff --git a/docs/examples/curlgtk.c b/docs/examples/curlgtk.c
index 2c44280..59dcd3f 100644
--- a/docs/examples/curlgtk.c
+++ b/docs/examples/curlgtk.c
@@ -13,7 +13,6 @@
 #include <gtk/gtk.h>
 
 #include <curl/curl.h>
-#include <curl/types.h> /* new for v7 */
 #include <curl/easy.h> /* new for v7 */
 
 GtkWidget *Bar;
diff --git a/docs/examples/ftpget.c b/docs/examples/ftpget.c
index 7b5e0d7..3ec17b9 100644
--- a/docs/examples/ftpget.c
+++ b/docs/examples/ftpget.c
@@ -22,7 +22,6 @@
 #include <stdio.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 /*
diff --git a/docs/examples/ftpgetinfo.c b/docs/examples/ftpgetinfo.c
index 52f87d6..9bf39c3 100644
--- a/docs/examples/ftpgetinfo.c
+++ b/docs/examples/ftpgetinfo.c
@@ -23,7 +23,6 @@
 #include <string.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 /*
diff --git a/docs/examples/ftpgetresp.c b/docs/examples/ftpgetresp.c
index ea882a0..49b2137 100644
--- a/docs/examples/ftpgetresp.c
+++ b/docs/examples/ftpgetresp.c
@@ -22,7 +22,6 @@
 #include <stdio.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 /*
diff --git a/docs/examples/postit2.c b/docs/examples/postit2.c
index bb7fd48..470fd8b 100644
--- a/docs/examples/postit2.c
+++ b/docs/examples/postit2.c
@@ -37,7 +37,6 @@
 #include <string.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 int main(int argc, char *argv[])
diff --git a/docs/examples/sepheaders.c b/docs/examples/sepheaders.c
index dac2878..973e465 100644
--- a/docs/examples/sepheaders.c
+++ b/docs/examples/sepheaders.c
@@ -24,7 +24,6 @@
 #include <unistd.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 static size_t write_data(void *ptr, size_t size, size_t nmemb, void *stream)
diff --git a/docs/examples/simplessl.c b/docs/examples/simplessl.c
index aeaadce..217622a 100644
--- a/docs/examples/simplessl.c
+++ b/docs/examples/simplessl.c
@@ -22,7 +22,6 @@
 #include <stdio.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 
diff --git a/docs/examples/smooth-gtk-thread.c b/docs/examples/smooth-gtk-thread.c
index 8cf106c..b679a27 100644
--- a/docs/examples/smooth-gtk-thread.c
+++ b/docs/examples/smooth-gtk-thread.c
@@ -37,7 +37,6 @@
 #include <pthread.h>
 
 #include <curl/curl.h>
-#include <curl/types.h> /* new for v7 */
 #include <curl/easy.h> /* new for v7 */
 
 #define NUMT 4
