diff -ru alsamixergui-0.9.0rc1-2.orig/configure.in alsamixergui-0.9.0rc1-2/configure.in
--- alsamixergui-0.9.0rc1-2.orig/configure.in	2015-02-22 17:56:28.389496411 +0100
+++ alsamixergui-0.9.0rc1-2/configure.in	2015-02-22 20:32:54.190556290 +0100
@@ -1,8 +1,9 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(src/alsamixer.cxx)
-AM_INIT_AUTOMAKE(alsamixergui, 0.9.0rc1-2)
+AC_INIT([alsamixergui], [0.9.0rc1-2])
+AC_CONFIG_SRCDIR(src/alsamixer.cxx)
+AC_CONFIG_HEADER(src/config.h)
 AC_PREFIX_DEFAULT(/usr)
-AM_CONFIG_HEADER(src/config.h)
+AM_INIT_AUTOMAKE
 
 dnl Checks for programs.
 AC_PROG_CC
@@ -11,11 +12,22 @@
 AC_PROG_LN_S
 
 dnl Checks for libraries.
-AC_CHECK_LIB(fltk,numericsort,,AC_MSG_ERROR("missing fltk"))
-AM_PATH_ALSA(0.9.0)
+AC_CHECK_PROG([HAVE_FLTK_CONFIG], [fltk-config], [yes])
+if test "x$HAVE_FLTK_CONFIG" = "xyes"; then
+  FLTK_CFLAGS=`fltk-config --cflags`
+  FLTK_LIBS=`fltk-config --ldflags`
+  AC_SUBST([FLTK_CFLAGS])
+  AC_SUBST([FLTK_LIBS])
+else
+  AC_MSG_ERROR("missing fltk")
+fi
+
+PKG_CHECK_MODULES([ALSA], [alsa >= 0.9.0])
+AC_SUBST([ALSA_CFLAGS])
+AC_SUBST([ALSA_LIBS])
 
 dnl Checks for header files.
 AC_HEADER_STDC
-AC_HAVE_HEADERS(stdio.h string.h stdlib.h getopt.h alsa/asoundlib.h FL/Fl.H)
+AC_CHECK_HEADERS(stdio.h getopt.h)
 
 AC_OUTPUT(Makefile src/Makefile)
diff -ru alsamixergui-0.9.0rc1-2.orig/src/Makefile.am alsamixergui-0.9.0rc1-2/src/Makefile.am
--- alsamixergui-0.9.0rc1-2.orig/src/Makefile.am	2015-02-22 17:56:28.389496411 +0100
+++ alsamixergui-0.9.0rc1-2/src/Makefile.am	2015-02-22 17:57:14.931482206 +0100
@@ -10,4 +10,6 @@
  alsamixer.cxx \
  ncurser_to_fl.H
 
-alsamixergui_LDADD = @LIBS@ @ALSA_LIBS@
+alsamixergui_CPPFLAGS = $(FLTK_CFLAGS) $(ALSA_CFLAGS)
+
+alsamixergui_LDADD = $(FLTK_LIBS) $(ALSA_LIBS)
