--- alsamixergui-0.9.0rc1-2/configure.BAD	2006-01-16 12:28:52.000000000 -0600
+++ alsamixergui-0.9.0rc1-2/configure	2006-01-16 12:39:24.000000000 -0600
@@ -1476,17 +1476,17 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lfltk  $LIBS"
+LIBS="-lfltk -lstdc++ $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 1482 "configure"
 #include "confdefs.h"
 /* Override any gcc2 internal prototype to avoid an error.  */
 /* We use char because int might match the return type of a gcc2
     builtin and then its argument prototype would still apply.  */
-char numericsort();
+char fl_numericsort();
 
 int main() {
-numericsort()
+fl_numericsort()
 ; return 0; }
 EOF
 if { (eval echo configure:1493: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
@@ -1664,7 +1664,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lasound  $LIBS"
+LIBS="-lasound -lstdc++ $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 1670 "configure"
 #include "confdefs.h"
--- alsamixergui-0.9.0rc1-2/src/alsamixer.cxx.BAD	2006-01-16 12:41:54.000000000 -0600
+++ alsamixergui-0.9.0rc1-2/src/alsamixer.cxx	2006-01-16 12:45:44.000000000 -0600
@@ -653,8 +653,8 @@
     if (vleft >= 0 && vright >= 0) {
       if (joined) {
 #ifdef ALSAMIXER_GUI
-	for (chn = snd_mixer_selem_channel_id_t(0); chn < SND_MIXER_SCHN_LAST; 
-	snd_mixer_selem_channel_id_t(int(chn)++))
+	for (chn = snd_mixer_selem_channel_id_t(0); chn < SND_MIXER_SCHN_LAST;
+	snd_mixer_selem_channel_id_t(int(chn)+1))
 #else
 	for (chn = 0; chn < SND_MIXER_SCHN_LAST; chn++)
 #endif
