diff -NrU5 xmlrpc-c-1.18.02.orig/src/Makefile xmlrpc-c-1.18.02/src/Makefile
--- xmlrpc-c-1.18.02.orig/src/Makefile	2009-05-02 12:53:43.000000000 +0200
+++ xmlrpc-c-1.18.02/src/Makefile	2009-05-02 13:30:00.000000000 +0200
@@ -128,11 +128,15 @@
   $(SUBDIRS:%=%/all) \
 
 # Extra dependencies to make parallel make work in spite of all the submakes
 # (See top level make file for details)
 ifeq ($MUST_BUILD_CLIENT),yes)
-cpp/all: $(BLDDIR)/transport_config.h
+  cpp/all: $(BLDDIR)/transport_config.h
+endif
+
+ifeq ($(ENABLE_CPLUSPLUS),yes)
+  cpp/all: $(TARGET_SHARED_LIBRARIES)
 endif
 
 #-----------------------------------------------------------------------------
 #  RULES TO LINK LIBRARIES
 #-----------------------------------------------------------------------------
