diff -NrU5 xmlrpc-c-1.18.02.orig/common.mk xmlrpc-c-1.18.02/common.mk
--- xmlrpc-c-1.18.02.orig/common.mk	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/common.mk	2009-04-28 00:37:21.000000000 +0200
@@ -282,32 +282,10 @@
 $(TARGET_MODS_PP:%=%.osh):%.osh:%.cpp
 	$(CXX) -c -o $@ $(INCLUDES) $(CXXFLAGS_ALL) $<
 
 
 ##############################################################################
-#                           MISC BUILD RULES                                 #
-##############################################################################
-
-# We use the srcdir symbolic link simply to make the make
-# rules easier to read in the make output.  We could use the $(SRCDIR)
-# variable, but that makes the compile and link commands
-# a mile long.  Note that Make sometime figures that a directory which
-# is a dependency is newer than the symbolic link pointing to it and wants
-# to rebuild the symbolic link.  So we don't make $(SRCDIR) a
-# dependency of 'srcdir'.
-
-# We should do the same for 'blddir'.  We did once before, then undid
-# it in an erroneous effort to enable parallel make.  It's a little harder
-# with blddir; when we did it before, we had to use the non-symlink
-# version in a few places.
-
-srcdir:
-	$(LN_S) $(SRCDIR) $@
-blddir:
-	$(LN_S) $(BLDDIR) $@
-
-##############################################################################
 #                    RECURSIVE SUBDIRECTORY BUILD RULES                      #
 ##############################################################################
 
 .PHONY: $(SUBDIRS:%=%/all)
 $(SUBDIRS:%=%/all): %/all: $(CURDIR)/%
@@ -594,11 +572,10 @@
 # especially with built .h files.  Better not to burden the user, who
 # gains nothing from it, with that.
 #
 	rm -f depend.mk
 	rm -f Makefile.depend  # We used to create a file by this name
-	rm -f srcdir blddir
 
 .PHONY: distdir-common
 distdir-common:
 	@for file in $(DISTFILES); do \
 	  d=$(SRCDIR); \
diff -NrU5 xmlrpc-c-1.18.02.orig/lib/abyss/src/Makefile xmlrpc-c-1.18.02/lib/abyss/src/Makefile
--- xmlrpc-c-1.18.02.orig/lib/abyss/src/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/lib/abyss/src/Makefile	2009-04-28 00:48:59.000000000 +0200
@@ -50,22 +50,17 @@
 
 include $(SRCDIR)/common.mk
 
 CFLAGS_LOCAL = -D_UNIX
 
-INCLUDES = -Iblddir -Isrcdir -Isrcdir/include -Isrcdir/lib/util/include
+INCLUDES = -I$(BLDDIR) -I$(SRCDIR) -I$(SRCDIR)/include -I$(SRCDIR)/lib/util/include
 
 ABYSS_SHLIB = $(call shlibfn,libxmlrpc_abyss)
 #ABYSS_SHLIB is e.g. libxmlrpc_abyss.so.3.1
 ABYSS_SHLIBLE = $(call shliblefn,libxmlrpc_abyss)
 #ABYSS_SHLIBLE is e.g. libxmlrpc_abyss.so
 
-# This 'common.mk' dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/common.mk: srcdir blddir
-
 .PHONY: all
 all: libxmlrpc_abyss.a $(TARGET_SHARED_LIBRARIES) $(TARGET_SHARED_LE_LIBS)
 
 
 # Rule for this is in common.mk, courtesy of TARGET_LIBRARY_NAMES
diff -NrU5 xmlrpc-c-1.18.02.orig/lib/expat/xmlparse/Makefile xmlrpc-c-1.18.02/lib/expat/xmlparse/Makefile
--- xmlrpc-c-1.18.02.orig/lib/expat/xmlparse/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/lib/expat/xmlparse/Makefile	2009-04-28 00:39:45.000000000 +0200
@@ -83,11 +83,6 @@
 distclean: clean distclean-common
 
 .PHONY: dep
 dep: dep-common
 
-# This 'common.mk' dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/common.mk: srcdir
-
 include depend.mk
diff -NrU5 xmlrpc-c-1.18.02.orig/lib/expat/xmltok/Makefile xmlrpc-c-1.18.02/lib/expat/xmltok/Makefile
--- xmlrpc-c-1.18.02.orig/lib/expat/xmltok/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/lib/expat/xmltok/Makefile	2009-04-28 00:38:11.000000000 +0200
@@ -24,11 +24,11 @@
 # we started using the Gcc -Wundef option, that generates a warning, so
 # se set it explicitly to 0 here.
 
 CFLAGS_LOCAL = -DXML_BYTE_ORDER=0
 
-# -I. is necessary when blddir != srcdir
+# -I. is necessary when $(BLDDIR) != $(SRCDIR)
 INCLUDES = -I. -I$(BLDDIR) -I$(SRCDIR)/lib/util/include
 
 default: all
 
 TARGET_LIBRARY_NAMES := libxmlrpc_xmltok
diff -NrU5 xmlrpc-c-1.18.02.orig/lib/libutil/Makefile xmlrpc-c-1.18.02/lib/libutil/Makefile
--- xmlrpc-c-1.18.02.orig/lib/libutil/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/lib/libutil/Makefile	2009-04-28 00:48:59.000000000 +0200
@@ -41,23 +41,18 @@
 MAJ=3
   # Major number of shared libraries in this directory
 
 include $(SRCDIR)/common.mk
 
-INCLUDES = -I$(BLDDIR) -Isrcdir \
-           -I$(BLDDIR)/include -Isrcdir/include -Isrcdir/lib/util/include
+INCLUDES = -I$(BLDDIR) -I$(SRCDIR) \
+           -I$(BLDDIR)/include -I$(SRCDIR)/include -I$(SRCDIR)/lib/util/include
 
 UTIL_SHLIB = $(call shlibfn,libxmlrpc_util)
 #UTIL_SHLIB is e.g. libxmlrpc_util.so.3.1
 UTIL_SHLIBLE = $(call shliblefn,libxmlrpc_util)
 #UTIL_SHLIBLE is e.g. libxmlrpc_util.so
 
-# This 'common.mk' dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/common.mk: srcdir blddir
-
 .PHONY: all
 all: libxmlrpc_util.a $(TARGET_SHARED_LIBRARIES) $(TARGET_SHARED_LE_LIBS)
 
 # Rule for this is in common.mk, courtesy of TARGET_LIBRARY_NAMES:
 $(UTIL_SHLIB): $(TARGET_MODS:%=%.osh)
diff -NrU5 xmlrpc-c-1.18.02.orig/src/cpp/Makefile xmlrpc-c-1.18.02/src/cpp/Makefile
--- xmlrpc-c-1.18.02.orig/src/cpp/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/src/cpp/Makefile	2009-04-28 00:38:11.000000000 +0200
@@ -37,18 +37,18 @@
 # INCLUDES and DEP_SOURCES are used by dep-common target
 INCLUDES = $(BASIC_INCLUDES) $(CLIENT_INCLUDES) $(LIBXML_INCLUDES) \
   $(SERVER_INCLUDES) $(SERVER_ABYSS_INCLUDES) $(TRANSPORT_INCLUDES)
 DEP_SOURCES = *.cpp
 
-WININET_TRANSPORT_DIR = srcdir/lib/wininet_transport
-CURL_TRANSPORT_DIR    = srcdir/lib/curl_transport
-LIBWWW_TRANSPORT_DIR  = srcdir/lib/libwww_transport
+WININET_TRANSPORT_DIR = $(SRCDIR)/lib/wininet_transport
+CURL_TRANSPORT_DIR    = $(SRCDIR)/lib/curl_transport
+LIBWWW_TRANSPORT_DIR  = $(SRCDIR)/lib/libwww_transport
 
 ifeq ($(ENABLE_LIBXML2_BACKEND),yes)
   LIBXML_INCLUDES = $(LIBXML2_CFLAGS)
 else
-  LIBXML_INCLUDES = -Isrcdir/lib/expat/xmlparse
+  LIBXML_INCLUDES = -I$(SRCDIR)/lib/expat/xmlparse
 endif
 
 ifeq ($(ENABLE_LIBXML2_BACKEND),yes)
   XML_PARSER_LIBDEP = $(shell xml2-config --libs)
 else
@@ -80,26 +80,20 @@
 MAJ = 5
   # Major number of shared libraries in this directory
 
 include $(SRCDIR)/common.mk
 
-
-# This 'common.mk' dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/common.mk: srcdir blddir
-
 TRANSPORT_INCLUDES = \
   -I$(WININET_TRANSPORT_DIR) \
   -I$(CURL_TRANSPORT_DIR) \
   -I$(LIBWWW_TRANSPORT_DIR) \
 
 BASIC_INCLUDES = \
-  -Iblddir/include \
-  -Isrcdir/include \
-  -Iblddir \
-  -Isrcdir/lib/util/include
+  -I$(BLDDIR)/include \
+  -I$(SRCDIR)/include \
+  -I$(BLDDIR) \
+  -I$(SRCDIR)/lib/util/include
 
 
 # TARGET_STATIC_LIBARIES, etc. are set by common.mk, based on
 # TARGET_LIB_NAMES_PP.
 
@@ -125,22 +119,22 @@
   $(LIBXMLRPC) \
   $(LIBXMLRPC_SERVER) \
   $(LIBXMLRPC_UTIL)
 $(LIBXMLRPC_CPP_SH): LIBOBJECTS = XmlRpcCpp.osh
 $(LIBXMLRPC_CPP_SH): LIBDEP = \
-  -Lblddir/src  -lxmlrpc_server -lxmlrpc \
+  -L$(BLDDIR)/src  -lxmlrpc_server -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
 
 LIBXMLRPCPP_SH = $(call shlibfn, lib$(LIBXMLRPCPP_NAME))
 
 $(LIBXMLRPCPP_SH): $(LIBXMLRPCPP_MODS:%=%.osh) \
   $(LIBXMLRPC) \
   $(LIBXMLRPC_UTIL)
 $(LIBXMLRPCPP_SH): LIBOBJECTS = $(LIBXMLRPCPP_MODS:%=%.osh)
 $(LIBXMLRPCPP_SH): LIBDEP = \
-  -Lblddir/src -lxmlrpc \
+  -L$(BLDDIR)/src -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
 
 
 LIBXMLRPC_SERVERPP_SH = $(call shlibfn, libxmlrpc_server++)
@@ -151,11 +145,11 @@
   $(LIBXMLRPC_UTIL) \
   $(LIBXMLRPC)
 $(LIBXMLRPC_SERVERPP_SH): LIBOBJECTS = $(LIBXMLRPC_SERVERPP_MODS:%=%.osh)
 $(LIBXMLRPC_SERVERPP_SH): LIBDEP = \
   -L. -l$(LIBXMLRPCPP_NAME) \
-  -Lblddir/src -lxmlrpc_server -lxmlrpc \
+  -L$(BLDDIR)/src -lxmlrpc_server -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
 
 LIBXMLRPC_SERVER_ABYSSPP_SH = $(call shlibfn, libxmlrpc_server_abyss++)
 
@@ -167,11 +161,11 @@
   $(LIBXMLRPC_UTIL) \
   $(LIBXMLRPC)
 $(LIBXMLRPC_SERVER_ABYSSPP_SH): LIBOBJECTS = $(LIBXMLRPC_SERVER_ABYSSPP_MODS:%=%.osh)
 $(LIBXMLRPC_SERVER_ABYSSPP_SH): LIBDEP = \
    -L. -lxmlrpc_server++  -l$(LIBXMLRPCPP_NAME) \
-   -Lblddir/src -lxmlrpc_server_abyss -lxmlrpc_server -lxmlrpc \
+   -L$(BLDDIR)/src -lxmlrpc_server_abyss -lxmlrpc_server -lxmlrpc \
    $(XML_PARSER_LIBDEP) \
    -L$(LIBXMLRPC_ABYSS_DIR) -lxmlrpc_abyss \
    -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util \
 
 $(LIBXMLRPC_SERVER_CGIPP_SH): $(LIBXMLRPC_SERVER_CGIPP_MODS:%=%.osh) \
@@ -179,11 +173,11 @@
   $(call shliblefn, lib$(LIBXMLRPCPP_NAME)) \
   $(LIBXMLRPC)
 $(LIBXMLRPC_SERVER_CGIPP_SH): LIBOBJECTS = $(LIBXMLRPC_SERVER_CGIPP_MODS:%=%.osh)
 $(LIBXMLRPC_SERVER_CGIPP_SH): LIBDEP = \
   -L. -lxmlrpc_server++  -l$(LIBXMLRPCPP_NAME) \
-  -Lblddir/src -lxmlrpc_server -lxmlrpc \
+  -L$(BLDDIR)/src -lxmlrpc_server -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
  -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
 
 LIBXMLRPC_SERVER_PSTREAMPP_SH = $(call shlibfn, libxmlrpc_server_pstream++)
 
@@ -193,11 +187,11 @@
   $(call shliblefn, libxmlrpc_packetsocket) \
   $(call shliblefn, libxmlrpc_server++)
 $(LIBXMLRPC_SERVER_PSTREAMPP_SH): LIBOBJECTS = $(LIBXMLRPC_SERVER_PSTREAMPP_MODS:%=%.osh)
 $(LIBXMLRPC_SERVER_PSTREAMPP_SH): LIBDEP = \
   -L. -lxmlrpc_server++ -l$(LIBXMLRPCPP_NAME) -lxmlrpc_packetsocket \
-  -Lblddir/src -lxmlrpc_server -lxmlrpc \
+  -L$(BLDDIR)/src -lxmlrpc_server -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util \
 
 LIBXMLRPC_CLIENTPP_SH = $(call shlibfn, libxmlrpc_client++)
 
@@ -208,22 +202,22 @@
   $(call shliblefn, lib$(LIBXMLRPCPP_NAME)) \
   $(LIBXMLRPC_CLIENT)
 $(LIBXMLRPC_CLIENTPP_SH): LIBOBJECTS = $(LIBXMLRPC_CLIENTPP_MODS:%=%.osh)
 $(LIBXMLRPC_CLIENTPP_SH): LIBDEP = \
   -L. -l$(LIBXMLRPCPP_NAME) -lxmlrpc_packetsocket \
-  -Lblddir/src -lxmlrpc_client -lxmlrpc \
+  -L$(BLDDIR)/src -lxmlrpc_client -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
 
 LIBXMLRPC_PACKETSOCKET_SH = $(call shlibfn, libxmlrpc_packetsocket)
 
 $(LIBXMLRPC_PACKETSOCKET_SH): $(LIBXMLRPC_PACKETSOCKET_MODS:%=%.osh) \
   $(call shliblefn, lib$(LIBXMLRPCPP_NAME))
 $(LIBXMLRPC_PACKETSOCKET_SH): LIBOBJECTS = $(LIBXMLRPC_PACKETSOCKET_MODS:%=%.osh)
 $(LIBXMLRPC_PACKETSOCKET_SH): LIBDEP = \
   -L. -l$(LIBXMLRPCPP_NAME) \
-  -Lblddir/src -lxmlrpc \
+  -L$(BLDDIR)/src -lxmlrpc \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util
 
 libxmlrpc_cpp.a: XmlRpcCpp.o
 libxmlrpc_cpp.a: LIBOBJECTS = XmlRpcCpp.o
diff -NrU5 xmlrpc-c-1.18.02.orig/src/cpp/test/Makefile xmlrpc-c-1.18.02/src/cpp/test/Makefile
--- xmlrpc-c-1.18.02.orig/src/cpp/test/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/src/cpp/test/Makefile	2009-04-28 00:48:59.000000000 +0200
@@ -26,16 +26,11 @@
   LIBS += $(shell libwww-config --libs)
 endif
 
 LIBS += -lpthread
 
-INCLUDES = -Isrcdir/include -I$(BLDDIR) -Isrcdir -Isrcdir/lib/util/include
-
-# This 'Makefile' dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-Makefile: srcdir
+INCLUDES = -I$(SRCDIR)/include -I$(BLDDIR) -I$(SRCDIR) -I$(SRCDIR)/lib/util/include
 
 include $(SRCDIR)/common.mk
 
 
 TEST_OBJS = test.o registry.o server_abyss.o server_pstream.o tools.o
diff -NrU5 xmlrpc-c-1.18.02.orig/src/Makefile xmlrpc-c-1.18.02/src/Makefile
--- xmlrpc-c-1.18.02.orig/src/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/src/Makefile	2009-04-28 00:32:07.000000000 +0200
@@ -115,15 +115,10 @@
 MAJ=3
   # Major number of shared libraries in this directory
 
 include $(SRCDIR)/common.mk
 
-# This 'common.mk' dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/common.mk: srcdir blddir
-
 # TARGET_STATIC_LIBRARIES, etc. are set by common.mk, based on
 # TARGET_LIBRARY_NAMES.
 
 
 all: \
diff -NrU5 xmlrpc-c-1.18.02.orig/tools/common.mk xmlrpc-c-1.18.02/tools/common.mk
--- xmlrpc-c-1.18.02.orig/tools/common.mk	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/tools/common.mk	2009-04-28 00:49:45.000000000 +0200
@@ -5,11 +5,11 @@
 # the corresponding -l.
 # 
 # Note that in a properly configured system, curl-config, etc. do not
 # generate -L options for general directories.
 
-CLIENT_LDLIBS = -Lblddir/src -Lblddir/lib/libutil
+CLIENT_LDLIBS = -L$(BLDDIR)/src -L$(BLDDIR)/lib/libutil
 
 CLIENT_LDLIBS += -lxmlrpc_client -lxmlrpc -lxmlrpc_util
 
 ifeq ($(MUST_BUILD_LIBWWW_CLIENT),yes)
   CLIENT_LDLIBS += $(shell libwww-config --libs)
@@ -21,17 +21,17 @@
   CLIENT_LDLIBS += $(shell wininet-config --libs)
 endif
 
 CLIENT_LDLIBS += $(LDLIBS_XML)
 
-CLIENTPP_LDLIBS = -Lblddir/src/cpp
+CLIENTPP_LDLIBS = -L$(BLDDIR)/src/cpp
 CLIENTPP_LDLIBS += -lxmlrpc_client++ -lxmlrpc_packetsocket -lxmlrpc++
 
 include $(SRCDIR)/common.mk
 
 ifneq ($(OMIT_LIB_RULE),Y)
-srcdir/tools/lib/dumpvalue.o: FORCE
+$(SRCDIR)/tools/lib/dumpvalue.o: FORCE
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/tools/lib/Makefile $(notdir $@) 
 endif
 
 .PHONY: install
 install: install-common
diff -NrU5 xmlrpc-c-1.18.02.orig/tools/lib/Makefile xmlrpc-c-1.18.02/tools/lib/Makefile
--- xmlrpc-c-1.18.02.orig/tools/lib/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/tools/lib/Makefile	2009-04-28 00:40:19.000000000 +0200
@@ -13,29 +13,24 @@
 OMIT_LIB_RULE = Y
 
 include $(SRCDIR)/tools/common.mk
 
 INCLUDES = \
-  -Isrcdir/$(SUBDIR)/include \
-  -Iblddir \
-  -Isrcdir/include \
-  -Isrcdir/lib/util/include \
+  -I$(SRCDIR)/$(SUBDIR)/include \
+  -I$(BLDDIR) \
+  -I$(SRCDIR)/include \
+  -I$(SRCDIR)/lib/util/include \
 
 LIBOBJS = \
   dumpvalue.o \
 
 .PHONY: all
 all: $(LIBOBJS)
 
 %.o:%.c
 	$(CC) -c $(CFLAGS_ALL) $<
 
-# This common.mk dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/tools/common.mk: srcdir blddir
-
 include depend.mk
 
 .PHONY: dep
 dep: dep-common
 
diff -NrU5 xmlrpc-c-1.18.02.orig/tools/xmlrpc/Makefile xmlrpc-c-1.18.02/tools/xmlrpc/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xmlrpc/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/tools/xmlrpc/Makefile	2009-04-28 00:38:11.000000000 +0200
@@ -14,14 +14,14 @@
 
 include $(SRCDIR)/tools/common.mk
 
 INCLUDES = \
   -I../lib/include \
-  -Iblddir \
-  -Iblddir/include \
-  -Isrcdir/include \
-  -Isrcdir/lib/util/include
+  -I$(BLDDIR) \
+  -I$(BLDDIR)/include \
+  -I$(SRCDIR)/include \
+  -I$(SRCDIR)/lib/util/include
 
 LDFLAGS = $(LADD)
 
 all: xmlrpc
 
@@ -32,11 +32,11 @@
   stripcaseeq.o \
   string_parser.o \
 
 UTILS = $(UTIL_OBJS:%=$(UTIL_DIR)/%)
 
-DUMPVALUE = srcdir/tools/lib/dumpvalue.o
+DUMPVALUE = $(SRCDIR)/tools/lib/dumpvalue.o
 
 # Can we just use $(LIBS) in the link?
 
 LIBS = $(LIBXMLRPC_CLIENT) $(LIBXMLRPC) $(LIBXMLRPC_XML) $(LIBXMLRPC_UTIL)
 
@@ -46,15 +46,10 @@
 	$(CCLD) -o $@ $(LDFLAGS) $(XMLRPC_OBJS) $(UTILS) $(CLIENT_LDLIBS)
 
 %.o:%.c $(BLDDIR)/include/xmlrpc-c/config.h
 	$(CC) -c $(CFLAGS_ALL) $<
 
-# This common.mk dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/tools/common.mk: srcdir blddir
-
 include depend.mk
 
 .PHONY: dep
 dep: dep-common
 
diff -NrU5 xmlrpc-c-1.18.02.orig/tools/xml-rpc-api2cpp/Makefile xmlrpc-c-1.18.02/tools/xml-rpc-api2cpp/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xml-rpc-api2cpp/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/tools/xml-rpc-api2cpp/Makefile	2009-04-28 00:31:08.000000000 +0200
@@ -46,15 +46,10 @@
 	$(CXXLD) -o $@ $(LDFLAGS) $(OBJECTS) $(LDLIBS)
 
 %.o:%.cpp
 	$(CXX) -c $(CXXFLAGS_ALL) $<
 
-# This common.mk dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/tools/common.mk: srcdir blddir
-
 include depend.mk
 
 .PHONY: clean
 clean: clean-common
 	rm -f $(PROGS)
diff -NrU5 xmlrpc-c-1.18.02.orig/tools/xml-rpc-api2txt/Makefile xmlrpc-c-1.18.02/tools/xml-rpc-api2txt/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xml-rpc-api2txt/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/tools/xml-rpc-api2txt/Makefile	2009-04-28 00:30:54.000000000 +0200
@@ -16,15 +16,10 @@
 
 all: $(PROGS)
 
 include $(SRCDIR)/tools/common.mk
 
-# This common.mk dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/tools/common.mk: srcdir blddir
-
 include depend.mk
 
 .PHONY: clean
 clean: clean-common
 	rm -f $(PROGS)
diff -NrU5 xmlrpc-c-1.18.02.orig/tools/xmlrpc_cpp_proxy/Makefile xmlrpc-c-1.18.02/tools/xmlrpc_cpp_proxy/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xmlrpc_cpp_proxy/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/tools/xmlrpc_cpp_proxy/Makefile	2009-04-28 00:28:49.000000000 +0200
@@ -45,15 +45,10 @@
 	$(CXXLD) -o $@ $(LDFLAGS) $(OBJECTS) $(LDLIBS) $(LADD)
 
 %.o:%.cpp $(BLDDIR)/include/xmlrpc-c/config.h
 	$(CXX) -c $(CXXFLAGS_ALL) $<
 
-# This common.mk dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/tools/common.mk: srcdir blddir
-
 include depend.mk
 
 .PHONY: clean
 clean: clean-common
 	rm -f xmlrpc_cpp_proxy
diff -NrU5 xmlrpc-c-1.18.02.orig/tools/xmlrpc_pstream/Makefile xmlrpc-c-1.18.02/tools/xmlrpc_pstream/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xmlrpc_pstream/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/tools/xmlrpc_pstream/Makefile	2009-04-28 00:38:11.000000000 +0200
@@ -16,22 +16,22 @@
 
 include $(SRCDIR)/tools/common.mk
 
 INCLUDES = \
   -I../lib/include \
-  -Isrcdir/lib/util/include \
-  -Iblddir \
-  -Iblddir/include \
-  -Isrcdir/include \
+  -I$(SRCDIR)/lib/util/include \
+  -I$(BLDDIR) \
+  -I$(BLDDIR)/include \
+  -I$(SRCDIR)/include \
 
 LDFLAGS = $(LADD)
 
 all: xmlrpc_pstream
 
 OBJECTS = \
   xmlrpc_pstream.o \
-  srcdir/tools/lib/dumpvalue.o \
+  $(SRCDIR)/tools/lib/dumpvalue.o \
 
 LIBS = \
   $(LIBXMLRPC_CLIENTPP) \
   $(LIBXMLRPCPP) \
   $(LIBXMLRPC_CLIENT) \
@@ -52,18 +52,13 @@
 UTILS = $(UTIL_OBJS:%=$(UTIL_DIR)/%)
 
 xmlrpc_pstream:  $(OBJECTS) $(LIBS) $(UTILS)
 	$(CXXLD) -o $@ $(LDFLAGS) $(OBJECTS) $(UTILS) $(LDLIBS) $(LADD)
 
-%.o:%.cpp blddir/include/xmlrpc-c/config.h
+%.o:%.cpp $(BLDDIR)/include/xmlrpc-c/config.h
 	$(CXX) -c $(CXXFLAGS_ALL) $<
 
-# This common.mk dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/tools/common.mk: srcdir blddir
-
 include depend.mk
 
 .PHONY: clean
 clean: clean-common
 	rm -f xmlrpc_pstream
diff -NrU5 xmlrpc-c-1.18.02.orig/tools/xmlrpc_transport/Makefile xmlrpc-c-1.18.02/tools/xmlrpc_transport/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xmlrpc_transport/Makefile	2009-04-28 00:23:33.000000000 +0200
+++ xmlrpc-c-1.18.02/tools/xmlrpc_transport/Makefile	2009-04-28 00:29:09.000000000 +0200
@@ -39,15 +39,10 @@
 	$(CCLD) -o $@ $(LDFLAGS) $< $(CLIENT_LDLIBS) $(UTILS)
 
 %.o:%.c
 	$(CC) -c $(CFLAGS_ALL) $<
 
-# This common.mk dependency makes sure the symlinks get built before
-# this make file is used for anything.
-
-$(SRCDIR)/tools/common.mk: srcdir blddir
-
 include depend.mk
 
 .PHONY: dep
 dep: dep-common
 
