diff -NrU5 xmlrpc-c-1.18.02.orig/tools/xmlrpc_pstream/Makefile xmlrpc-c-1.18.02/tools/xmlrpc_pstream/Makefile
--- xmlrpc-c-1.18.02.orig/tools/xmlrpc_pstream/Makefile	2009-04-26 16:05:12.000000000 +0200
+++ xmlrpc-c-1.18.02/tools/xmlrpc_pstream/Makefile	2009-04-26 22:37:06.000000000 +0200
@@ -27,11 +27,11 @@
 
 all: xmlrpc_pstream
 
 OBJECTS = \
   xmlrpc_pstream.o \
-  $(TOOLSDIR)/lib/dumpvalue.o \
+  srcdir/tools/lib/dumpvalue.o \
 
 LIBS = \
   $(LIBXMLRPC_CLIENTPP) \
   $(LIBXMLRPCPP) \
   $(LIBXMLRPC_CLIENT) \
