diff -Nuar slim-1.3.6/CMakeLists.txt slim-1.3.6-fix/CMakeLists.txt
--- slim-1.3.6/CMakeLists.txt	2013-10-01 22:38:05.000000000 +0000
+++ slim-1.3.6-fix/CMakeLists.txt	2014-12-05 06:51:29.360997817 +0000
@@ -171,6 +171,7 @@
 	${X11_Xrender_INCLUDE_PATH}
 	${X11_Xrandr_INCLUDE_PATH}
 	${FREETYPE_INCLUDE_DIR_freetype2}
+	${FREETYPE_INCLUDE_DIRS}
 	${X11_Xmu_INCLUDE_PATH}
 	${ZLIB_INCLUDE_DIR}
 	${JPEG_INCLUDE_DIR}
