diff -urN linux-2.6.32/net/bridge/br_input.c linux-2.6.32.bridgefix/net/bridge/br_input.c
--- linux-2.6.32/net/bridge/br_input.c	2011-11-11 21:17:05.079458352 -0700
+++ linux-2.6.32.bridgefix/net/bridge/br_input.c	2011-11-11 21:20:07.972224324 -0700
@@ -90,10 +90,12 @@
 			skb2 = skb;
 
 		br->dev->stats.multicast++;
-	} else if ((dst = __br_fdb_get(br, dest)) && dst->is_local) {
+	} else if ((dst = __br_fdb_get(br, dest)) != NULL) {
 		skb2 = skb;
 		/* Do not forward the packet since it's local. */
-		skb = NULL;
+		if (dst->is_local) {
+			skb = NULL;
+		}
 	}
 
 	if (skb2 == skb)
