diff -Nuar debian/config/config debian-fix/config/config
--- debian/config/config	2014-04-16 18:07:22.000000000 +0000
+++ debian-fix/config/config	2014-05-16 15:27:25.121252179 +0000
@@ -3783,7 +3783,7 @@
 ##
 ## file: fs/xfs/Kconfig
 ##
-CONFIG_XFS_FS=m
+CONFIG_XFS_FS=y
 CONFIG_XFS_QUOTA=y
 CONFIG_XFS_POSIX_ACL=y
 CONFIG_XFS_RT=y
@@ -3952,7 +3952,7 @@
 CONFIG_CRC_ITU_T=m
 CONFIG_CRC32=y
 CONFIG_CRC7=m
-CONFIG_LIBCRC32C=m
+CONFIG_LIBCRC32C=y
 
 ##
 ## file: lib/Kconfig.debug
