--- xen-3.1.3/tools/examples/network-bridge.orig	2008-01-31 02:06:54.000000000 -0800
+++ xen-3.1.3/tools/examples/network-bridge	2008-02-02 17:21:25.000000000 -0800
@@ -70,7 +70,7 @@
 vif0="vif0.${vifnum}"
 
 get_ip_info() {
-    addr_pfx=`ip addr show dev $1 | egrep '^ *inet' | sed -e 's/ *inet //' -e 's/ .*//'`
+    addr_pfx=`ip addr show dev $1 | egrep '^ *inet' | sed -e 's/ *inet //' -e "s/$1//"`
     gateway=`ip route show dev $1 | fgrep default | sed 's/default via //'`
 }
     
