diff -rpuN xen-3.0.4_1-src.orig/tools/xenmon/Makefile xen-3.0.4_1-src/tools/xenmon/Makefile
--- xen-3.0.4_1-src.orig/tools/xenmon/Makefile	2007-01-08 15:00:51.000000000 +0000
+++ xen-3.0.4_1-src/tools/xenmon/Makefile	2007-02-09 01:02:08.000000000 +0000
@@ -42,6 +42,6 @@ clean:
 
 
 %: %.c Makefile
-	$(CC) $(CFLAGS) $(LDFLAGS) -lxenctrl -o $@ $<
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< -lxenctrl
 xentrace_%: %.c Makefile
-	$(CC) $(CFLAGS) $(LDFLAGS) -lxenctrl -o $@ $<
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< -lxenctrl
