diff -Nru a/tools/libxc/xc_dom_x86.c b/tools/libxc/xc_dom_x86.c
--- a/tools/libxc/xc_dom_x86.c	2008-08-22 17:49:08.000000000 +0800
+++ b/tools/libxc/xc_dom_x86.c	2009-02-19 19:08:23.000000000 +0800
@@ -418,7 +418,7 @@
     xc_dom_printf("%s: called\n", __FUNCTION__);
 
     memset(start_info, 0, sizeof(*start_info));
-    snprintf(start_info->magic, sizeof(start_info->magic), dom->guest_type);
+    snprintf(start_info->magic, sizeof(start_info->magic), "%s", dom->guest_type);
     start_info->nr_pages = dom->total_pages;
     start_info->shared_info = shinfo << PAGE_SHIFT_X86;
     start_info->pt_base = dom->pgtables_seg.vstart;
@@ -457,7 +457,7 @@
     xc_dom_printf("%s: called\n", __FUNCTION__);
 
     memset(start_info, 0, sizeof(*start_info));
-    snprintf(start_info->magic, sizeof(start_info->magic), dom->guest_type);
+    snprintf(start_info->magic, sizeof(start_info->magic), "%s", dom->guest_type);
     start_info->nr_pages = dom->total_pages;
     start_info->shared_info = shinfo << PAGE_SHIFT_X86;
     start_info->pt_base = dom->pgtables_seg.vstart;
diff -Nru a/tools/xenstat/xentop/xentop.c b/tools/xenstat/xentop/xentop.c
--- a/tools/xenstat/xentop/xentop.c	2008-08-22 17:49:08.000000000 +0800
+++ b/tools/xenstat/xentop/xentop.c	2009-02-19 19:11:46.000000000 +0800
@@ -254,7 +254,7 @@
 {
 	if(cwin != NULL && !isendwin())
 		endwin();
-	fprintf(stderr, str);
+	fprintf(stderr, "%s", str);
 	exit(1);
 }
 
