diff -urN xen-3.3.1/tools/ioemu-qemu-xen/xen-setup xen-3.3.1-fix/tools/ioemu-qemu-xen/xen-setup
--- xen-3.3.1/tools/ioemu-qemu-xen/xen-setup	2009-01-05 04:36:22.000000000 -0700
+++ xen-3.3.1-fix/tools/ioemu-qemu-xen/xen-setup	2009-04-25 17:55:03.000000000 -0600
@@ -3,6 +3,8 @@
 
 # git-clean -x -d && ./xen-setup && make prefix=/usr CMDLINE_CFLAGS='-O0 -g' -j4 && make install DESTDIR=`pwd`/dist/ prefix=/usr && rsync -a --stats --delete . thule:shadow/qemu-iwj.git/ && rsync -a --stats dist/. root@thule:/
 
+target=i386-dm
+
 rm -f $target/Makefile
 rm -f $target/config.mak
 rm -f config-host.mak
@@ -11,8 +13,6 @@
 
 ./configure --disable-gfx-check --disable-gcc-check --disable-curses --disable-slirp "$@" --prefix=/usr
 
-target=i386-dm
-
 if [ "x$XEN_ROOT" != x ]; then
 	echo "XEN_ROOT=$XEN_ROOT" >>config-host.mak
 fi
