--- tools/ioemu-qemu-xen/xen-setup~	2009-01-05 11:36:22.000000000 +0000
+++ tools/ioemu-qemu-xen/xen-setup	2009-03-11 13:08:51.000000000 +0000
@@ -3,9 +3,9 @@
 
 # git-clean -x -d && ./xen-setup && make prefix=/usr CMDLINE_CFLAGS='-O0 -g' -j4 && make install DESTDIR=`pwd`/dist/ prefix=/usr && rsync -a --stats --delete . thule:shadow/qemu-iwj.git/ && rsync -a --stats dist/. root@thule:/
 
-rm -f $target/Makefile
-rm -f $target/config.mak
-rm -f config-host.mak
+[[ -f $target/Makefile ]] && rm -f $target/Makefile
+[[ -f $target/config.mak ]] && rm -f $target/config.mak
+[[ -f config-host.mak ]] && rm -f config-host.mak
 
 if test -f config-host.h; then mv config-host.h config-host.h~; fi
 
