diff -urN xen-3.3.1/install.sh xen-3.3.1-udev/install.sh
--- xen-3.3.1/install.sh	2009-01-05 04:26:58.000000000 -0700
+++ xen-3.3.1-udev/install.sh	2009-04-25 17:34:32.000000000 -0600
@@ -27,8 +27,8 @@
 echo "Installing Xen from '$src' to '$dst'..."
 (cd $src; tar -cf - * ) | tar -C "$tmp" -xf -
 
-[ -x "$(which udevinfo)" ] && \
-  UDEV_VERSION=$(udevinfo -V | sed -e 's/^[^0-9]* \([0-9]\{1,\}\)[^0-9]\{0,\}/\1/')
+[ -x "$(which udevadm)" ] && \
+  UDEV_VERSION=$(udevadm --version | sed -e 's/^[^0-9]* \([0-9]\{1,\}\)[^0-9]\{0,\}/\1/')
 
 if [ -n "$UDEV_VERSION" ] && [ $UDEV_VERSION -ge 059 ]; then
   echo " - installing for udev-based system"
diff -urN xen-3.3.1/tools/check/check_udev xen-3.3.1-udev/tools/check/check_udev
--- xen-3.3.1/tools/check/check_udev	2009-01-05 04:26:58.000000000 -0700
+++ xen-3.3.1-udev/tools/check/check_udev	2009-04-25 17:35:20.000000000 -0600
@@ -8,8 +8,8 @@
 	has_or_fail vnconfig
 	;;
 Linux)
-	has_or_fail udevinfo
-	[ "`udevinfo -V | awk '{print $NF}'`" -ge 59 ] 2>/dev/null || \
+	has_or_fail udevadm
+	[ "`udevadm --version | awk '{print $NF}'`" -ge 59 ] 2>/dev/null || \
 		has hotplug || \
 		fail "udev is too old, upgrade to version 59 or later"
 	;;
diff -urN xen-3.3.1/tools/examples/Makefile xen-3.3.1-udev/tools/examples/Makefile
--- xen-3.3.1/tools/examples/Makefile	2009-01-05 04:26:58.000000000 -0700
+++ xen-3.3.1-udev/tools/examples/Makefile	2009-04-25 17:36:10.000000000 -0600
@@ -52,7 +52,7 @@
 ifeq ($(findstring $(DI),$(DE)),$(DI))
 HOTPLUGS=install-hotplug install-udev
 else
-ifeq ($(shell [ -x /usr/bin/udevinfo ] && [ `/usr/bin/udevinfo -V | sed -e 's/^[^0-9]* \([0-9]\{1,\}\)[^0-9]\{0,\}/\1/'` -ge 059 ] && echo 1),1)
+ifeq ($(shell [ -x /sbin/udevadm ] && [ `/sbin/udevadm --version | sed -e 's/^[^0-9]* \([0-9]\{1,\}\)[^0-9]\{0,\}/\1/'` -ge 059 ] && echo 1),1)
 HOTPLUGS=install-udev
 else
 HOTPLUGS=install-hotplug
