diff -urN xen-3.3.1/tools/python/xen/util/acmpolicy.py xen-3.3.1-python-2.6/tools/python/xen/util/acmpolicy.py
--- xen-3.3.1/tools/python/xen/util/acmpolicy.py	2009-01-05 04:26:59.000000000 -0700
+++ xen-3.3.1-python-2.6/tools/python/xen/util/acmpolicy.py	2009-04-28 15:51:54.000000000 -0600
@@ -17,7 +17,7 @@
 #============================================================================
 
 import os
-import sha
+import hashlib
 import stat
 import array
 import struct
@@ -1103,7 +1103,9 @@
 
     def hash(self):
         """ Calculate a SAH1 hash of the XML policy """
-        return sha.sha(self.toxml())
+	myhash=hashlib.sha()
+	myhash.update(self.toxml())
+        return myhash 
 
     def save(self):
         ### Save the XML policy into a file ###
diff -urN xen-3.3.1/tools/python/xen/xend/XendAPI.py xen-3.3.1-python-2.6/tools/python/xen/xend/XendAPI.py
--- xen-3.3.1/tools/python/xen/xend/XendAPI.py	2009-01-05 04:26:59.000000000 -0700
+++ xen-3.3.1-python-2.6/tools/python/xen/xend/XendAPI.py	2009-04-28 15:56:50.000000000 -0600
@@ -18,7 +18,6 @@
 import inspect
 import os
 import Queue
-import sets
 import string
 import sys
 import traceback
@@ -116,7 +115,7 @@
 def event_register(session, reg_classes):
     if session not in event_registrations:
         event_registrations[session] = {
-            'classes' : sets.Set(),
+            'classes' : set(),
             'queue'   : Queue.Queue(EVENT_QUEUE_LENGTH),
             'next-id' : 1
             }
diff -urN xen-3.3.1/tools/python/xen/xend/encode.py xen-3.3.1-python-2.6/tools/python/xen/xend/encode.py
--- xen-3.3.1/tools/python/xen/xend/encode.py	2009-01-05 04:26:59.000000000 -0700
+++ xen-3.3.1-python-2.6/tools/python/xen/xend/encode.py	2009-04-28 15:21:45.000000000 -0600
@@ -27,7 +27,7 @@
 
 import urllib
 import random
-import md5
+import hashlib
 
 # Extract from HTML4 spec.
 ## The following example illustrates "multipart/form-data"
@@ -154,7 +154,7 @@
 
 def mime_boundary():
     random.seed()
-    m = md5.new()
+    m = hashlib.md5()
     for _ in range(0, 10):
         c = chr(random.randint(1, 255))
         m.update(c)
