diff -ru xen-3.0.4_1-src.orig/tools/examples/network-bridge xen-3.0.4_1-src/tools/examples/network-bridge
--- xen-3.0.4_1-src.orig/tools/examples/network-bridge	2007-01-08 07:00:48.000000000 -0800
+++ xen-3.0.4_1-src/tools/examples/network-bridge	2007-03-05 17:37:32.000000000 -0800
@@ -70,13 +70,13 @@
 vif0="vif0.${vifnum}"
 
 get_ip_info() {
-    addr_pfx=`ip addr show dev $1 | egrep '^ *inet' | sed -e 's/ *inet //' -e 's/ .*//'`
+    addr_pfx=`ip addr show dev $1 | egrep '^ *inet' | sed -e 's/ *inet //' -e "s/$1//"`
     gateway=`ip route show dev $1 | fgrep default | sed 's/default via //'`
 }
     
 do_ifup() {
     if ! ifup $1 ; then
-        if [ ${addr_pfx} ] ; then
+        if [ "${addr_pfx}" ] ; then
             # use the info from get_ip_info()
             ip addr flush $1
             ip addr add ${addr_pfx} dev $1
