--- gmult/DigitChooser.cpp
+++ gmult/DigitChooser.cpp
@@ -19,6 +19,7 @@
 #include "DigitChooser.h"
 #include "config.h"
 #include "gettext.h"
+#include <cstring>
 
 DigitChooser::DigitChooser (Gtk::Window &parent, MultPuzzle &puzzle, char letter)
 	: Gtk::Dialog ("", parent, true, false)
--- gmult/GtkMult.cpp
+++ gmult/GtkMult.cpp
@@ -3,6 +3,7 @@
 #include "config.h"
 #include "gettext.h"
 
+#include <cstring>
 #include <gtkmm/icontheme.h>
 
 GtkMult::GtkMult (void)
--- gmult/GtkMultTable.cpp
+++ gmult/GtkMultTable.cpp
@@ -22,7 +22,8 @@
 #include "DigitChooser.h"
 #include "config.h"
 #include "gettext.h"
-#include <ctype.h>
+#include <cctype>
+#include <cstring>
 
 GtkMultTable::GtkMultTable (MultPuzzle *p)
 	: Gtk::Table (1, 1, TRUE), puzzle (p)
