--- gcc/configure.orig	2005-08-20 04:28:13.000000000 -0400
+++ gcc/configure	2005-08-20 04:28:38.000000000 -0400
@@ -6790,7 +6790,7 @@
 # This prevents libgcc2 from containing any code which requires libc
 # support.
 inhibit_libc=
-if test x$host != x$target && test x$with_headers = x; then
+if test x$host != x$target && { test "x$with_headers" = x || test "x$with_headers" = xno ; } ; then
        inhibit_libc=-Dinhibit_libc
 else
        if test x$with_newlib = xyes; then
