--- gcc-3.2.2/gcc/ggc-page.c.orig	2003-03-22 03:52:54.000000000 +0200
+++ gcc-3.2.2/gcc/ggc-page.c	2003-03-22 03:54:17.000000000 +0200
@@ -344,11 +344,11 @@
    this factor times the allocation at the end of the last collection.
    In other words, total allocation must expand by (this factor minus
    one) before collection is performed.  */
-#define GGC_MIN_EXPAND_FOR_GC (1.3)
+#define GGC_MIN_EXPAND_FOR_GC (2.0)
 
-/* Bound `allocated_last_gc' to 16MB, to prevent the memory expansion
+/* Bound `allocated_last_gc' to 32MB, to prevent the memory expansion
    test from triggering too often when the heap is small.  */
-#define GGC_MIN_LAST_ALLOCATED (16 * 1024 * 1024)
+#define GGC_MIN_LAST_ALLOCATED (32 * 1024 * 1024)
 
 /* Allocate pages in chunks of this size, to throttle calls to memory
    allocation routines.  The first page is used, the rest go onto the
