http://xbmc.org/trac/ticket/5283

--- xbmc/utils/ArabicShaping.cpp
+++ xbmc/utils/ArabicShaping.cpp
@@ -148,10 +148,10 @@
   for (int i = 0; i < MAP_LENGTH; i++)
   {
     if (charsMap[i].code == c)
-      return FRIBIDI_TRUE;
+      return true;
   }
 
-  return FRIBIDI_FALSE;
+  return false;
 }
 
 static CharRep GetCharRep(FriBidiChar c)
@@ -186,9 +186,9 @@
   for (i = 0; i < TRANS_CHARS_LENGTH; i++)
   {
     if (transChars[i] == c)
-      return FRIBIDI_TRUE;
+      return true;
   }
-  return FRIBIDI_FALSE;
+  return false;
 }
 
 FriBidiChar* shape_arabic(FriBidiChar * str, int len)
--- xbmc/utils/ArabicShaping.h
+++ xbmc/utils/ArabicShaping.h
@@ -30,7 +30,7 @@
 #include "lib/libfribidi/fribidi.h"
 #else
 #include <fribidi/fribidi.h>
-#include <fribidi/fribidi_char_sets.h>
+#include <fribidi/fribidi-char-sets.h>
 #endif
 
 /**
--- xbmc/utils/CharsetConverter.cpp
+++ xbmc/utils/CharsetConverter.cpp
@@ -30,7 +30,7 @@
 #else
 #include <iconv.h>
 #include <fribidi/fribidi.h>
-#include <fribidi/fribidi_char_sets.h>
+#include <fribidi/fribidi-char-sets.h>
 #endif
 
 #ifdef __APPLE__
