--- prosody.cfg.lua.old	2010-05-27 11:10:45.000000000 +0200
+++ prosody.cfg.lua.dist	2010-05-27 11:12:21.000000000 +0200
@@ -16,6 +16,14 @@
 -- Settings in this section apply to the whole server and are the default settings
 -- for any virtual hosts
 
+prosody_user = "jabber"
+prosody_group = "jabber"
+pidfile = "/var/run/jabber/prosody.pid"
+
+log = "/var/log/jabber/prosody.log"
+-- Requires mod_posix to be loaded
+-- log = "*syslog"
+
 -- This is a (by default, empty) list of accounts that are admins
 -- for the server. Note that you must create the accounts separately
 -- (see http://prosody.im/doc/creating_accounts for info)
