From 6863b6979dd30781ff680aee346f164b38788103 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ralf=20Cors=C3=A9pius?= <corsepiu@fedoraproject.org>
Date: Sun, 3 Aug 2014 03:49:21 +0200
Subject: [PATCH 3/3] grass-6.4.2-docfiles

---
 Makefile | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 8f860c6..d2db723 100644
--- a/Makefile
+++ b/Makefile
@@ -60,10 +60,7 @@ else
 	LOCALE=0
 endif
 
-FILES = AUTHORS COPYING CHANGES REQUIREMENTS.html GPL.TXT contributors.csv  contributors_extra.csv  translators.csv
-
-BIN_DIST_FILES = $(FILES) \
-	grass${GRASS_VERSION_MAJOR}${GRASS_VERSION_MINOR}.tmp \
+BIN_DIST_FILES = grass${GRASS_VERSION_MAJOR}${GRASS_VERSION_MINOR}.tmp \
 	bin \
 	bwidget \
 	docs \
-- 
1.9.3

