diff --git a/Makefile b/Makefile
index 038ab84..f3385d3 100644
--- a/Makefile
+++ b/Makefile
@@ -9,37 +9,37 @@ squeezelite: main.o slimproto.o utils.o output.o buffer.o stream.o decode.o flac
 	$(CC) $(CFLAGS) main.o slimproto.o utils.o output.o buffer.o stream.o decode.o flac.o pcm.o mad.o vorbis.o faad.o $(LDFLAGS) $(MYLIBS) -o squeezelite
 
 main.o: main.c squeezelite.h
-	$(CC) $(CFLAGS) -c main.c -o main.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c main.c -o main.o
 
 slimproto.o: slimproto.c squeezelite.h slimproto.h
-	$(CC) $(CFLAGS) -c slimproto.c -o slimproto.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c slimproto.c -o slimproto.o
 
 utils.o: utils.c squeezelite.h
-	$(CC) $(CFLAGS) -c utils.c -o utils.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c utils.c -o utils.o
 
 output.o: output.c squeezelite.h
-	$(CC) $(CFLAGS) -c output.c -o output.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c output.c -o output.o
 
 buffer.o: buffer.c squeezelite.h
-	$(CC) $(CFLAGS) -c buffer.c -o buffer.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c buffer.c -o buffer.o
 
 stream.o: stream.c squeezelite.h
-	$(CC) $(CFLAGS) -c stream.c -o stream.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c stream.c -o stream.o
 
 decode.o: decode.c squeezelite.h
-	$(CC) $(CFLAGS) -c decode.c -o decode.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c decode.c -o decode.o
 
 flac.o: flac.c squeezelite.h
-	$(CC) $(CFLAGS) -c flac.c -o flac.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c flac.c -o flac.o
 
 pcm.o: pcm.c squeezelite.h
-	$(CC) $(CFLAGS) -c pcm.c -o pcm.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c pcm.c -o pcm.o
 
 mad.o: mad.c squeezelite.h
-	$(CC) $(CFLAGS) -c mad.c -o mad.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c mad.c -o mad.o
 
 vorbis.o: vorbis.c squeezelite.h
-	$(CC) $(CFLAGS) -c vorbis.c -o vorbis.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c vorbis.c -o vorbis.o
 
 faad.o: faad.c squeezelite.h
-	$(CC) $(CFLAGS) -c faad.c -o faad.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c faad.c -o faad.o
diff --git a/decode.c b/decode.c
index 9839d2e..340f580 100644
--- a/decode.c
+++ b/decode.c
@@ -95,11 +95,19 @@ void decode_init(log_level level, const char *opt) {
 	// register codecs
 	// alc,wma,wmap,wmal,aac,spt,ogg,ogf,flc,aif,pcm,mp3
 	int i = 0;
+#ifndef SL_NO_AAC
 	if (!opt || !strcmp(opt, "aac"))  codecs[i++] = register_faad();
+#endif
+#ifndef SL_NO_OGG
 	if (!opt || !strcmp(opt, "ogg"))  codecs[i++] = register_vorbis();
+#endif
+#ifndef SL_NO_FLAC
 	if (!opt || !strcmp(opt, "flac")) codecs[i++] = register_flac();
+#endif
 	if (!opt || !strcmp(opt, "pcm"))  codecs[i++] = register_pcm();
+#ifndef SL_NO_MP3
 	if (!opt || !strcmp(opt, "mp3"))  codecs[i++] = register_mad();
+#endif
 
 	pthread_attr_t attr;
 	pthread_attr_init(&attr);
diff --git a/faad.c b/faad.c
index 40b18a0..71cc77f 100644
--- a/faad.c
+++ b/faad.c
@@ -20,6 +20,8 @@
 
 #include "squeezelite.h"
 
+#ifndef SL_NO_AAC
+
 #include <neaacdec.h>
 #include <dlfcn.h>
 
@@ -569,3 +571,5 @@ struct codec *register_faad(void) {
 
 	return &ret;
 }
+
+#endif
diff --git a/flac.c b/flac.c
index 07e9c55..020a8eb 100644
--- a/flac.c
+++ b/flac.c
@@ -20,6 +20,8 @@
 
 #include "squeezelite.h"
 
+#ifndef SL_NO_FLAC
+
 #include <FLAC/stream_decoder.h>
 #include <dlfcn.h>
 
@@ -212,3 +214,5 @@ struct codec *register_flac(void) {
 
 	return &ret;
 }
+
+#endif
diff --git a/mad.c b/mad.c
index 842ec77..3ae14c7 100644
--- a/mad.c
+++ b/mad.c
@@ -20,6 +20,8 @@
 
 #include "squeezelite.h"
 
+#ifndef SL_NO_MP3
+
 #include <mad.h>
 #include <dlfcn.h>
 
@@ -298,3 +300,5 @@ struct codec *register_mad(void) {
 
 	return &ret;
 }
+
+#endif
diff --git a/vorbis.c b/vorbis.c
index 3aa820e..5ecc024 100644
--- a/vorbis.c
+++ b/vorbis.c
@@ -20,6 +20,8 @@
 
 #include "squeezelite.h"
 
+#ifndef SL_NO_OGG
+
 // automatically select between floating point (preferred) and fixed point libraries:
 #define LIBVORBIS "libvorbisfile.so.3"
 #define LIBTREMOR "libvorbisidec.so.1"
@@ -236,3 +238,5 @@ struct codec *register_vorbis(void) {
 
 	return &ret;
 }
+
+#endif
