diff --git a/Makefile b/Makefile
index d1aaf3c..038ab84 100644
--- a/Makefile
+++ b/Makefile
@@ -1,10 +1,12 @@
-CFLAGS  = -Wall -fPIC -O2
-LDFLAGS = -lasound -lpthread -ldl -lrt
+MYLIBS = -lasound -lpthread -ldl -lrt
 
 all: squeezelite
 
+clean:
+	rm *.o squeezelite 2>/dev/null || true
+
 squeezelite: main.o slimproto.o utils.o output.o buffer.o stream.o decode.o flac.o pcm.o mad.o vorbis.o faad.o
-	$(CC) $(CFLAGS) main.o slimproto.o utils.o output.o buffer.o stream.o decode.o flac.o pcm.o mad.o vorbis.o faad.o $(LDFLAGS) -o squeezelite
+	$(CC) $(CFLAGS) main.o slimproto.o utils.o output.o buffer.o stream.o decode.o flac.o pcm.o mad.o vorbis.o faad.o $(LDFLAGS) $(MYLIBS) -o squeezelite
 
 main.o: main.c squeezelite.h
 	$(CC) $(CFLAGS) -c main.c -o main.o
