--- gcc/version.c.old	2004-05-20 05:20:46.352804640 -0400
+++ gcc/version.c	2004-05-20 05:23:53.833303272 -0400
@@ -5,7 +5,7 @@
    please modify this string to indicate that, e.g. by putting your
    organization's name in parentheses at the end of the string.  */
 
-const char version_string[] = "3.4.2 20040803 (prerelease)";
+const char version_string[] = "3.4.1 @GENTOO@";
 
 /* This is the location of the online document giving instructions for
    reporting bugs.  If you distribute a modified version of GCC,
@@ -14,4 +14,4 @@
    forward us bugs reported to you, if you determine that they are
    not bugs in your modifications.)  */
 
-const char bug_report_url[] = "<URL:http://gcc.gnu.org/bugs.html>";
+const char bug_report_url[] = "<URL:http://bugs.gentoo.org/>";
