diff -urN pciutils-3.1.8/lib/init.c pciutils-3.1.8-r1/lib/init.c
--- pciutils-3.1.8/lib/init.c	2011-01-07 14:04:28.000000000 -0700
+++ pciutils-3.1.8-r1/lib/init.c	2011-12-16 10:49:42.534965034 -0700
@@ -198,8 +198,11 @@
 	      }
 	    a->debug("...No.\n");
 	  }
-      if (!a->methods)
+      if (!a->methods) {
 	a->error("Cannot find any working access method.");
+	/* Just in case error() does not call exit(), return safely: */
+	return;
+      }
     }
   a->debug("Decided to use %s\n", a->methods->name);
   a->methods->init(a);
