diff -Nuar gucharmap-3.6.1/configure.ac gucharmap-3.6.1-fixed/configure.ac
--- gucharmap-3.6.1/configure.ac	2012-11-12 22:14:29.000000000 +0000
+++ gucharmap-3.6.1-fixed/configure.ac	2013-12-17 11:04:22.337241541 +0000
@@ -237,7 +237,7 @@
   enable_vala=no
 fi
 m4_ifdef([VAPIGEN_CHECK],
-  [VAPIGEN_CHECK([0.16],[0.18])],
+  [VAPIGEN_CHECK([0.16])],
   [AM_CONDITIONAL([ENABLE_VAPIGEN],[false])])
 
 case "$with_gtk" in
