--- pyzy-0.1.0.orig/data/db/open-phrase/Makefile.am	2013-01-20 16:26:27.643846902 +0800
+++ pyzy-0.1.0/data/db/open-phrase/Makefile.am	2013-01-20 16:27:04.921846433 +0800
@@ -31,7 +31,7 @@
 
 $(DBTAR):
 	$(AM_V_GEN) \
-	wget http://pyzy.googlecode.com/files/$(DBTAR) || \
+	ln -s ${PORTDIR}/distfiles/$(DBTAR) || \
 	( $(RM) $@; exit 1)
 
 stamp-db: $(DBTAR)
--- pyzy-0.1.0.orig/data/db/open-phrase/Makefile.in	2013-01-20 16:35:16.609840253 +0800
+++ pyzy-0.1.0/data/db/open-phrase/Makefile.in	2013-01-20 16:36:30.448839325 +0800
@@ -492,7 +492,7 @@
 
 @PYZY_BUILD_DB_OPEN_PHRASE_TRUE@$(DBTAR):
 @PYZY_BUILD_DB_OPEN_PHRASE_TRUE@	$(AM_V_GEN) \
-@PYZY_BUILD_DB_OPEN_PHRASE_TRUE@	wget http://pyzy.googlecode.com/files/$(DBTAR) || \
+@PYZY_BUILD_DB_OPEN_PHRASE_TRUE@	ln -s ${PORTDIR}/distfiles/$(DBTAR) || \
 @PYZY_BUILD_DB_OPEN_PHRASE_TRUE@	( $(RM) $@; exit 1)
 
 @PYZY_BUILD_DB_OPEN_PHRASE_TRUE@stamp-db: $(DBTAR)
