diff -ur glurp-0.11.6.orig/configure.in glurp-0.11.6/configure.in
--- glurp-0.11.6.orig/configure.in	2005-06-05 18:23:54.000000000 +0300
+++ glurp-0.11.6/configure.in	2009-06-08 17:58:20.000000000 +0300
@@ -40,6 +40,10 @@
 AC_SUBST(PACKAGE_CFLAGS)
 AC_SUBST(PACKAGE_LIBS)
 
+PKG_CHECK_MODULES([libmpd], libmpd >= 0.17.0)
+AC_SUBST(libmpd_LIBS)
+AC_SUBST(libmpd_CFLAGS)
+
 dnl AC_SUBST(GETTEXT_PACKAGE)
 dnl AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"glurp")
 
Only in glurp-0.11.6.orig: glurp-0.11.6
diff -ur glurp-0.11.6.orig/src/libmpdclient.c glurp-0.11.6/src/libmpdclient.c
--- glurp-0.11.6.orig/src/libmpdclient.c	2005-04-03 04:56:08.000000000 +0300
+++ glurp-0.11.6/src/libmpdclient.c	2009-06-08 18:02:29.000000000 +0300
@@ -31,7 +31,7 @@
 
 */
 
-#include "libmpdclient.h"
+#include <libmpd/libmpdclient.h>
 
 #include <errno.h>
 #include <sys/types.h>
diff -ur glurp-0.11.6.orig/src/Makefile.am glurp-0.11.6/src/Makefile.am
--- glurp-0.11.6.orig/src/Makefile.am	2005-06-05 18:28:03.000000000 +0300
+++ glurp-0.11.6/src/Makefile.am	2009-06-08 17:59:27.000000000 +0300
@@ -2,7 +2,7 @@
 
 INCLUDES = \
 	-DDATADIR=\""$(datadir)/glurp/"\" \
-	@PACKAGE_CFLAGS@ -Wall
+	@PACKAGE_CFLAGS@ @libmpd_CFLAGS@ -Wall
 
 bin_PROGRAMS = glurp
 
@@ -12,7 +12,6 @@
 	gui.h gui.c \
 	gui-callbacks.c \
 	keyboard.c \
-	libmpdclient.h libmpdclient.c \
 	main.c \
 	player.h player.c \
 	structs.h \
@@ -20,4 +19,4 @@
 
 glurp_LDFLAGS = -export-dynamic
 
-glurp_LDADD = @PACKAGE_LIBS@
+glurp_LDADD = @PACKAGE_LIBS@ @libmpd_LIBS@
diff -ur glurp-0.11.6.orig/src/structs.h glurp-0.11.6/src/structs.h
--- glurp-0.11.6.orig/src/structs.h	2005-04-05 04:35:39.000000000 +0300
+++ glurp-0.11.6/src/structs.h	2009-06-08 18:02:10.000000000 +0300
@@ -20,7 +20,7 @@
 
 */
 
-#include "libmpdclient.h"
+#include <libmpd/libmpdclient.h>
 
 enum {
     PL_FILENAME,
