diff -NrU5 boo-0.9.1.3287.orig/default.build boo-0.9.1.3287/default.build
--- boo-0.9.1.3287.orig/default.build	2009-05-06 12:07:51.000000000 +0200
+++ boo-0.9.1.3287/default.build	2009-05-06 12:25:31.000000000 +0200
@@ -561,17 +561,24 @@
 				<include name="*.exe"/>
 				<include name="*.rsp"/>
 				<include name="*.config"/>
 				<include name="*.targets"/>
 				<include name="*.Tasks.dll" />
+				<include name="Boo.Lang.Extensions.dll" />
 			</fileset>
 		</copy>
 
 		<foreach item="File" property="filename">
 		<in>
 			<items>
-				<include name="${build.dir}/Boo.Lang*.dll" />
+				<include name="${build.dir}/Boo.Lang.dll" />
+				<include name="${build.dir}/Boo.Lang.CodeDom.dll" />
+				<include name="${build.dir}/Boo.Lang.Compiler.dll" />
+				<include name="${build.dir}/Boo.Lang.Interpreter.dll" />
+				<include name="${build.dir}/Boo.Lang.Parser.dll" />
+				<include name="${build.dir}/Boo.Lang.PatternMatching.dll" />
+				<include name="${build.dir}/Boo.Lang.Useful.dll" />
 			</items>
 		</in>
 		<do>
 			<exec program="gacutil" useruntimeengine="false" workingdir="${build.dir}">
 				<arg value="/i" />
