--- boo-0.7.8.2559/default.build~	2007-06-12 18:58:29.000000000 +0200
+++ boo-0.7.8.2559/default.build	2007-06-12 18:58:36.000000000 +0200
@@ -504,7 +504,7 @@
 
 		<property name="sharedmime.prefix" value="${pkg-config::get-variable('shared-mime-info','prefix')}" />
 		<property name="fakeroot.sharedmime" value="${install.destdir}/${sharedmime.prefix}" />
-		<property name="gsv.prefix" value="${pkg-config::get-variable('gtksourceview-1.0','prefix')}" />
+		<property name="gsv.prefix" value="${pkg-config::get-variable('gtksourceview-2.0','prefix')}" />
 		<property name="fakeroot.gsv" value="${install.destdir}/${gsv.prefix}" />
 
 		<mkdir dir="${fakeroot.boolib}"/>
