*** default.build	2011-01-20 13:25:04.000000000 +0100
--- default.build	2013-04-22 11:16:57.070753922 +0200
***************
*** 612,623 ****
  			</exec>
  		</do>
  		</foreach>
- 		
- 		<copy todir="${fakeroot.boolib}/pt">
- 			<fileset basedir="${build.dir}/pt">
- 				<include name="*"/>
- 			</fileset>
- 		</copy>
  
  		<copy todir="${fakeroot.bindir}">
  			<fileset basedir="${build.dir}">
--- 612,617 ----
