Build failure fix with GCC4.4
Index: fusecompress-2.6/src/rlog/rlog.h
===================================================================
--- fusecompress-2.6.orig/src/rlog/rlog.h	2009-10-16 16:52:25.022223417 +0530
+++ fusecompress-2.6/src/rlog/rlog.h	2009-10-16 16:52:41.478275160 +0530
@@ -1,6 +1,7 @@
 #ifndef RLOG_H
 #define RLOG_H
 
+#include <cstdio>
 #include <syslog.h>
 #include <stdarg.h>
 #include <string>
