Patch for fixing build failures with boost version >1.40
Index: fusecompress-2.6/src/boost/archive/portable_binary_iarchive.hpp
===================================================================
--- fusecompress-2.6.orig/src/boost/archive/portable_binary_iarchive.hpp	2009-10-06 21:56:47.625445953 +0530
+++ fusecompress-2.6/src/boost/archive/portable_binary_iarchive.hpp	2009-10-06 22:09:18.421446213 +0530
@@ -239,7 +239,13 @@
 };
 
 #include <boost/archive/impl/basic_binary_iarchive.ipp>
+
+#if BOOST_VERSION < 104000
 #include <boost/archive/impl/archive_pointer_iserializer.ipp>
+#else
+#include <boost/archive/impl/archive_serializer_map.ipp>
+#endif
+
 #include <boost/archive/impl/basic_binary_iprimitive.ipp>
 
 namespace boost {
@@ -266,7 +272,11 @@
 #endif
 >;
 
+#if BOOST_VERSION < 104000
 template class detail::archive_pointer_iserializer<portable_binary_iarchive> ;
+#else
+template class detail::archive_serializer_map<portable_binary_iarchive> ;
+#endif
 
 } // namespace archive
 } // namespace boost
Index: fusecompress-2.6/src/boost/archive/portable_binary_oarchive.hpp
===================================================================
--- fusecompress-2.6.orig/src/boost/archive/portable_binary_oarchive.hpp	2009-10-06 21:56:47.605444597 +0530
+++ fusecompress-2.6/src/boost/archive/portable_binary_oarchive.hpp	2009-10-06 22:09:18.421446213 +0530
@@ -229,7 +229,12 @@
 };
 
 #include <boost/archive/impl/basic_binary_oarchive.ipp>
+
+#if BOOST_VERSION < 104000
 #include <boost/archive/impl/archive_pointer_oserializer.ipp>
+// archive_serializer_map will get included in portable_binary_iarchive.hpp
+#endif
+
 #include <boost/archive/impl/basic_binary_oprimitive.ipp>
 
 namespace boost {
@@ -254,7 +259,11 @@
 	, std::ostream::traits_type
 #endif
 > ;
+
+#if BOOST_VERSION < 104000
 template class detail::archive_pointer_oserializer<portable_binary_oarchive> ;
+// defined in portable_binary_iarchive.hpp
+#endif
 
 } // namespace archive
 } // namespace boost
