diff -urN tk8.6.1/generic/tk.h tk8.6.1-patched/generic/tk.h
--- tk8.6.1/generic/tk.h	2013-09-03 08:52:34.000000000 -0400
+++ tk8.6.1-patched/generic/tk.h	2014-08-02 10:18:53.734757750 -0400
@@ -17,7 +17,7 @@
 #define _TK
 
 #include <tcl.h>
-#if (TCL_MAJOR_VERSION != 8) || (TCL_MINOR_VERSION < 6)
+#if (TCL_MAJOR_VERSION < 8) && (TCL_MINOR_VERSION < 6)
 #	error Tk 8.6 must be compiled with tcl.h from Tcl 8.6 or better
 #endif
 
