--- src/fe-gtk/maingui.c.old	2010-05-16 05:15:55.000000000 +0200
+++ src/fe-gtk/maingui.c	2010-07-27 05:10:22.000000000 +0200
@@ -2828,7 +2828,7 @@
 static void
 mg_create_entry (session *sess, GtkWidget *box)
 {
-	GtkWidget *sw, *hbox, *but, *entry;
+	GtkWidget *sw, *hbox, *but, *entry, *mybox;
 	session_gui *gui = sess->gui;
 
 	hbox = gtk_hbox_new (FALSE, 0);
@@ -2846,7 +2846,7 @@
 
 #ifdef USE_GTKSPELL
 	gui->input_box = entry = gtk_text_view_new ();
-	gtk_widget_set_size_request (entry, 0, 1);
+	gtk_widget_set_size_request (entry, 0, -1);
 	gtk_text_view_set_wrap_mode (GTK_TEXT_VIEW (entry), GTK_WRAP_NONE);
 	gtk_text_view_set_accepts_tab (GTK_TEXT_VIEW (entry), FALSE);
 	if (prefs.gui_input_spell)
@@ -2859,7 +2859,9 @@
 												GTK_POLICY_NEVER,
 												GTK_POLICY_NEVER);
 	gtk_container_add (GTK_CONTAINER (sw), entry);
-	gtk_container_add (GTK_CONTAINER (hbox), sw);
+	mybox = gtk_vbox_new(FALSE, 0);
+	gtk_box_pack_start (GTK_BOX (mybox), sw, 1, 0, 0);
+	gtk_box_pack_start (GTK_BOX (hbox), mybox, 1, 1, 0);
 #else
 #ifdef USE_LIBSEXY
 	gui->input_box = entry = sexy_spell_entry_new ();
