diff -ur xchat-2.8.8.orig/configure.in xchat-2.8.8/configure.in
--- xchat-2.8.8.orig/configure.in	2012-06-08 15:22:38.000000000 -0400
+++ xchat-2.8.8/configure.in	2012-06-08 15:23:27.000000000 -0400
@@ -151,7 +151,7 @@
 dnl ** GLIB *************************************************************
 dnl *********************************************************************
 
-AM_PATH_GLIB_2_0(2.0.3, glib=yes, glib=no)
+AM_PATH_GLIB_2_0(2.0.3, glib=yes, glib=no, gmodule)
 if test "$glib" = no; then
 	AC_MSG_ERROR("Cannot find glib")
 fi
