diff -Nurp genkernel-3.4.10.907.orig/arch/x86/modules_load genkernel-3.4.10.907/arch/x86/modules_load
--- genkernel-3.4.10.907.orig/arch/x86/modules_load	2009-12-17 05:21:37.000000000 +0100
+++ genkernel-3.4.10.907/arch/x86/modules_load	2010-04-10 21:53:00.590161869 +0200
@@ -28,3 +28,5 @@ MODULES_USB="ehci-hcd uhci usb-ohci hid
 # Filesystems
 MODULES_FS="ext2 ext3 reiserfs jfs nfs xfs fuse"
 
+# Crypto stuff, required for Sabayon crypt installs
+MODULES_CRYPT="aes sha1 sha256 sha512 xts" 
diff -Nurp genkernel-3.4.10.907.orig/arch/x86_64/modules_load genkernel-3.4.10.907/arch/x86_64/modules_load
--- genkernel-3.4.10.907.orig/arch/x86_64/modules_load	2009-12-17 05:21:37.000000000 +0100
+++ genkernel-3.4.10.907/arch/x86_64/modules_load	2010-04-10 21:53:12.358096340 +0200
@@ -26,3 +26,6 @@ MODULES_USB="ehci-hcd uhci usb-ohci hid
 
 # Filesystems
 MODULES_FS="ext2 ext3 reiserfs jfs nfs xfs fuse"
+
+# Crypto stuff, required for Sabayon crypt installs
+MODULES_CRYPT="aes sha1 sha256 sha512 xts"
