--- bash-3.0/config.h.in
+++ bash-3.0/config.h.in
@@ -197,7 +197,7 @@
 
 /* System paths */
 
-#define DEFAULT_MAIL_DIRECTORY "/usr/spool/mail"
+#define DEFAULT_MAIL_DIRECTORY "/var/spool/mail"
 
 /* Characteristics of the system's header files and libraries that affect
    the compilation environment. */
--- bash-3.0/config-bot.h
+++ bash-3.0/config-bot.h
@@ -178,4 +178,18 @@
 /******************************************************************/
 
 /* If you don't want bash to provide a default mail file to check. */
-/* #undef DEFAULT_MAIL_DIRECTORY */
+/* DP: - don't define a default DEFAULT_MAIL_DIRECTORY, because it
+ * DP:   can cause a timeout on NFS mounts.
+ */
+#undef DEFAULT_MAIL_DIRECTORY
+
+/* Force pgrp synchronization
+ * (https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=81653)
+ *
+ * The session will hang cases where you 'su' (not 'su -') and
+ * then run a piped command in emacs.
+ * This problem seem to happen due to scheduler changes kernel
+ * side - although reproduceble with later 2.4 kernels, it is
+ * especially easy with 2.6 kernels.
+ */
+#define PGRP_PIPE 1
--- bash-3.0/config-top.h
+++ bash-3.0/config-top.h
@@ -52,14 +52,14 @@
 /* The default value of the PATH variable. */
 #ifndef DEFAULT_PATH_VALUE
 #define DEFAULT_PATH_VALUE \
-  "/usr/gnu/bin:/usr/local/bin:/bin:/usr/bin:."
+  "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
 #endif
 
 /* The value for PATH when invoking `command -p'.  This is only used when
    the Posix.2 confstr () function, or CS_PATH define are not present. */
 #ifndef STANDARD_UTILS_PATH
 #define STANDARD_UTILS_PATH \
-  "/bin:/usr/bin:/sbin:/usr/sbin:/etc:/usr/etc"
+  "/bin:/usr/bin:/sbin:/usr/sbin"
 #endif
 
 /* Default primary and secondary prompt strings. */
@@ -74,14 +74,16 @@
 
 /* System-wide .bashrc file for interactive shells. */
 /* #define SYS_BASHRC "/etc/bash.bashrc" */
+#define SYS_BASHRC "/etc/bash/bashrc" /* #26952 */
 
 /* System-wide .bash_logout for login shells. */
 /* #define SYS_BASH_LOGOUT "/etc/bash.bash_logout" */
+#define SYS_BASH_LOGOUT "/etc/bash/bash_logout" /* #90488 */
 
 /* Define this to make non-interactive shells begun with argv[0][0] == '-'
    run the startup files when not in posix mode. */
-/* #define NON_INTERACTIVE_LOGIN_SHELLS */
+#define NON_INTERACTIVE_LOGIN_SHELLS
 
 /* Define this if you want bash to try to check whether it's being run by
    sshd and source the .bashrc if so (like the rshd behavior). */
-/* #define SSH_SOURCE_BASHRC */
+#define SSH_SOURCE_BASHRC /* #24762 */
