Index: jack-audio-connection-kit-0.103.0/configure.ac
===================================================================
--- jack-audio-connection-kit-0.103.0.orig/configure.ac
+++ jack-audio-connection-kit-0.103.0/configure.ac
@@ -210,8 +210,6 @@ AM_CONDITIONAL(USE_POSIX_SHM, $USE_POSIX
 
 JACK_CORE_CFLAGS="-I\$(top_srcdir)/config -I\$(top_srcdir) \
 -I\$(top_srcdir) -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -Wall -g"
-JACK_OPT_CFLAGS="$JACK_CORE_CFLAGS -march=pentium2 -mcpu=pentium4 -O3 \
--ffast-math -funroll-loops -fprefetch-loop-arrays"
 JACK_CFLAGS="$JACK_CORE_CFLAGS $CFLAGS"
 
 dnl
@@ -253,7 +251,7 @@ if test "$target_cpu" = "powerpc"; then
 
     dnl -mcpu=7450 does not reliably work with gcc 3.*
     
-    JACK_OPT_CFLAGS="-D_REENTRANT -O2 -mcpu=7400 $ALTIVEC_OPT_FLAGS_FLAGS -mhard-float -mpowerpc-gfxopt"
+    JACK_OPT_CFLAGS="-D_REENTRANT -mcpu=7400 $ALTIVEC_OPT_FLAGS_FLAGS -mhard-float -mpowerpc-gfxopt"
 
 elif echo $target_cpu | egrep '(i.86|x86_64)' >/dev/null; then
 
@@ -339,17 +337,7 @@ elif echo $target_cpu | egrep '(i.86|x86
     fi
 
     AC_DEFINE(x86, 1, "Nope it's intel")
-    COMMON_X86_OPT_FLAGS="-DREENTRANT -O3 -fomit-frame-pointer -ffast-math -funroll-loops"
-    
-    if test "$target_cpu" = "i586"; then
-	JACK_OPT_CFLAGS="-march=i586 "
-    elif test "$target_cpu" = "i686"; then
-	JACK_OPT_CFLAGS="-march=i686"
-    elif test "$target_cpu" = "x86_64"; then
-	JACK_OPT_CFLAGS="-march=k8"
-    else
-	:
-    fi
+    COMMON_X86_OPT_FLAGS="-DREENTRANT"
     
     JACK_OPT_CFLAGS="$COMMON_X86_OPT_FLAGS $JACK_OPT_CFLAGS $MMX_FLAGS $SSE_FLAGS"
 
