--- gcc-3.4.3/gcc/common.opt
+++ gcc-3.4.3/gcc/common.opt
@@ -152,6 +152,10 @@ Wunused-variable
 Common
 Warn when a variable is unused
 
+Wstack-protector
+Common
+Warn when not issuing stack smashing protection for some reason
+
 aux-info
 Common Separate
 -aux-info <file>	Emit declaration information into <file>
@@ -743,6 +747,14 @@ fzero-initialized-in-bss
 Common
 Put zero initialized data in the bss section
 
+fstack-protector
+Common
+Enables stack protection
+
+fstack-protector-all
+Common
+Enables stack protection of every function
+
 g
 Common JoinedOrMissing
 Generate debug information in default format
--- gcc-3.4.3/gcc/opts.c
+++ gcc-3.4.3/gcc/opts.c
@@ -804,6 +807,13 @@ common_handle_option (size_t scode, cons
       warn_unused_variable = value;
       break;
 
+    case OPT_fstack_protector:
+    case OPT_fstack_protector_all:
+    case OPT_Wstack_protector:
+      if (value)
+        warning ("ssp stub: stack protector is not supported");
+      break;
+
     case OPT_aux_info:
     case OPT_aux_info_:
       aux_info_file_name = arg;
