--- extensions/xt_TARPIT.c	2014-09-29 03:33:13.000000000 -0500
+++ extensions/xt_TARPIT.c.new	2015-06-29 11:38:26.582851153 -0500
@@ -250,7 +250,7 @@
 
 #ifdef CONFIG_BRIDGE_NETFILTER
 	if (hook != NF_INET_FORWARD || (nskb->nf_bridge != NULL &&
-	    nskb->nf_bridge->mask & BRNF_BRIDGED))
+	    nskb->nf_bridge->physoutdev))
 #else
 	if (hook != NF_INET_FORWARD)
 #endif
@@ -279,7 +279,7 @@
 
 	nf_ct_attach(nskb, oldskb);
 
-	NF_HOOK(NFPROTO_IPV4, NF_INET_LOCAL_OUT, nskb, NULL,
+	NF_HOOK(NFPROTO_IPV4, NF_INET_LOCAL_OUT, NULL, nskb, NULL,
 		skb_dst(nskb)->dev, dst_output);
 	return;
 
@@ -394,7 +394,7 @@
 
 	nf_ct_attach(nskb, oldskb);
 
-	NF_HOOK(NFPROTO_IPV6, NF_INET_LOCAL_OUT, nskb, NULL,
+	NF_HOOK(NFPROTO_IPV6, NF_INET_LOCAL_OUT, NULL, nskb, NULL,
 	        skb_dst(nskb)->dev, dst_output);
 	return;
 
