diff -urN linux/net/bridge/br_stp_if.c drobtmp/net/bridge/br_stp_if.c
--- linux/net/bridge/br_stp_if.c	2009-06-20 01:46:25.000000000 -0600
+++ drobtmp/net/bridge/br_stp_if.c	2009-06-27 16:47:57.000000000 -0600
@@ -163,10 +163,7 @@
 	struct net_bridge_port *p;
 
 	list_for_each_entry(p, &br->port_list, list) {
-		if (addr == br_mac_zero ||
-		    memcmp(p->dev->dev_addr, addr, ETH_ALEN) < 0)
-			addr = p->dev->dev_addr;
-
+	    addr = p->dev->dev_addr;
 	}
 
 	if (compare_ether_addr(br->bridge_id.addr, addr))
