--- src/freenet-old/node/NodeStarter.java	2010-05-10 21:19:25.000000000 +0200
+++ src/freenet/node/NodeStarter.java	2010-05-10 21:35:30.000000000 +0200
@@ -168,37 +168,8 @@
 	}
 
 	private void getExtBuild() {
-		try {
-			extBuildNumber = ExtVersion.buildNumber;
-			extRevisionNumber = ExtVersion.cvsRevision;
-			String builtWithMessage = "freenet.jar built with freenet-ext.jar Build #" + extBuildNumber + " r" + extRevisionNumber;
-			Logger.normal(this, builtWithMessage);
-			System.out.println(builtWithMessage);
-			extBuildNumber = ExtVersion.buildNumber();
-			if(extBuildNumber == -42) {
-				extBuildNumber = ExtVersion.extBuildNumber();
-				extRevisionNumber = ExtVersion.extRevisionNumber();
-			}
-			if(extBuildNumber == 0) {
-				String buildMessage = "extBuildNumber is 0; perhaps your freenet-ext.jar file is corrupted?";
-				Logger.error(this, buildMessage);
-				System.err.println(buildMessage);
-				extBuildNumber = -1;
-			}
-			if(extRevisionNumber == null) {
-				String revisionMessage = "extRevisionNumber is null; perhaps your freenet-ext.jar file is corrupted?";
-				Logger.error(this, revisionMessage);
-				System.err.println(revisionMessage);
-				extRevisionNumber = "INVALID";
-			}
-		} catch(Throwable t) {
-			// Compatibility code ... will be removed
-			Logger.error(this, "Unable to get the version of your freenet-ext file : it's probably corrupted!");
-			System.err.println("Unable to get the version of your freenet-ext file : it's probably corrupted!");
-			System.err.println(t.getMessage());
-			extRevisionNumber = "INVALID";
-			extBuildNumber = -1;
-		}
+		extRevisionNumber = "";
+		extBuildNumber = RECOMMENDED_EXT_BUILD_NUMBER;
 	}
 
 	/**
--- build.xml	2010-05-10 21:38:15.000000000 +0200
+++ build.xml.new	2010-05-10 21:43:29.000000000 +0200
@@ -124,7 +124,7 @@
 	</condition>
 	<!-- ================================================== -->
 
-	<target name="compile" depends="get-extjar, generate-js">
+	<target name="compile" depends="mkdir">
 		<!-- Create the time stamp -->
 		<tstamp/>
 
