--- setup/Linux/oss/build/osscore.c	2011-12-29 10:30:56.071144892 +0800
+++ setup/Linux/oss/build/osscore.c	2011-12-29 10:36:19.735147341 +0800
@@ -911,7 +911,7 @@
  *	are replaced by wrapper handles.
  */
 
-static struct file_operations *
+static file_operations_no_const *
 alloc_fop (oss_device_t * osdev, struct oss_file_operation_handle *op)
 {
 /*
@@ -927,7 +927,7 @@
  *	and WRAPPER_VERSION must be incremented.
  */
 
-  struct file_operations *fop;
+  file_operations_no_const *fop;
 
   poll_t tmp_poll = (poll_t) op->poll;
   read_t tmp_read = (read_t) op->read;
@@ -942,10 +942,10 @@
   new_ioctl_t tmp_unlocked_ioctl = (new_ioctl_t) op->unlocked_ioctl;
   new_ioctl_t tmp_compat_ioctl = (new_ioctl_t) op->compat_ioctl;
 
-  fop = (struct file_operations *)
-    oss_kmem_alloc (sizeof (struct file_operations));
+  fop = (file_operations_no_const *)
+    oss_kmem_alloc (sizeof (file_operations_no_const));
 
-  memset ((char *) fop, 0, sizeof (struct file_operations));
+  memset ((char *) fop, 0, sizeof (file_operations_no_const));
 
 /*
  *	Now the assignment
