diff -Naur samba-3.0.28a.orig/source/Makefile.in samba-3.0.28a/source/Makefile.in
--- samba-3.0.28a.orig/source/Makefile.in	2008-05-17 12:20:23.000000000 +0200
+++ samba-3.0.28a/source/Makefile.in	2008-05-17 12:23:38.000000000 +0200
@@ -1666,7 +1666,8 @@
 	fi
 	PYTHON_OBJS="$(PYTHON_OBJ)" \
 	PYTHON_CFLAGS="$(CFLAGS) $(CPPFLAGS) $(FLAGS)" \
-	LIBS="$(LDFLAGS) $(LIBS) $(PASSDB_LIBS) $(IDMAP_LIBS) $(KRB5LIBS) $(LDAP_LIBS)" \
+	LDFLAGS="$(LDSHFLAGS)" \
+	LIBS="$(LDSHFLAGS) $(LIBS) $(PASSDB_LIBS) $(IDMAP_LIBS) $(KRB5LIBS) $(LDAP_LIBS)" \
 		$(PYTHON) python/setup.py build
 
 python_install: $(PYTHON_OBJ)
@@ -1676,8 +1677,9 @@
 	fi
 	PYTHON_OBJS="$(PYTHON_OBJ)" \
 	PYTHON_CFLAGS="$(CFLAGS) $(CPPFLAGS)" \
-	LIBS="$(LDFLAGS) $(LIBS)" \
-		$(PYTHON) python/setup.py install --root=$(DESTDIR)
+	LDFLAGS="$(LDSHFLAGS)" \
+	LIBS="$(LDSHFLAGS) $(LIBS)" \
+		$(PYTHON) python/setup.py install --root="$(DESTDIR)"
 
 python_clean:
 	@-if test -n "$(PYTHON)"; then $(PYTHON) python/setup.py clean; fi
