diff --git a/source/libsmb/libsmbclient.c b/source/libsmb/libsmbclient.c
index 80b09fa..374a86a 100644
--- a/source/libsmb/libsmbclient.c
+++ b/source/libsmb/libsmbclient.c
@@ -2644,7 +2644,11 @@ smbc_opendir_ctx(SMBCCTX *context,
                                 return NULL;
                         }
 
-                        ip_list = &server_addr;
+			ip_list = memdup(&server_addr, sizeof(server_addr));
+			if (ip_list == NULL) {
+				errno = ENOMEM;
+				return NULL;
+			}
                         count = 1;
                 }
 
