diff -Naur samba-3.0.28.orig/source/tests/os2_delete.c samba-3.0.28/source/tests/os2_delete.c
--- samba-3.0.28.orig/source/tests/os2_delete.c	2007-11-15 04:15:04.000000000 +0100
+++ samba-3.0.28/source/tests/os2_delete.c	2008-02-20 20:47:53.878055947 +0100
@@ -105,6 +105,9 @@
 
 	return 0;
 }
+
+#if 0
+
 /*
   test readdir/unlink pattern that OS/2 uses
   tridge@samba.org July 2005
@@ -212,3 +215,4 @@
 
 	return 0;
 }
+#endif
