diff --git a/mcomix/histogram.py b/mcomix/histogram.py
index 9ab63b3..bef1f92 100644
--- a/mcomix/histogram.py
+++ b/mcomix/histogram.py
@@ -1,8 +1,6 @@
 """histogram.py - Draw histograms (RGB) from pixbufs."""
 
-import Image
-import ImageDraw
-import ImageOps
+from PIL import Image, ImageDraw, ImageOps
 
 from mcomix import image_tools
 
diff --git a/mcomix/image_tools.py b/mcomix/image_tools.py
index c1f39c8..4c87088 100644
--- a/mcomix/image_tools.py
+++ b/mcomix/image_tools.py
@@ -6,9 +6,7 @@ import operator
 import itertools
 import bisect
 import gtk
-import Image
-import ImageEnhance
-import ImageOps
+from PIL import Image, ImageEnhance, ImageOps
 
 from mcomix.preferences import prefs
 
diff --git a/mcomix/library/book_area.py b/mcomix/library/book_area.py
index 79c44ed..ab53a00 100644
--- a/mcomix/library/book_area.py
+++ b/mcomix/library/book_area.py
@@ -4,8 +4,7 @@ import os
 import urllib
 import gtk
 import gobject
-import Image
-import ImageDraw
+from PIL import Image, ImageDraw
 
 from mcomix.preferences import prefs
 from mcomix import thumbnail_view
diff --git a/mcomix/run.py b/mcomix/run.py
index 85c856b..541167b 100644
--- a/mcomix/run.py
+++ b/mcomix/run.py
@@ -138,7 +138,7 @@ except ImportError:
 
 # Check PIL library
 try:
-    import Image
+    from PIL import Image
     assert Image.VERSION >= '1.1.5'
 
 except AssertionError:
diff --git a/mcomix/thumbnail_tools.py b/mcomix/thumbnail_tools.py
index 542624a..9bdd970 100644
--- a/mcomix/thumbnail_tools.py
+++ b/mcomix/thumbnail_tools.py
@@ -9,7 +9,7 @@ import tempfile
 import mimetypes
 import threading
 import itertools
-import Image
+from PIL import Image
 from urllib import pathname2url
 
 try:  # The md5 module is deprecated as of Python 2.5, replaced by hashlib.
