--- bindings/c/src/conf2xml.c.orig	2006-11-28 09:49:23 +0100
+++ bindings/c/src/conf2xml.c	2006-11-28 09:49:34 +0100
@@ -7,7 +7,6 @@
 #include <string.h>
 #include "conf2xml.h"
 
-#include <error.h>
 #include <errno.h>
 /* #define ARGC(a) ((sizeof(a)/sizeof(a[0]))) */
 /* #define STR_MAX_SIZE 256 */
@@ -39,17 +38,17 @@
 
 char * conf2xml (char *filename, char *template, char *template_options) {
     STRLEN l;
-    if (my_perl == NULL) error (NOT_INITIALIZED, 0, "conf2xml has not been initialized. Call conf2xml_init first");
+    if (my_perl == NULL) errc (NOT_INITIALIZED, 0, "conf2xml has not been initialized. Call conf2xml_init first");
     
     if (!template) {
         template = malloc(sizeof(char));
-        if (!template) error (ENOMEM, ENOMEM, NULL);
+        if (!template) errc (ENOMEM, ENOMEM, NULL);
         *template = '\0';
     }
 
     if (!template_options) {
         template_options = malloc(sizeof(char));
-        if (!template_options) error (ENOMEM, ENOMEM, NULL);;
+        if (!template_options) errc (ENOMEM, ENOMEM, NULL);;
         *template_options = '\0';
     }
 
@@ -66,7 +65,7 @@
                                       1
                                       );
     char *perl_code = malloc(sizeof(char) * code_length);
-    if (!perl_code) error (ENOMEM, ENOMEM, NULL);;
+    if (!perl_code) errc (ENOMEM, ENOMEM, NULL);;
     *perl_code = '\0';
     strncat(perl_code, perl_code1, strlen(perl_code1));
     strncat(perl_code, filename, strlen(filename));
--- bindings/python/src/setup.py.orig	2006-11-28 09:50:43 +0100
+++ bindings/python/src/setup.py	2006-11-28 09:50:50 +0100
@@ -12,7 +12,6 @@
     ext_modules=[
         Extension(
             'conf2xml',
-            sources=['conf2xmlmodule.c'],
-            libraries=['dl'],
+            sources=['conf2xmlmodule.c']
             )]
     )
