diff -Naur grub-0.97/stage2/Makefile.am grub-0.97.final/stage2/Makefile.am
--- grub-0.97/stage2/Makefile.am	2011-09-07 14:04:55.605902544 -0400
+++ grub-0.97.final/stage2/Makefile.am	2011-09-07 14:08:15.822903632 -0400
@@ -85,7 +85,7 @@
 GRAPHICS_FLAGS =
 endif
 
-STAGE2_COMPILE = $(STAGE2_CFLAGS) -fno-builtin -nostdinc \
+STAGE2_COMPILE = $(STAGE2_CFLAGS) -fno-reorder-functions -fno-builtin -nostdinc \
 	$(NETBOOT_FLAGS) $(SERIAL_FLAGS) $(HERCULES_FLAGS) $(GRAPHICS_FLAGS)
 
 STAGE1_5_LINK = -nostdlib -Wl,-N -Wl,-Ttext -Wl,2000
