diff -urN openrc-0.12.3/src/librc/librc-daemon.c openrc-0.12.3-fixed/src/librc/librc-daemon.c
--- openrc-0.12.3/src/librc/librc-daemon.c	2013-10-22 20:59:16.000000000 -0600
+++ openrc-0.12.3-fixed/src/librc/librc-daemon.c	2014-05-20 13:38:17.399787235 -0600
@@ -90,6 +90,10 @@
 {
 	DIR *procdir;
 	struct dirent *entry;
+	FILE *fp;
+	bool host_process = true;
+	char *line = NULL;
+	size_t len = 0;
 	pid_t p;
 	char buffer[PATH_MAX];
 	struct stat sb;
@@ -134,6 +138,22 @@
 		if (argv &&
 		    !pid_is_argv(p, (const char *const *)argv))
 			continue;
+		snprintf(buffer, sizeof(buffer), "/proc/%d/status", p);
+		if (exists(buffer)) {
+			fp = fopen(buffer, "r");
+			if (! fp)
+				continue;
+			while (! feof(fp)) {
+				rc_getline(&line, &len, fp);
+				if (strncmp(line, "envID:", 6) == 0) {
+					host_process = (strncmp(line, "envID:\t0", 8) == 0);
+					break;
+				}
+			}
+			fclose(fp);
+		}
+		if (! host_process)
+			continue;
 		if (!pids) {
 			pids = xmalloc(sizeof(*pids));
 			LIST_INIT(pids);
@@ -142,6 +162,8 @@
 		pi->pid = p;
 		LIST_INSERT_HEAD(pids, pi, entries);
 	}
+	if (line != NULL)
+		free(line);
 	closedir(procdir);
 	return pids;
 }
