diff -Nuar openrc-0.10.2/init.d/Makefile openrc-0.10.2-nodevfs/init.d/Makefile
--- openrc-0.10.2/init.d/Makefile	2012-11-25 19:19:35.000000000 +0000
+++ openrc-0.10.2-nodevfs/init.d/Makefile	2013-06-27 01:33:33.439315203 +0000
@@ -20,7 +20,7 @@
 		powerd.in syscons.in
 
 NET_LO-Linux=	net.lo
-SRCS-Linux=	devfs.in dmesg.in hwclock.in consolefont.in keymaps.in \
+SRCS-Linux=	dmesg.in hwclock.in consolefont.in keymaps.in \
 		killprocs.in modules.in mount-ro.in mtab.in numlock.in \
 		procfs.in sysfs.in termencoding.in
 
diff -Nuar openrc-0.10.2/runlevels/Makefile openrc-0.10.2-nodevfs/runlevels/Makefile
--- openrc-0.10.2/runlevels/Makefile	2012-11-25 19:19:35.000000000 +0000
+++ openrc-0.10.2-nodevfs/runlevels/Makefile	2013-06-27 01:34:35.018382905 +0000
@@ -28,7 +28,7 @@
 
 BOOT-Linux=	hwclock keymaps modules mtab net.lo procfs termencoding
 SHUTDOWN-Linux=	killprocs mount-ro
-SYSINIT-Linux=	devfs dmesg
+SYSINIT-Linux=	dmesg
 
 # Generic BSD stuff
 BOOT-NetBSD=	hostid net.lo0 newsyslog savecore syslogd
