diff -urN logitechmediaserver-7.7.1-33735-noCPAN.orig/slimserver.pl logitechmediaserver-7.7.1-33735-noCPAN/slimserver.pl
--- logitechmediaserver-7.7.1-33735-noCPAN.orig/slimserver.pl	2011-11-28 23:43:24.000000000 +0000
+++ logitechmediaserver-7.7.1-33735-noCPAN/slimserver.pl	2012-01-22 22:26:50.365777949 +0000
@@ -440,8 +440,9 @@
 	
 	# Generate a UUID for this SC instance on first-run
 	if ( !$prefs->get('server_uuid') ) {
-		require UUID::Tiny;
-		$prefs->set( server_uuid => UUID::Tiny::create_UUID_as_string( UUID::Tiny::UUID_V4() ) );
+		require Data:UUID;
+		my $ug = new Data::UUID;
+		$prefs->set( server_uuid => lc $ug->create_str() );
 	}
 
 	main::INFOLOG && $log->info("Server binary search path init...");
