diff -NrU5 iodine-0.5.2.original/src/Makefile iodine-0.5.2/src/Makefile
--- iodine-0.5.2.original/src/Makefile	2009-08-07 19:58:30.000000000 -0600
+++ iodine-0.5.2/src/Makefile	2009-08-07 20:34:01.000000000 -0600
@@ -1,18 +1,18 @@
-CC = gcc
+#CC = gcc
 COMMONOBJS = tun.o dns.o read.o encoding.o login.o base32.o base64.o md5.o common.o
 CLIENTOBJS = iodine.o
 CLIENT = ../bin/iodine
 SERVEROBJS = iodined.o user.o fw_query.o
 SERVER = ../bin/iodined
 
 OS = `echo $(TARGETOS) | tr "a-z" "A-Z"`
 ARCH = `uname -m`
 
 LIBPATH = -L.
-LDFLAGS =  -lz `sh osflags $(TARGETOS) link` $(LIBPATH)
-CFLAGS = -c -g -Wall -D$(OS) -pedantic `sh osflags $(TARGETOS) cflags`
+LDFLAGS +=  -lz `sh osflags $(TARGETOS) link` $(LIBPATH)
+CFLAGS += -c -g -Wall -D$(OS) -pedantic `sh osflags $(TARGETOS) cflags`
 
 all: stateos $(CLIENT) $(SERVER)
 
 stateos:
 	@echo OS is $(OS), arch is $(ARCH)
@@ -27,11 +27,11 @@
 	@mkdir -p ../bin
 	@$(CC) $(COMMONOBJS) $(SERVEROBJS) -o $(SERVER) $(LDFLAGS)
 
 .c.o: 
 	@echo CC $<
-	@$(CC) $(CFLAGS) $< -o $@
+	@$(CC) $(CFLAGS) $(LDFLAGS) $< -o $@
 
 clean:
 	@echo "Cleaning src/"
 	@rm -f $(CLIENT){,.exe} $(SERVER){,.exe} *~ *.o *.core
 
