diff -ur portage-utils-0.21.orig/qlop.c portage-utils-0.21/qlop.c
--- portage-utils-0.21.orig/qlop.c	2013-01-09 13:30:29.991861128 -0500
+++ portage-utils-0.21/qlop.c	2013-01-09 13:31:21.397881153 -0500
@@ -311,35 +311,32 @@
 {
 	FILE *fp;
 	size_t buflen;
-	char *buf, *p, *q;
+	char *buf, *path, *pathend;
 	time_t t;
 
 	if ((fp = fopen(logfile, "r")) == NULL) {
-		warnp("Could not open logfile '%s'", logfile);
+		printf("Could not open logfile '%s'", logfile);
 		return;
 	}
 
 	buf = NULL;
 	while (getline(&buf, &buflen, fp) != -1) {
-		if (strlen(buf) < 35)
+		if (strlen(buf) < 24)
 			continue;
-		if (strncmp(buf+12, "=== Sync completed with", 23) != 0)
+
+		if (strncmp(buf+12, ">>> Git pull", 12) != 0)
 			continue;
 
-		if ((p = strchr(buf, '\n')) != NULL)
-			*p = 0;
-		if ((p = strchr(buf, ':')) == NULL)
+		if (strstr(buf, "success") == NULL)
 			continue;
-		*p = 0;
-		q = p+2;
 
-		t = (time_t)atol(buf);
+		path = strchr(buf, '/');
+		pathend = strchr(path, ' ');
+               *pathend = 0;
 
-		if ((p = strstr(q, "with")) == NULL)
-			continue;
-		q = p + 5;
+		t = (time_t)atol(buf);
 
-		printf("%s >>> %s%s%s\n", chop_ctime(t), GREEN, q, NORM);
+		printf("%s >>> %sSync to %s successful %s \n", chop_ctime(t), GREEN, path, NORM);
 	}
 
 	free(buf);
