--- gcc-3.2.2/gcc/mklibgcc.in.mklibgcc-serialize-crtfiles	2003-02-04 05:04:51.000000000 -0500
+++ gcc-3.2.2/gcc/mklibgcc.in	2003-03-16 18:01:48.000000000 -0500
@@ -404,6 +404,9 @@ echo '	  if [ -d $$d ]; then true; else 
 echo '	done'
 echo '	if [ -f stmp-dirs ]; then true; else touch stmp-dirs; fi'
 
+# serialize build, since they may have to generate a new c-parse.y but
+# two jobs could be started in parallel thus colliding
+prev_out=""
 for f in $EXTRA_MULTILIB_PARTS; do
   for ml in $MULTILIBS; do
     dir=`echo ${ml} | sed -e 's/;.*$//' -e 's/=/$(EQ)/g'`
@@ -418,12 +421,13 @@ for f in $EXTRA_MULTILIB_PARTS; do
     *) targ=$out ;;
     esac
 
-    echo $out: stmp-dirs
+    echo $out: stmp-dirs $prev_out
     echo "	$make_compile" \\
     echo '	  LIBGCC2_CFLAGS="$(LIBGCC2_CFLAGS)' $flags '" ' \\
     echo '	  MULTILIB_CFLAGS="'$flags'"' T=$t $targ
 
     all="$all $out"
+    prev_out="$out"
   done
 done
 
