diff -Naurp gcc-3.2.3.orig/gcc/ada/gnatvsn.ads gcc-3.2.3/gcc/ada/gnatvsn.ads
--- gcc-3.2.3.orig/gcc/ada/gnatvsn.ads	2003-04-22 02:16:06.000000000 -0400
+++ gcc-3.2.3/gcc/ada/gnatvsn.ads	2003-04-26 12:20:22.000000000 -0400
@@ -38,7 +38,7 @@
 
 package Gnatvsn is
 
-   Gnat_Version_String : constant String := "3.2.3 20030422 (release)";
+   Gnat_Version_String : constant String := "3.2.3 20030422 (Gentoo 1.4)";
    --  Version output when GNAT (compiler), or its related tools, including
    --  GNATBIND, GNATCHOP, GNATFIND, GNATLINK, GNATMAKE, GNATXREF, are run
    --  (with appropriate verbose option switch set).
diff -Naurp gcc-3.2.3.orig/gcc/f/version.c gcc-3.2.3/gcc/f/version.c
--- gcc-3.2.3.orig/gcc/f/version.c	2003-04-22 02:16:09.000000000 -0400
+++ gcc-3.2.3/gcc/f/version.c	2003-04-26 12:25:38.000000000 -0400
@@ -1,4 +1,4 @@
 #include "ansidecl.h"
 #include "f/version.h"
 
-const char *const ffe_version_string = "3.2.3 20030422 (release)";
+const char *const ffe_version_string = "3.2.3 20030422 (Gentoo Linux 1.4 @PV@)";
diff -Naurp gcc-3.2.3.orig/gcc/system.h gcc-3.2.3/gcc/system.h
--- gcc-3.2.3.orig/gcc/system.h	2003-03-30 20:14:02.000000000 -0500
+++ gcc-3.2.3/gcc/system.h	2003-04-26 12:24:55.000000000 -0400
@@ -26,7 +26,7 @@ Software Foundation, 59 Temple Place - S
 /* This is the location of the online document giving information how
    to report bugs. If you change this string, also check for strings
    not under control of the preprocessor.  */
-#define GCCBUGURL "<URL:http://gcc.gnu.org/bugs.html>"
+#define GCCBUGURL "<URL:http://bugs.gentoo.org/>"
 
 /* We must include stdarg.h/varargs.h before stdio.h.  */
 #ifdef ANSI_PROTOTYPES
diff -Naurp gcc-3.2.3.orig/gcc/version.c gcc-3.2.3/gcc/version.c
--- gcc-3.2.3.orig/gcc/version.c	2003-04-22 02:16:03.000000000 -0400
+++ gcc-3.2.3/gcc/version.c	2003-04-26 12:25:28.000000000 -0400
@@ -1,4 +1,4 @@
 #include "ansidecl.h"
 #include "version.h"
 
-const char *const version_string = "3.2.3";
+const char *const version_string = "3.2.3 20030422 (Gentoo Linux 1.4 @PV@)";
