diff -u -r -N gcc-3.2.3.orig/gcc/output.h gcc-3.2.3/gcc/output.h
--- gcc-3.2.3.orig/gcc/output.h	2003-07-14 17:55:33.000000000 +0200
+++ gcc-3.2.3/gcc/output.h	2003-07-14 17:58:16.000000000 +0200
@@ -257,8 +257,13 @@
 
 extern void assemble_alias		PARAMS ((tree, tree));
 
+#endif
+
+/* hppa breaks when this is put into TREE_CODE */
 extern void default_assemble_visibility	PARAMS ((tree, const char *));
 
+#ifdef TREE_CODE
+
 /* Output a string of literal assembler code
    for an `asm' keyword used between functions.  */
 extern void assemble_asm		PARAMS ((tree));
