diff --git a/netcat.c b/netcat.c
index b31dce2..792112f 100644
--- a/netcat.c
+++ b/netcat.c
@@ -64,6 +64,8 @@
 #define PORT_MAX_LEN	6
 #define UNIX_DG_TMP_SOCKET_SIZE	19
 
+#define UDP_SCAN_TIMEOUT 3
+
 /* Command Line Options */
 int	dflag;					/* detached, no stdin */
 unsigned int iflag;				/* Interval Flag */
@@ -413,7 +415,7 @@ main(int argc, char *argv[])
 				continue;
 
 			ret = 0;
-			if (vflag || zflag) {
+			if ((vflag && !uflag) || zflag) {
 				/* For UDP, make sure we are connected. */
 				if (uflag) {
 					if (udptest(s) == -1) {
@@ -860,15 +862,20 @@ build_ports(char *p)
 int
 udptest(int s)
 {
-	int i, ret;
-
-	for (i = 0; i <= 3; i++) {
-		if (write(s, "X", 1) == 1)
-			ret = 1;
-		else
-			ret = -1;
+	int i, t;
+
+	if ((write(s, "X", 1) != 1) ||
+		((write(s, "X", 1) != 1) && ( errno == ECONNREFUSED)))
+		return -1;
+	
+	/* Give the remote host some time to reply */
+	for (i = 0, t = (timeout == -1) ? UDP_SCAN_TIMEOUT : (timeout / 1000);
+		 i < t; i++) {
+		sleep(1);
+		if ((write(s, "X", 1) != 1) && (errno == ECONNREFUSED))
+			return -1;
 	}
-	return (ret);
+	return 1;
 }
 
 void
