diff --git a/netcat.c b/netcat.c
index fca6ad5..b3143ef 100644
--- a/netcat.c
+++ b/netcat.c
@@ -953,14 +953,14 @@ map_tos(char *s, int *val)
 		{ "af42",		IPTOS_DSCP_AF42 },
 		{ "af43",		IPTOS_DSCP_AF43 },
 		{ "critical",		IPTOS_PREC_CRITIC_ECP },
-		{ "cs0",		IPTOS_DSCP_CS0 },
-		{ "cs1",		IPTOS_DSCP_CS1 },
-		{ "cs2",		IPTOS_DSCP_CS2 },
-		{ "cs3",		IPTOS_DSCP_CS3 },
-		{ "cs4",		IPTOS_DSCP_CS4 },
-		{ "cs5",		IPTOS_DSCP_CS5 },
-		{ "cs6",		IPTOS_DSCP_CS6 },
-		{ "cs7",		IPTOS_DSCP_CS7 },
+		{ "cs0",		IPTOS_CLASS_CS0 },
+		{ "cs1",		IPTOS_CLASS_CS1 },
+		{ "cs2",		IPTOS_CLASS_CS2 },
+		{ "cs3",		IPTOS_CLASS_CS3 },
+		{ "cs4",		IPTOS_CLASS_CS4 },
+		{ "cs5",		IPTOS_CLASS_CS5 },
+		{ "cs6",		IPTOS_CLASS_CS6 },
+		{ "cs7",		IPTOS_CLASS_CS7 },
 		{ "ef",			IPTOS_DSCP_EF },
 		{ "inetcontrol",	IPTOS_PREC_INTERNETCONTROL },
 		{ "lowdelay",		IPTOS_LOWDELAY },
