diff --git a/netcat.c b/netcat.c
index cb36f88..b7e4d5e 100644
--- a/netcat.c
+++ b/netcat.c
@@ -48,7 +48,7 @@
 #include <poll.h>
 #include <stdarg.h>
 #include <stdio.h>
-#include <stdlib.h>
+#include <bsd/stdlib.h>
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
diff --git a/socks.c b/socks.c
index 909c86e..c71453b 100644
--- a/socks.c
+++ b/socks.c
@@ -34,7 +34,7 @@
 #include <errno.h>
 #include <netdb.h>
 #include <stdio.h>
-#include <stdlib.h>
+#include <bsd/stdlib.h>
 #include <string.h>
 #include <unistd.h>
 #include <resolv.h>
