--- openntpd-3.9p1/client.c	2009-01-08 10:47:49.000000000 +0100
+++ openntpd-3.9p1-noipv6/client.c	2009-01-08 18:14:32.000000000 +0100
@@ -130,8 +130,20 @@
 		struct sockaddr *sa = (struct sockaddr *)&p->addr->ss;

 		if ((p->query->fd = socket(p->addr->ss.ss_family, SOCK_DGRAM,
-		    0)) == -1)
+		    0)) == -1) {
+		
+		   if (errno == EAFNOSUPPORT) {
+		      log_warn("client_query socket");
+		      client_nextaddr(p);
+		      set_next(p, error_interval());
+		      return (-1);
+		   }
+		   else
+		   {
 			fatal("client_query socket");
+		   }
+                }
+
 		if (connect(p->query->fd, sa, SA_LEN(sa)) == -1) {
 			if (errno == ECONNREFUSED || errno == ENETUNREACH ||
 			    errno == EHOSTUNREACH) {
