diff -Nuar autogen-5.18.4/autoopts/mk-tpl-config.sh autogen-5.18.4-fix/autoopts/mk-tpl-config.sh
--- autogen-5.18.4/autoopts/mk-tpl-config.sh	2014-07-07 23:47:25.000000000 +0000
+++ autogen-5.18.4-fix/autoopts/mk-tpl-config.sh	2015-01-16 10:52:02.870207643 +0000
@@ -214,7 +214,7 @@
 
 fix_guile() {
     cd ${builddir}
-    find_libguiledir "${LGCFLAGS}"
+    find_libguiledir ${LGCFLAGS}
 
     list=`exec 2>/dev/null
         find ${libguiledir}/libguile* -type f | \
