diff --git a/src/engine/client/ec_snd.c b/src/engine/client/ec_snd.c
index 034dfb2..67e819c 100644
--- a/src/engine/client/ec_snd.c
+++ b/src/engine/client/ec_snd.c
@@ -359,13 +359,6 @@ static void rate_convert(int sid)
 }
 
 
-static IOHANDLE file = NULL;
-
-static int read_data(void *buffer, int size)
-{
-	return io_read(file, buffer, size);
-}
-
 int snd_load_wv(const char *filename)
 {
 	SAMPLE *snd;
@@ -381,19 +374,12 @@ int snd_load_wv(const char *filename)
 	if(!sound_enabled)
 		return 1;
 
-	file = engine_openfile(filename, IOFLAG_READ); /* TODO: use system.h stuff for this */
-	if(!file)
-	{
-		dbg_msg("sound/wv", "failed to open %s", filename);
-		return -1;
-	}
-
 	sid = snd_alloc_id();
 	if(sid < 0)
 		return -1;
 	snd = &samples[sid];
 
-	context = WavpackOpenFileInput(read_data, error);
+	context = WavpackOpenFileInput(filename, error, OPEN_2CH_MAX, 0);
 	if (context)
 	{
 		int samples = WavpackGetNumSamples(context);
@@ -448,9 +434,6 @@ int snd_load_wv(const char *filename)
 		dbg_msg("sound/wv", "failed to open %s: %s", filename, error);
 	}
 
-	io_close(file);
-	file = NULL;
-
 	if(config.debug)
 		dbg_msg("sound/wv", "loaded %s", filename);
 
