diff -Naur ATLAS/tune/blas/ger/r1hgen.c ATLAS.new/tune/blas/ger/r1hgen.c
--- ATLAS/tune/blas/ger/r1hgen.c	2010-02-07 19:41:01.000000000 -0500
+++ ATLAS.new/tune/blas/ger/r1hgen.c	2010-02-18 11:54:46.000000000 -0500
@@ -1030,7 +1030,7 @@
       r1b->XU = xu;
       r1b->YU = yu;
       r1b->flag = iflag;
-      r1b->CacheElts = l1mul * GetL1CacheElts(pre);
+      r1b->CacheElts = (l1mul/100.0) * GetL1CacheElts(pre);
       r1hgen(pre, path, -1, r1b, NULL);
       exit(0);
    }
