http://bugs.gentoo.org/211712

--- Make.top
+++ Make.top
@@ -27,70 +27,70 @@
 
 
 shared: INSTALLER = cp
-shared: libatlas.so libblas.so libcblas.so
+shared: libatlas.dylib libblas.dylib libcblas.dylib
 
 shared-strip: INSTALLER = install -s
-shared-strip: libatlas.so libblas.so libcblas.so
+shared-strip: libatlas.dylib libblas.dylib libcblas.dylib
 
 ptshared: INSTALLER = cp
-ptshared: libptblas.so libptcblas.so
+ptshared: libptblas.dylib libptcblas.dylib
 
 ptshared-strip: INSTALLER = install -s
-ptshared-strip: libptblas.so libptcblas.so
+ptshared-strip: libptblas.dylib libptcblas.dylib
 
-libatlas.so:
+libatlas.dylib:
 	mkdir -p gentoo/libs
 	@echo
 	@echo Linking a really big library, please be patient...
 	@echo
 	cd gentoo/libatlas.a ; \
-	libtool --mode=link --tag=CC $(CC) $(LDFLAGS) $(LIBS) -o libatlas.la \
-		*.lo -rpath /usr/$(LIBDIR) ; \
-	libtool --mode=install $(INSTALLER) libatlas.la $(BLDdir)/gentoo/libs
+	glibtool --mode=link --tag=CC $(CC) $(LDFLAGS) $(LIBS) -o libatlas.la \
+		*.lo -rpath $(RPATH) ; \
+	glibtool --mode=install $(INSTALLER) libatlas.la $(BLDdir)/gentoo/libs
 
-libblas.so:
+libblas.dylib:
 	cd gentoo/libf77blas.a ; \
-       	libtool --mode=link --tag=CC $(CC) $(LDFLAGS) $(F77SYSLIB) -o libblas.la \
+       	glibtool --mode=link --tag=CC $(CC) $(LDFLAGS) $(F77SYSLIB) -o libblas.la \
        		../libs/libatlas.la *.lo -rpath $(RPATH); \
-	rm -f .libs/libblas.so.0.0.0; \
-	$(CC) -shared $(LDFLAGS) .libs/*.o $(F77SYSLIB) $(BLDdir)/gentoo/libs/libatlas.so \
-       	      -Wl,-soname -Wl,libblas.so.0 -o .libs/libblas.so.0.0.0; \
-	libtool --mode=install $(INSTALLER) libblas.la $(BLDdir)/gentoo/libs
+	rm -f .libs/libblas.0.0.0.dylib; \
+	$(CC) -dynamiclib $(LDFLAGS) .libs/*.o $(F77SYSLIB) $(BLDdir)/gentoo/libs/libatlas.dylib \
+       	      -install_name $(RPATH)/libblas.0.dylib -o .libs/libblas.0.0.0.dylib; \
+	glibtool --mode=install $(INSTALLER) libblas.la $(BLDdir)/gentoo/libs
 
-libcblas.so:
+libcblas.dylib:
 	cd gentoo/libcblas.a ; \
-       	libtool --mode=link --tag=CC $(CC) $(LDFLAGS) -o libcblas.la \
+       	glibtool --mode=link --tag=CC $(CC) $(LDFLAGS) -o libcblas.la \
 		../libs/libatlas.la *.lo -rpath $(RPATH) ; \
-       	rm -f .libs/libcblas.so.0.0.0; \
-       	$(CC) -shared $(LDFLAGS) .libs/*.o $(BLDdir)/gentoo/libs/libatlas.so \
-	       -Wl,-soname -Wl,libcblas.so.0 -o .libs/libcblas.so.0.0.0; \
-       libtool --mode=install $(INSTALLER) libcblas.la $(BLDdir)/gentoo/libs
+       	rm -f .libs/libcblas.0.0.0.dylib; \
+       	$(CC) -dynamiclib $(LDFLAGS) .libs/*.o $(BLDdir)/gentoo/libs/libatlas.dylib \
+	       -install_name $(RPATH)/libcblas.0.dylib -o .libs/libcblas.0.0.0.dylib; \
+       glibtool --mode=install $(INSTALLER) libcblas.la $(BLDdir)/gentoo/libs
 
-libptblas.so:
+libptblas.dylib:
 	mkdir -p gentoo/threaded-libs
 	cd gentoo/libptf77blas.a ; \
-	libtool --mode=link --tag=CC $(CC) $(LDFLAGS) $(F77SYSLIB) -o libblas.la \
+	glibtool --mode=link --tag=CC $(CC) $(LDFLAGS) $(F77SYSLIB) -o libblas.la \
 		../libs/libatlas.la *.lo -rpath $(RPATH); \
-	rm -f .libs/libblas.so.0.0.0; \
-	$(CC) -shared $(LDFLAGS) .libs/*.o  $(F77SYSLIB) $(BLDdir)/gentoo/libs/libatlas.so \
-	      -Wl,-soname -Wl,libblas.so.0 -o .libs/libblas.so.0.0.0; \
-	libtool --mode=install $(INSTALLER) libblas.la $(BLDdir)/gentoo/threaded-libs
+	rm -f .libs/libblas.0.0.0.dylib; \
+	$(CC) -dynamiclib $(LDFLAGS) .libs/*.o  $(F77SYSLIB) $(BLDdir)/gentoo/libs/libatlas.dylib \
+	      -install_name $(RPATH)/libblas.0.dylib -o .libs/libblas.0.0.0.dylib; \
+	glibtool --mode=install $(INSTALLER) libblas.la $(BLDdir)/gentoo/threaded-libs
 
-libptcblas.so:
+libptcblas.dylib:
 	cd gentoo/libptcblas.a ; \
-	libtool --mode=link --tag=CC $(CC) $(LDFLAGS) -o libcblas.la \
+	glibtool --mode=link --tag=CC $(CC) $(LDFLAGS) -o libcblas.la \
 		../libs/libatlas.la *.lo -rpath $(RPATH); \
-       	rm -f .libs/libcblas.so.0.0.0; \
-       	$(CC) -shared $(LDFLAGS) .libs/*.o $(BLDdir)/gentoo/libs/libatlas.so \
-	      -Wl,-soname -Wl,libcblas.so.0 -o .libs/libcblas.so.0.0.0; \
-       	libtool --mode=install $(INSTALLER) libcblas.la $(BLDdir)/gentoo/threaded-libs
+       	rm -f .libs/libcblas.0.0.0.dylib; \
+       	$(CC) -dynamiclib $(LDFLAGS) .libs/*.o $(BLDdir)/gentoo/libs/libatlas.dylib \
+	      -install_name $(RPATH)/libcblas.0.dylib -o .libs/libcblas.0.0.0.dylib; \
+       	glibtool --mode=install $(INSTALLER) libcblas.la $(BLDdir)/gentoo/threaded-libs
 
-liblapack.so:
+liblapack.dylib:
 	mkdir -p gentoo/libs
 	cd gentoo/liblapack.a ; \
-       	libtool --mode=link --tag=CC $(CC) $(LDFLAGS) $(F77SYSLIB) -o liblapack.la \
+       	glibtool --mode=link --tag=CC $(CC) $(LDFLAGS) $(F77SYSLIB) -o liblapack.la \
                 *.lo -rpath $(RPATH) -lblas -lcblas -latlas ; \
-	libtool --mode=install $(INSTALLER) liblapack.la $(BLDdir)/gentoo/libs
+	glibtool --mode=install $(INSTALLER) liblapack.la $(BLDdir)/gentoo/libs
 
 
 sanity_test :
