diff -urN postfix-2.8.12/conf/master.cf postfix-2.8.12-fixed/conf/master.cf
--- postfix-2.8.12/conf/master.cf	2010-12-31 14:14:51.000000000 +0000
+++ postfix-2.8.12-fixed/conf/master.cf	2012-08-11 19:26:03.327178818 +0000
@@ -8,7 +8,7 @@
 # service type  private unpriv  chroot  wakeup  maxproc command + args
 #               (yes)   (yes)   (yes)   (never) (100)
 # ==========================================================================
-smtp      inet  n       -       n       -       -       smtpd
+smtp      unix  n       -       y       -       -       smtpd
 #smtp      inet  n       -       n       -       1       postscreen
 #smtpd     pass  -       -       n       -       -       smtpd
 #dnsblog   unix  -       -       n       -       0       dnsblog
