From 54ce24f4947256e1991970eed40bafd1b7028cf6 Mon Sep 17 00:00:00 2001
From: Josselin Mouette <joss@debian.org>
Date: Tue, 12 Oct 2010 19:19:44 +0000
Subject: housekeeping: More network filesystems not to monitor

https://bugzilla.gnome.org/show_bug.cgi?id=606421
---
diff --git a/plugins/housekeeping/gsd-disk-space.c b/plugins/housekeeping/gsd-disk-space.c
index e1385ea..2fcc523 100644
--- a/plugins/housekeeping/gsd-disk-space.c
+++ b/plugins/housekeeping/gsd-disk-space.c
@@ -304,25 +304,41 @@ ldsm_mount_should_ignore (GUnixMountEntry *mount)
          * expose this in a way that allows it to be used for this
          * purpose
          */
+         
+         /* We also ignore network filesystems */
                  
         const gchar *ignore_fs[] = {
+                "adfs",
+                "afs",
                 "auto",
                 "autofs",
+                "autofs4",
+                "cifs",
+                "cxfs",
                 "devfs",
                 "devpts",
                 "ecryptfs",
+                "gfs",
+                "gfs2",
                 "kernfs",
                 "linprocfs",
+                "linsysfs",
+                "lustre",
+                "lustre_lite",
+                "ncpfs",
+                "nfs",
+                "nfs4",
+                "nfsd",
+                "ocfs2",
                 "proc",
                 "procfs",
                 "ptyfs",
+                "rpc_pipefs",
                 "selinuxfs",
-                "linsysfs",
+                "smbfs",
                 "sysfs",
                 "tmpfs",
                 "usbfs",
-                "nfsd",
-                "rpc_pipefs",
                 "zfs",
                 NULL
         };
--
cgit v0.8.3.1
