diff -Naur gnome-settings-daemon-3.3.92//configure.ac gnome-settings-daemon-3.3.92.new//configure.ac
--- gnome-settings-daemon-3.3.92//configure.ac	2012-03-19 14:23:38.000000000 +0530
+++ gnome-settings-daemon-3.3.92.new//configure.ac	2012-03-25 11:36:35.833356226 +0530
@@ -194,13 +194,39 @@
 dnl - color
 dnl ---------------------------------------------------------------------------
 
-PKG_CHECK_MODULES(COLOR, [colord >= 0.1.9 gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION libcanberra-gtk3])
+AC_ARG_ENABLE([color],
+  AS_HELP_STRING([--disable-color],
+                 [turn off color plugin]),
+       [with_color=$enableval],
+       [with_color=yes]) dnl Default value
+
+if test x$with_color = xyes; then
+       PKG_CHECK_MODULES(COLOR, [colord >= 0.1.9 gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION libcanberra-gtk3])
+       AC_DEFINE(HAVE_COLOR, 1, [Define if wacom plugin is enabled])
+fi
+AM_CONDITIONAL(WITH_COLOR, test "x$with_color" = "xyes")
+
+AC_SUBST(COLOR_CFLAGS)
+AC_SUBST(COLOR_LIBS)
 
 dnl ---------------------------------------------------------------------------
 dnl - wacom
 dnl ---------------------------------------------------------------------------
 
-PKG_CHECK_MODULES(WACOM, [libwacom >= $LIBWACOM_REQUIRED_VERSION x11 xi xtst $GUDEV_PKG gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION xorg-wacom])
+AC_ARG_ENABLE([wacom],
+  AS_HELP_STRING([--disable-wacom],
+                 [turn off wacom plugin]),
+       [with_wacom=$enableval],
+       [with_wacom=no]) dnl Default value
+
+if test x$with_wacom = xyes; then
+       PKG_CHECK_MODULES(WACOM, [libwacom >= $LIBWACOM_REQUIRED_VERSION x11 xi xtst $GUDEV_PKG gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION xorg-wacom])
+       AC_DEFINE(HAVE_WACOM, 1, [Define if wacom plugin is enabled])
+fi
+AM_CONDITIONAL(WITH_WACOM, test "x$with_wacom" = "xyes")
+
+AC_SUBST(WACOM_CFLAGS)
+AC_SUBST(WACOM_LIBS)
 
 dnl ==============================================
 dnl PackageKit section
diff -Naur gnome-settings-daemon-3.3.92//data/Makefile.am gnome-settings-daemon-3.3.92.new//data/Makefile.am
--- gnome-settings-daemon-3.3.92//data/Makefile.am	2012-03-16 22:59:35.000000000 +0530
+++ gnome-settings-daemon-3.3.92.new//data/Makefile.am	2012-03-25 11:39:24.548364338 +0530
@@ -11,16 +11,20 @@
 	org.gnome.settings-daemon.plugins.gschema.xml			\
 	org.gnome.settings-daemon.plugins.keyboard.gschema.xml		\
 	org.gnome.settings-daemon.plugins.power.gschema.xml		\
-	org.gnome.settings-daemon.plugins.color.gschema.xml		\
 	org.gnome.settings-daemon.plugins.media-keys.gschema.xml	\
 	org.gnome.settings-daemon.plugins.xsettings.gschema.xml		\
 	org.gnome.settings-daemon.plugins.housekeeping.gschema.xml	\
 	org.gnome.settings-daemon.plugins.print-notifications.gschema.xml	\
-	org.gnome.settings-daemon.peripherals.wacom.gschema.xml		\
 	org.gnome.settings-daemon.plugins.xrandr.gschema.xml
 
 all_schemas = $(gsettings_SCHEMAS)
 
+if WITH_COLOR
+gsettings_SCHEMAS += org.gnome.settings-daemon.plugins.color.gschema.xml
+else
+all_schemas += org.gnome.settings-daemon.plugins.color.gschema.xml
+endif
+
 if HAVE_PACKAGEKIT
 gsettings_SCHEMAS += org.gnome.settings-daemon.plugins.updates.gschema.xml
 else
@@ -33,6 +37,12 @@
 all_schemas += org.gnome.settings-daemon.plugins.orientation.gschema.xml
 endif
 
+if WITH_WACOM
+gsettings_SCHEMAS += org.gnome.settings-daemon.peripherals.wacom.gschema.xml
+else
+all_schemas += org.gnome.settings-daemon.peripherals.wacom.gschema.xml
+endif
+
 @INTLTOOL_XML_NOMERGE_RULE@
 
 @GSETTINGS_RULES@
diff -Naur gnome-settings-daemon-3.3.92//plugins/Makefile.am gnome-settings-daemon-3.3.92.new//plugins/Makefile.am
--- gnome-settings-daemon-3.3.92//plugins/Makefile.am	2012-03-16 22:59:35.000000000 +0530
+++ gnome-settings-daemon-3.3.92.new//plugins/Makefile.am	2012-03-25 11:37:47.296359651 +0530
@@ -6,7 +6,6 @@
 	automount	\
 	background	\
 	clipboard	\
-	color		\
 	cursor		\
 	dummy		\
 	power		\
@@ -15,13 +14,18 @@
 	media-keys	\
 	mouse		\
 	sound		\
-	wacom		\
 	xrandr		\
 	xsettings	\
 	$(NULL)
 
 disabled_plugins = $(NULL)
 
+if WITH_COLOR
+enabled_plugins += color
+else
+disabled_plugins += color
+endif
+
 if HAVE_PACKAGEKIT
 enabled_plugins += updates
 else
@@ -46,5 +50,11 @@
 disabled_plugins += print-notifications
 endif
 
+if WITH_WACOM
+enabled_plugins += wacom
+else
+disabled_plugins += wacom
+endif
+
 SUBDIRS = common $(enabled_plugins)
 DIST_SUBDIRS = $(SUBDIRS) $(disabled_plugins)
