From 4a1fe09bfdff516e28aab23eea102984fedb429a Mon Sep 17 00:00:00 2001
From: Christian Ruppert <idl0r@gentoo.org>
Date: Wed, 7 Sep 2011 14:39:10 +0200
Subject: [PATCH] Add missing quotes

---
 bin/euse |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/bin/euse b/bin/euse
index bedba35..febe732 100755
--- a/bin/euse
+++ b/bin/euse
@@ -275,21 +275,21 @@ get_useflags() {
 	# backup portdir so get_portdir() doesn't give false results later
 	portdir_backup="${PORTDIR}"
 	
-	ACTIVE_FLAGS[0]="$(reduce_incrementals ${USE})"
+	ACTIVE_FLAGS[0]="$(reduce_incrementals "${USE}")"
 	USE=""
 	for x in $(get_all_make_conf); do
 		source "${x}"
-		ACTIVE_FLAGS[1]="$(reduce_incrementals ${ACTIVE_FLAGS[1]} ${USE})"
+		ACTIVE_FLAGS[1]="$(reduce_incrementals "${ACTIVE_FLAGS[1]}" "${USE}")"
 	done
 	USE=""
 	for x in $(get_all_make_defaults); do
 		source "${x}"
 		ACTIVE_FLAGS[2]="${ACTIVE_FLAGS[2]} ${USE}"
 	done
-	ACTIVE_FLAGS[2]="$(reduce_incrementals ${ACTIVE_FLAGS[2]})"
+	ACTIVE_FLAGS[2]="$(reduce_incrementals "${ACTIVE_FLAGS[2]}")"
 	USE=""
 	source "${MAKE_GLOBALS_PATH}"
-	ACTIVE_FLAGS[3]="$(reduce_incrementals ${USE})"
+	ACTIVE_FLAGS[3]="$(reduce_incrementals "${USE}")"
 
 	# restore saved env variables
 	USE="${ACTIVE_FLAGS[0]}"
-- 
1.7.3.4

