From 92e6be628307b0643307202ba9994cab9a4dd439 Mon Sep 17 00:00:00 2001
From: dol-sen <brian.dolbec@gmail.com>
Date: Thu, 25 Aug 2011 00:04:47 -0700
Subject: [PATCH] fix bug 380573.

---
 pym/gentoolkit/equery/meta.py |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/pym/gentoolkit/equery/meta.py b/pym/gentoolkit/equery/meta.py
index 01e090a..f617cfa 100644
--- a/pym/gentoolkit/equery/meta.py
+++ b/pym/gentoolkit/equery/meta.py
@@ -22,7 +22,7 @@ from functools import partial
 
 import gentoolkit.pprinter as pp
 from gentoolkit import errors
-from gentoolkit import keyword
+from gentoolkit.keyword import Keyword
 from gentoolkit.equery import format_options, mod_usage, CONFIG
 from gentoolkit.helpers import print_sequence, print_file
 from gentoolkit.textwrap_ import TextWrapper
@@ -217,7 +217,7 @@ def format_keywords(keywords):
 
 	result = []
 
-	for kw in sorted(keywords, keyword.compare_strs):
+	for kw in sorted(keywords, key=Keyword):
 		if kw.startswith('-'):
 			# arch masked
 			kw = pp.keyword(kw, stable=False, hard_masked=True)
-- 
1.7.3.4

