From 0382df255affced93b59322611a58bf98aec6c64 Mon Sep 17 00:00:00 2001
From: Paul Varner <fuzzyray@gentoo.org>
Date: Fri, 15 Jul 2011 14:59:23 -0500
Subject: [PATCH] Do not index ChangeLog entries with an invalid CPV. (Bug 375293)

---
 pym/gentoolkit/helpers.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/pym/gentoolkit/helpers.py b/pym/gentoolkit/helpers.py
index 225a198..a0b29ab 100644
--- a/pym/gentoolkit/helpers.py
+++ b/pym/gentoolkit/helpers.py
@@ -175,7 +175,7 @@ class ChangeLog(object):
 			if not pkg_name.strip():
 				continue
 			try:
-				entry_ver = CPV(pkg_name)
+				entry_ver = CPV(pkg_name, validate=True)
 			except errors.GentoolkitInvalidCPV:
 				if self.invalid_entry_is_fatal:
 					raise ValueError(entry_ver)
-- 
1.7.3.4

