--- build/basics.rb	2014-01-26 13:41:56.879695794 +0700
+++ build/basics.rb	2014-01-26 13:46:27.116093311 +0700
@@ -187,6 +187,9 @@
 	compiler_flag_option('EXTRA_CXX_LDFLAGS')
 
 
+EXTRA_LDFLAGS = "#{ENV['LDFLAGS']}"
+
+
 AGENT_OUTPUT_DIR          = string_option('AGENT_OUTPUT_DIR', OUTPUT_DIR + "support-binaries") + "/"
 COMMON_OUTPUT_DIR         = string_option('COMMON_OUTPUT_DIR', OUTPUT_DIR + "common") + "/"
 APACHE2_OUTPUT_DIR        = string_option('APACHE2_OUTPUT_DIR', OUTPUT_DIR + "apache2") + "/"
