--- ext/nginx/ContentHandler.c	2011-10-19 22:44:40.000000000 +0400
+++ ext/nginx/ContentHandler.c	2011-10-30 00:58:06.700129856 +0400
@@ -761,6 +761,7 @@
     }
 
     u = r->upstream;
+    u->headers_in.content_length_n = r->headers_out.content_length_n;
 
     if (rc == NGX_HTTP_SCGI_PARSE_NO_HEADER) {
         ngx_log_error(NGX_LOG_ERR, r->connection->log, 0,
@@ -824,6 +825,7 @@
     } state;
 
     u = r->upstream;
+    u->headers_in.content_length_n = r->headers_out.content_length_n;
 
     state = r->state;
 
@@ -1363,6 +1364,7 @@
         return NGX_HTTP_INTERNAL_SERVER_ERROR;
     }
     u = r->upstream;
+    u->headers_in.content_length_n = r->headers_out.content_length_n;
 
     u->schema = pp_schema_string;
     u->output.tag = (ngx_buf_tag_t) &ngx_http_passenger_module;
