diff -Naur zsh-completion-20080310.old/_eix zsh-completion-20080310/_eix
--- zsh-completion-20080310.old/_eix	2008-03-10 14:25:10.000000000 +0100
+++ zsh-completion-20080310/_eix	2009-12-04 01:15:05.000000000 +0100
@@ -19,6 +19,14 @@
     '--dump[dump variables to stdout]'
     '--dump-defaults[dump default values of variables]'
     '--print[print the expanded value of a variable]'
+    '--print-all-useflags[print all IUSE words used in some version]'
+    '--print-all-keywords[print all KEYWORDS used in some version]'
+    '--print-all-slots[print all SLOT strings used in some version]'
+    '--print-all-provides[print all PROVIDE strings used in some package]'
+    '--print-all-licenses[print all LICENSE strings used in some package]'
+    '--print-world-sets[print the world sets]'
+    '--print-overlay-path[print the path of specified overlay]'
+    '--print-overlay-label[print label of specified overlay]'
 )
 
 global_special=(
@@ -55,6 +63,12 @@
     '--testing[Match packages with a testing or stable version]'
     '--non-masked[Match packages with a non-masked version]'
     '--system[Match system packages]'
+    '--installed-unstable[Match packages with a non-stable installed version]'
+    '--installed-testing[Match packages with a testing installed version]'
+    '--installed-masked[Match packages with a masked installed version]'
+    '--world[Match world packages]'
+    '--world-all[Match packages of world or of a world set]'
+    '--world-set[Match packages of a world set]'
     {'(--overlay)-O','(-O)--overlay'}'[Match packages from overlays]'
     '--in-overlay[OVERLAY Match packages from OVERLAY]'
     '--only-in-overlay[OVERLAY Match packages only in OVERLAY]'
@@ -62,8 +76,19 @@
     '--installed-from-overlay[OVERLAY Packages installed from OVERLAY]'
     '--installed-in-some-overlay[Packages with an installed version provided by some overlay]'
     '--installed-in-overlay[OVERLAY Packages with an installed version provided from OVERLAY]'
-    '--fetch[Match packages with a fetch restriction]'
-    '--mirror[Match packages with a mirror restriction]'
+    '--restrict-fetch[Match packages with a fetch restriction]'
+    '--restrict-mirror[Match packages with a mirror restriction]'
+    '--restrict-primaryuri[Match packages with RESTRICT=primaryuri]'
+    '--restrict-binchecks[Match packages with RESTRICT=binchecks]'
+    '--restrict-strip[Match packages with RESTRICT=strip]'
+    '--restrict-test[Match packages with RESTRICT=test]'
+    '--restrict-userpriv[Match packages with RESTRICT=userpriv]'
+    '--restrict-installsources[Match packages with RESTRICT=installsources]'
+    '--restrict-bindist[Match packages with RESTRICT=bindist]'
+    '--properties-interactive[Match packages with PROPERTIES=interactive]'
+    '--properties-live[Match packages with PROPERTIES=live]'
+    '--properties-virtual[ Match packages with PROPERTIES=virtual]'
+    '--properties-set[Match packages with PROPERTIES=set]'
     {'(--test-obsolete)-T','(-T)--test-obsolete'}'[Match packages with obsolete entries in /etc/portage/package.*]'
     {'(--not)-!','(-!)--not'}'[(toggle) Invert the expression]'
     {'(--pipe)-\\|','(-\\|)--pipe'}'[Use input from pipe of emerge -pv]'
@@ -85,6 +110,9 @@
 local_type_of_pattern=(
     {'(--regex)-r','(-r)--regex'}'[Pattern is a regexp (default)]'
     {'(--exact)-e','(-e)--exact'}'[Pattern is the exact string]'
+    {'(--substring)-z','(-z)--substring'}'[Pattern is a substring]'
+    {'(--begin)-b','(-b)--begin'}'[Pattern is the beginning of the string]'
+    '--end[Pattern is the end       of the string]'
     {'(--pattern)-p','(-p)--pattern'}'[Pattern is a wildcards-pattern]'
     {'(--fuzzy)-f','(-f)--fuzzy'}'[Use fuzzy-search with a max. levenshtein-distance m (default: 2)]'
 )
