From ea688b0ae9669ba28f5b557851ebe43bb20ad2f6 Mon Sep 17 00:00:00 2001
From: "Gavin Pryke (gavlee)" <gavinlee303@googlemail.com>
Date: Wed, 13 Mar 2013 09:59:28 +0000
Subject: [PATCH 1/6] fix creation of packaging install dir

---
 SConstruct | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index 8c70a33..91d81fc 100644
--- a/SConstruct
+++ b/SConstruct
@@ -123,7 +123,7 @@ def setup_build_options( env ) :
 	
 	# global options
 	opts.Add( PathVariable( 'prefix', 'Install location for CLAM', '/usr/local'))
-	opts.Add( PathVariable( 'prefix_for_packaging', 'Install location when packaging (just for .deb creation)', '.'))
+	opts.Add( PathVariable( 'prefix_for_packaging', 'Install location when packaging (just for .deb creation)', '.', PathVariable.PathIsDirCreate))
 	if isWindowsPlatform :
 		opts.Add( PathVariable( 'sandbox_path', 'The working directory in mingw where the external libraries are placed', '' ) )
 	opts.Add( BoolVariable( 'release', 'Build CLAM with optimizations and stripping debug symbols', 'no'))
-- 
1.7.12.4

