From 39eafabf7e3fc41e658fc1ee26926d4da47bbdce Mon Sep 17 00:00:00 2001
From: "Gavin Pryke (gavlee)" <gavinlee303@googlemail.com>
Date: Wed, 13 Mar 2013 10:43:46 +0000
Subject: [PATCH 3/6] honour toolchain environment

---
 SConstruct | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index 91d81fc..f762227 100644
--- a/SConstruct
+++ b/SConstruct
@@ -230,12 +230,25 @@ if crosscompiling :
 clam_env.Tool('textfile',toolpath=['scons/sconstools'])
 clam_env.Tool('pc',toolpath=['scons/sconstools'])
 
+# from http://dev.gentoo.org/~vapier/scons-blows.txt
+if os.environ.has_key('CC'):
+	clam_env['CC'] = os.environ['CC']
+if os.environ.has_key('CFLAGS'):
+	clam_env['CCFLAGS'] += SCons.Util.CLVar(os.environ['CFLAGS'])
+if os.environ.has_key('CXX'):
+	clam_env['CXX'] = os.environ['CXX']
+if os.environ.has_key('CXXFLAGS'):
+	clam_env['CXXFLAGS'] += SCons.Util.CLVar(os.environ['CXXFLAGS'])
+if os.environ.has_key('CPPFLAGS'):
+	clam_env['CCFLAGS'] += SCons.Util.CLVar(os.environ['CPPFLAGS'])
+if os.environ.has_key('LDFLAGS'):
+	clam_env['LINKFLAGS'] += SCons.Util.CLVar(os.environ['LDFLAGS'])
+
 if clam_env['distcc_hosts'] :
 	clam_env['CXX'] = 'distcc '+clam_env['CXX']
 	clam_env['ENV']['DISTCC_HOSTS'] = clam_env['distcc_hosts']
 	SetOption('num_jobs', len( clam_env['distcc_hosts'].split() ))
 
-
 sys.path.append('scons/sconstools')
 import versionInfo
 version, fullVersion = versionInfo.versionFromLocalInfo("CLAM", "CHANGES")
-- 
1.7.12.4

