From 7ab2ae6dd0b38bd47526cdb48978d7c08b183634 Mon Sep 17 00:00:00 2001
From: "Gavin Pryke (gavlee)" <gavinlee303@googlemail.com>
Date: Thu, 14 Mar 2013 17:25:54 +0000
Subject: [PATCH 6/6] correct prefix in .pc files building modules

---
 scons/sconstools/clam.py | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/scons/sconstools/clam.py b/scons/sconstools/clam.py
index 4e78e8e..9f5dc82 100644
--- a/scons/sconstools/clam.py
+++ b/scons/sconstools/clam.py
@@ -71,12 +71,20 @@ def ClamModule(env, moduleName, version,
 		libdir = env['libdir']
 	elif not env['libdir']:
 		libdir = 'lib'
+
+	# try to set correct prefix directories in pkg-config files when
+        # building modules (networkeditor), this doesn't apply to clam lib
+        # and clam_prefix needs to be specified to build networkeditor anyway
+	if env['clam_prefix']:
+		prefix = env['clam_prefix']
+	elif not env['clam_prefix']:
+		prefix = env['prefix']
 	
 	# pkg-config file
 	pcfile = env.PkgConfigFile(
 		package = libraryName,
 		version = version,
-                prefix = env['prefix'],
+                prefix = prefix,
 		libdir = libdir,
 		description = description,
 		url = url,
-- 
1.7.12.4

