Description: support Xerces-c 3
Origin: upstream
        svn diff -c15306

Index: clam-1.4.0/src/Flow/Ports/OutPort.hxx
===================================================================
--- clam-1.4.0.orig/src/Flow/Ports/OutPort.hxx	2014-02-15 16:36:01.654895560 -0500
+++ clam-1.4.0/src/Flow/Ports/OutPort.hxx	2014-02-15 17:15:31.955139980 -0500
@@ -28,6 +28,7 @@
 #include <string>
 #include "InPort.hxx"
 #include "InPortPublisher.hxx"
+#include "TypeInfo.hxx"
 
 namespace CLAM
 {
