Description: fix FTBFS with TIME_UTC difinition
Origin: upstream
        svn diff -r 15397:15398
Bug-Debian: http://bugs.debian.org/701257

Index: clam-1.4.0/src/System/Threads/xtime.hxx
===================================================================
--- clam-1.4.0.orig/src/System/Threads/xtime.hxx	2014-02-15 17:00:47.975907239 -0500
+++ clam-1.4.0/src/System/Threads/xtime.hxx	2014-02-15 17:02:27.892272416 -0500
@@ -30,6 +30,11 @@
 #endif
 #include <pthread.h>
 
+// KLUDGE: C11 defines TIME_UTC as macro in time.h
+#ifdef TIME_UTC
+#undef TIME_UTC
+#endif
+
 namespace CLAM
 {
 
