--- screen-4.0.2/display.c	2003-12-05 05:45:41.000000000 -0800
+++ screen-4.0.2.abez/display.c	2005-01-27 00:31:35.000000000 -0800
@@ -23,6 +23,7 @@
 
 #include <sys/types.h>
 #include <signal.h>
+#include <setjmp.h>
 #include <fcntl.h>
 #ifndef sun
 # include <sys/ioctl.h>
@@ -307,6 +308,17 @@ struct mode *Mode;
   return display;
 }
 
+static jmp_buf alarm_jump;
+
+static sigret_t 
+FlushSigAlarm SIGDEFARG
+{
+#ifdef DEBUG
+    debug("FlushSigAlarm\n");
+#endif
+    siglongjmp (alarm_jump, 1);
+}
+
 
 void
 FreeDisplay()
@@ -328,7 +340,12 @@ FreeDisplay()
       Flush();
       if (!display)
 	return;
-      SetTTY(D_userfd, &D_OldMode);
+      if (sigsetjmp(alarm_jump,1)==0) {
+        signal(SIGALRM, FlushSigAlarm);
+        alarm(10);
+        SetTTY(D_userfd, &D_OldMode);
+        alarm(0);
+      }
       fcntl(D_userfd, F_SETFL, 0);
     }
   freetty();
@@ -3132,6 +3149,7 @@ int n;
     AddChar(' ');
 }
 
+
 void
 Flush()
 {
@@ -3156,7 +3174,15 @@ Flush()
   while (l)
     {
       register int wr;
-      wr = write(D_userfd, p, l);
+      if (sigsetjmp (alarm_jump,1) == 0) {
+        signal(SIGALRM, FlushSigAlarm);
+	alarm(10);
+	wr = write(D_userfd, p, l);
+	alarm(0);
+      } else {
+    	debug1("Warning: We spent too long in flush: %d\n", errno);
+	return;
+      }
       if (wr <= 0) 
 	{
 	  if (errno == EINTR) 
