diff --git a/display.h b/display.h
index ef99954..b5a0420 100644
--- a/display.h
+++ b/display.h
@@ -85,7 +85,7 @@ struct display
   struct win *d_other;		/* pointer to other window */
   int   d_nonblock;		/* -1 don't block if obufmax reached */
 				/* >0: block after nonblock secs */
-  char  d_termname[20 + 1];	/* $TERM */
+  char  d_termname[30 + 1];	/* $TERM */
   char	*d_tentry;		/* buffer for tgetstr */
   char	d_tcinited;		/* termcap inited flag */
   int	d_width, d_height;	/* width/height of the screen */
diff --git a/screen.h b/screen.h
index 4f9f354..9d45a2e 100644
--- a/screen.h
+++ b/screen.h
@@ -202,7 +202,7 @@ struct msg
 	  char preselect[20];
 	  int esc;		/* his new escape character unless -1 */
 	  int meta_esc;		/* his new meta esc character unless -1 */
-	  char envterm[20 + 1];	/* terminal type */
+	  char envterm[30 + 1];	/* terminal type */
 	  int encoding;		/* encoding of display */
 	}
       attach;
