diff -Nuar -Nuar vzctl-4.7.2/etc/vz.conf.in vzctl-4.7.2-fix/etc/vz.conf.in
--- vzctl-4.7.2/etc/vz.conf.in	2014-05-02 20:54:34.000000000 +0000
+++ vzctl-4.7.2-fix/etc/vz.conf.in	2014-09-16 09:48:32.136912815 +0000
@@ -44,7 +44,7 @@
 DEF_OSTEMPLATE="centos-6-x86"
 NAMESERVER=inherit # Copy from host system's /etc/resolv.conf
 ## Filesystem layout for new CTs: either simfs or ploop
-VE_LAYOUT=ploop
+VE_LAYOUT=simfs
 
 # User namespace configuration
 LOCAL_UID=100000
