diff -Nuar -Nuar eix-0.30.3/src/eix-sync.in eix-0.30.3-fi/src/eix-sync.in
--- eix-0.30.3/src/eix-sync.in	2014-08-22 15:22:42.000000000 +0000
+++ eix-0.30.3-fi/src/eix-sync.in	2014-09-15 05:08:55.027434267 +0000
@@ -43,13 +43,7 @@
      (useful to ignore ${eixsyncconf} options).
 -a   Always assume that some tree/overlay changes
      (set have_changed=: in a !-hook to force this)
--s [USER@]SERVER[:DIR] Sync via rsync from SERVER.
-     USER defaults to current user and DIR defaults to PORTDIR.
-     An empty SERVER cancels this option. This option implies -0
 -0   No eix --sync. This implies -a
--2 [USER@]CLIENT[:DIR] Sync via rsync *to* CLIENT after successful syncing;
-     you should later call eix-sync -u locally on CLIENT.
-     If you already have synced you might want to combine this option with -uU.
 -U   Do not touch the database and omit the hooks after eix-update (@ entries)
      and do not show differences. This option implies -R
 -u   Update database only and show differences. This is equivalent to -0l@As ""
@@ -68,8 +62,6 @@
 -t   Use temporary file to save the current database (excluding eix-diff later)
 -T   Do not measure time
 -q   Be quiet (close stdout)
--w   Run emerge-webrsync instead of emerge --sync.
--W   Run emerge-delta-webrsync instead of emerge --sync.
 -c CMD Run CMD instead of emerge --sync.
 -C OPT Add OPT to the emerge --sync command (or whatever is used instead).
        This option can be used accumulatively.
@@ -279,9 +271,7 @@
 do	case ${opt} in
 	i)	DefaultOpts;;
 	a)	have_changed=:;;
-	s)	server=${OPTARG};;
 	0)	skip_sync=:;;
-	2)	client=${OPTARG};;
 	U)	doupdate=false;;
 	u)	nolayman=:
 		nohooks=:
@@ -299,8 +289,6 @@
 	T)	measure_time=false;;
 	q)	quiet=:;;
 	L)	Push layman_opt "${OPTARG}";;
-	w)	Push -c emergecmd 'emerge-webrsync';;
-	W)	Push -c emergecmd 'emerge-delta-webrsync';;
 	c)	Push -c emergecmd "${OPTARG}";;
 	C)	Push emergecmd "${OPTARG}";;
 	o)	Push updatecmd "${OPTARG}";;
@@ -456,41 +444,6 @@
 	trap ExitAll EXIT HUP INT TERM
 }
 
-preprsync=false
-PrepRsync() {
-	GetPortdir
-	hostdir=${1#*:}
-	if [ -n "${hostdir:++}" ] && [ "${hostdir}" != "${1}" ]
-	then	hostdir=${1}
-	else	hostdir="${1%%:*}:${local_portdir}"
-	fi
-	hostdir="${hostdir%/}/"
-	${preprsync} || [ -n "${rsync_opts:++}" ] && return
-	ReadVar portage_rsync_opts PORTAGE_RSYNC_OPTS || \
-		portage_rsync_opts='--recursive --links --safe-links --perms --times --compress --force --whole-file --delete --stats --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages'
-	ReadVar portage_rsync_extra_opts PORTAGE_RSYNC_EXTRA_OPTS
-	case ${-} in
-	*f*)
-		eval "set -- ${before_rsync}";;
-	*)
-		set -f
-		eval "set -- ${before_rsync}"
-		set +f;;
-	esac
-	[ ${#} -eq 0 ] || StatusInfo "`gettext 'Running ~-hooks'`"
-	for curr_cmd
-	do	! ${printcmd} || PrintCmd 'eval "$(eval "'"${curr_cmd}"'")"'
-		${dryrun} && continue
-		if c=`eval "${curr_cmd}"`
-		then	eval "${c}" || WarnOrDie "`eval_gettext \
-				'${c} (output of ${curr_cmd}) failed'`"
-		else	WarnOrDie "`eval_gettext '${curr_cmd} failed'`"
-		fi
-	done
-	rsync_opts="${portage_rsync_opts} ${portage_rsync_extra_opts} --exclude=/.unionfs"
-	preprsync=:
-}
-
 ClearCache() {
 	${clearcache} || return 0
 	# Cleaning old cache
@@ -510,25 +463,10 @@
 }
 
 CallEmergeSync() {
-	if [ -n "${server:++}" ]
-	then	PrepRsync "${server}"
-		MyRunCommand "rsyncing from ${hostdir}" -t sync \
-		rsync ${rsync_opts} -- "${hostdir}" "${local_portdir}" || \
-			die "`eval_gettext 'Could not rsync from ${hostdir}'`"
-		return
-	fi
 	${skip_sync} && return
 	MyVarCommand "${emergecmd}" -t sync
 }
 
-CallSyncClient() {
-	[ -z "${client:++}" ] && return
-	PrepRsync "${client}"
-	MyRunCommand "`eval_gettext 'rsyncing to ${hostdir}'`" -t client \
-		rsync ${rsync_opts} -- "${local_portdir}" "${hostdir}" || \
-			die "`eval_gettext 'Could not rsync to ${hostdir}'`"
-}
-
 CallEmergeMetadata() {
 	${doupdate} && ${metadata} || return 0
 	MyVarCommand 'emerge --metadata' -t metadata
@@ -634,7 +572,6 @@
 	[ "${previous_timestamp}" != "${current_timestamp}" ] && have_changed=:
 	${have_changed} || NotChanged
 	${synchooks} && DoHook "${after_sync}" "`gettext 'Running @-hooks'`"
-	CallSyncClient
 	CallEmergeMetadata
 	CopyPrevious
 	UpdateDiff
