diff --git a/a/src/eix-sync.in b/b/src/eix-sync.in
index c8a46d6..2275415 100755
--- a/a/src/eix-sync.in
+++ b/b/src/eix-sync.in
@@ -43,13 +43,7 @@ The following options are available:
 -d   Only show differences to the previously saved database and exit.
      Except for executing the !! lines in ${eixsyncconf}
      this is the same as eix-diff ${eixprevious}.
--s [USER@]SERVER[:DIR] Sync via rsync from SERVER.
-     USER defaults to current user and DIR defaults to PORTDIR.
-     An empty SERVER cancels this option. This option implies -0
 -0   No eix --sync
--2 [USER@]CLIENT[:DIR] Sync via rsync *to* CLIENT after successfull syncing;
-     you should later call eix-sync -u locally on CLIENT.
-     If you already have synced you might want to combine this option with -uU.
 -U   Do not touch the database and omit the hooks after eix-update (@ entries)
      and do not show differences. This option implies -R
 -u   Update database only and show differences. This is equivalent to -0l@s ""
@@ -67,8 +61,6 @@ The following options are available:
 -t   Use temporary file to save the current database (-d will not be possible).
 -T   Do not measure time
 -q   Be quiet (close stdout)
--w   Run emerge-webrsync instead of emerge --sync.
--W   Run emerge-delta-webrsync instead of emerge --sync.
 -c CMD Run CMD instead of emerge --sync.
 -C OPT Add OPT to the emerge --sync command (or whatever is used instead).
        This option can be used accumulatively.
@@ -215,8 +207,6 @@ DefaultOpts() {
 	measure_time=:
 	metadata=false
 	skip_sync=false
-	server=''
-	client=''
 	doupdate=:
 	synchooks=:
 	test_for_root=:
@@ -236,14 +226,11 @@ while getopts 'ids:02:UuFl@SMNtTqwWL:c:C:o:rRemHh?' opt
 do	case "${opt}" in
 	i)	DefaultOpts;;
 	d)	diffonly=:;;
-	s)	server="${OPTARG}";;
 	0)	skip_sync=:;;
-	2)	client="${OPTARG}";;
 	U)	doupdate=false;;
 	u)	nolayman=:
 		nohooks=:
-		skip_sync=:
-		server='';;
+		skip_sync=:;;
 	F)	FATAL_HOOKS=':';;
 	l)	nolayman=:;;
 	'@')	nohooks=:;;
@@ -270,8 +257,6 @@ do	case "${opt}" in
 done
 opt=''
 ${diffonly} && DiffOnly
-[ -z "${server}" ] || skip_sync=:
-
 ${measure_time} || time_begin=''
 measure_time=false
 case "${time_begin}" in
@@ -397,38 +382,6 @@ MakeTempFile() {
 	trap ExitAll EXIT HUP INT TERM
 }
 
-preprsync=false
-PrepRsync() {
-	GetPortdir
-	hostdir="${1#*:}"
-	if [ -n "${hostdir}" ] && [ "${hostdir}" != "${1}" ]
-	then	hostdir="${1}"
-	else	hostdir="${1%%:*}:${local_portdir}"
-	fi
-	hostdir="${hostdir%/}/"
-	${preprsync} || [ -n "${rsync_opts}" ] && return
-	ReadVar portage_rsync_opts PORTAGE_RSYNC_OPTS || \
-		portage_rsync_opts="--recursive --links --safe-links --perms --times --compress --force --whole-file --delete --stats --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages"
-	ReadVar portage_rsync_extra_opts PORTAGE_RSYNC_EXTRA_OPTS
-	case "${-}" in
-	*f*)
-		eval "set -- ${before_rsync}";;
-	*)
-		set -f
-		eval "set -- ${before_rsync}"
-		set +f;;
-	esac
-	for curr_cmd
-	do	if c=`eval "${curr_cmd}"`
-		then	eval "${c}" || WarnOrDie "`eval_gettext \
-				'${c} (output of ${curr_cmd}) failed'`"
-		else	WarnOrDie "`eval_gettext '${curr_cmd} failed'`"
-		fi
-	done
-	rsync_opts="${portage_rsync_opts} ${portage_rsync_extra_opts} --exclude=/.unionfs"
-	preprsync=:
-}
-
 ClearCache() {
 	${clearcache} || return 0
 	# Cleaning old cache
@@ -449,28 +402,11 @@ ClearCache() {
 }
 
 CallEmergeSync() {
-	if [ -n "${server}" ]
-	then	${emulatemode} && return
-		PrepRsync "${server}"
-		MyRunCommand "rsyncing from ${hostdir}" -t sync \
-		rsync ${rsync_opts} -- "${hostdir}" "${local_portdir}" || \
-			die "`eval_gettext 'Could not rsync from ${hostdir}'`"
-		return
-	fi
 	${skip_sync} && return
 	RootTest
 	MyVarCommand "${emergecmd}" -t sync
 }
 
-CallSyncClient() {
-	${emulatemode} && return
-	[ -z "${client}" ] && return
-	PrepRsync "${client}"
-	MyRunCommand "`eval_gettext 'rsyncing to ${hostdir}'`" -t client \
-		rsync ${rsync_opts} -- "${local_portdir}" "${hostdir}" || \
-			die "`eval_gettext 'Could not rsync to ${hostdir}'`"
-}
-
 CallEmergeMetadata() {
 	${doupdate} && ${metadata} || return 0
 	RootTest
@@ -543,7 +479,6 @@ MainSync() {
 	ExecuteConfig 'sync'
 	CallEmergeSync
 	${synchooks} && DoHook "${after_sync}"
-	CallSyncClient
 	CallEmergeMetadata
 	UpdateDiff
 }
