diff -ur eix-0.23.6-a/src/eix-sync.in eix-0.23.6-b/src/eix-sync.in
--- eix-0.23.6-a/src/eix-sync.in	2012-01-17 11:52:44.765023945 -0500
+++ eix-0.23.6-b/src/eix-sync.in	2012-01-17 11:53:45.193024991 -0500
@@ -449,7 +449,7 @@
 	MakeTempFile
 	cp -- "${eixcache}" "${tmpfile}" || die "`eval_gettext \
 		'Could not copy database to temporary file ${tmpfile}'`"
-	umask "${previouus_umask}"
+	umask "${previous_umask}"
 }
 
 MovePrevious() {
