diff -urN eix-0.22.8-orig//src/eix-sync.in eix-0.22.8/src/eix-sync.in
--- eix-0.22.8-orig//src/eix-sync.in	2011-03-15 12:48:39.271002405 -0400
+++ eix-0.22.8/src/eix-sync.in	2011-03-15 13:05:34.823002844 -0400
@@ -8,6 +8,8 @@
 #
 # This script calls emerge --sync and shows the differences.
 # See the eix manpage for details. (@PACKAGE_STRING@).
+#
+# Patched by Jean-Francis Roy for Funtoo.org to prevent rsync.
 
 time_begin="`date +%s 2>/dev/null`" || time_begin=''
 
@@ -43,13 +45,7 @@
 -d   Only show differences to the previously saved database and exit.
      Except for executing the !! lines in ${eixsyncconf}
      this is the same as eix-diff ${eixprevious}.
--s [USER@]SERVER[:DIR] Sync via rsync from SERVER.
-     USER defaults to current user and DIR defaults to PORTDIR.
-     An empty SERVER cancels this option. This option implies -0
 -0   No eix --sync
--2 [USER@]CLIENT[:DIR] Sync via rsync *to* CLIENT after successfull syncing;
-     you should later call eix-sync -u locally on CLIENT.
-     If you already have synced you might want to combine this option with -uU.
 -U   Do not touch the database and omit the hooks after eix-update (@ entries)
      and do not show differences. This option implies -R
 -u   Update database only and show differences. This is equivalent to -0l@s ""
@@ -65,8 +61,6 @@
 -t   Use temporary file to save the current database (-d will not be possible).
 -T   Do not measure time
 -q   Be quiet (close stdout)
--w   Run emerge-webrsync instead of emerge --sync.
--W   Run emerge-delta-webrsync instead of emerge --sync.
 -c CMD Run CMD instead of emerge --sync.
 -C OPT Add OPT to the emerge --sync command (or whatever is used instead).
        This option can be used accumulatively.
@@ -173,8 +167,6 @@
 	measure_time=:
 	metadata=false
 	skip_sync=false
-	server=''
-	client=''
 	doupdate=:
 	synchooks=:
 	test_for_root=:
@@ -191,11 +183,9 @@
 do	case "${opt}" in
 		i) DefaultOpts;;
 		d) diffonly=:;;
-		s) server="${OPTARG}";;
 		0) skip_sync=:;;
-		2) client="${OPTARG}";;
 		U) doupdate=false;;
-		u) nolayman=:; nohooks=:; skip_sync=:; server='';;
+		u) nolayman=:; nohooks=:; skip_sync=: ;;
 		l) nolayman=:;;
 		@) nohooks=:;;
 		S) synchooks=false;;
@@ -205,8 +195,6 @@
 		T) measure_time=false;;
 		q) quiet=:;;
 		L) Push layman_opt "${OPTARG}";;
-		w) Push -c emergecmd 'emerge-webrsync';;
-		W) Push -c emergecmd 'emerge-delta-webrsync';;
 		c) Push -c emergecmd "${OPTARG}";;
 		C) Push emergecmd "${OPTARG}";;
 		o) Push updatecmd "${OPTARG}";;
@@ -219,7 +207,6 @@
 done
 opt=''
 ${diffonly} && DiffOnly
-[ -z "${server}" ] || skip_sync=:
 
 ${measure_time} || time_begin=''
 measure_time=false
@@ -319,28 +306,6 @@
 	trap ExitAll EXIT HUP INT TERM
 }
 
-preprsync=false
-PrepRsync() {
-	GetPortdir
-	hostdir="${1#*:}"
-	if [ -n "${hostdir}" ] && [ "${hostdir}" != "${1}" ]
-	then	hostdir="${1}"
-	else	hostdir="${1%%:*}:${local_portdir}"
-	fi
-	hostdir="${hostdir%/}/"
-	${preprsync} && return
-	portage_rsync_opts="`portageq envvar PORTAGE_RSYNC_OPTS`"
-	portage_rsync_extra_opts="`portageq envvar PORTAGE_RSYNC_EXTRA_OPTS`"
-	eval "set -- ${before_rsync}"
-	for curr_cmd
-	do	c=`eval "${curr_cmd}"` || \
-			die "`eval_gettext '${curr_cmd} failed'`"
-		eval "${c}"
-	done
-	rsync_opts="${portage_rsync_opts} ${portage_rsync_extra_opts}"
-	preprsync=:
-}
-
 ClearCache() {
 	${clearcache} || return 0
 	# Cleaning old cache
@@ -354,14 +319,6 @@
 }
 
 CallEmergeSync() {
-	if [ -n "${server}" ]
-	then	${emulatemode} && return
-		PrepRsync "${server}"
-		MyRunCommand "rsyncing from ${hostdir}" -t sync \
-		rsync ${rsync_opts} "${hostdir}" "${local_portdir}" || \
-			die "`eval_gettext 'Could not rsync from ${hostdir}'`"
-		return
-	fi
 	${skip_sync} && return
 	RootTest
 	eval "set -- ${emergecmd}"
@@ -370,15 +327,6 @@
 		|| die "`eval_gettext '${runarg} failed'`"
 }
 
-CallSyncClient() {
-	${emulatemode} && return
-	[ -z "${client}" ] && return
-	PrepRsync "${client}"
-	MyRunCommand "`eval_gettext 'rsyncing to ${hostdir}'`" -t client \
-		rsync ${rsync_opts} "${local_portdir}" "${hostdir}" || \
-			die "`eval_gettext 'Could not rsync to ${hostdir}'`"
-}
-
 TESTRESULT=''
 CallEmergeMetadata() {
 	${doupdate} && ${metadata} || return 0
@@ -464,7 +412,6 @@
 	ExecuteConfig 'sync'
 	CallEmergeSync
 	${synchooks} && DoHook "${after_sync}"
-	CallSyncClient
 	CallEmergeMetadata
 	UpdateDiff
 }
