--- setup.py.orig	2004-12-12 22:52:50.000000000 +0100
+++ setup.py	2010-07-02 23:56:17.983511538 +0200
@@ -13,9 +13,13 @@
     url = "http://www.home.unix-ag.org/arne/jack/",
 
     # Description of the modules and packages in the distribution
-    ext_modules = [ Extension('jack_cursesmodule',
-    ['cursesmodule/jack_cursesmodule.c'], libraries=["ncurses"],
-    extra_compile_args=["-Wno-strict-prototypes"]) ],
+    # we dont need jack_cursemodule any more because its included in python 
+    # since release 2.0
+    # see http://docs.python.org/whatsnew/2.0.html
+    #
+    # ext_modules = [ Extension('jack_cursesmodule',
+    # ['cursesmodule/jack_cursesmodule.c'], libraries=["ncurses"],
+    # extra_compile_args=["-Wno-strict-prototypes"]) ],
 
     py_modules = [ 'jack_CDTime', 'jack_TOC', 'jack_TOCentry', 'jack_argv',
     'jack_checkopts', 'jack_children', 'jack_config', 'jack_constants',
--- jack_t_curses.py.orig	2010-07-03 00:40:11.000000000 +0200
+++ jack_t_curses.py	2010-07-03 12:32:37.947527893 +0200
@@ -35,11 +35,14 @@
 try:
     from jack_curses import endwin, resizeterm, A_REVERSE, newwin, newpad, initscr, noecho, cbreak, echo, nocbreak
 except ImportError:
-    warning("jack_curses module not found, trying normal curses...")
+    # we dont need jack_cursemodule any more because its included in python 
+    # since release 2.0
+    # see http://docs.python.org/whatsnew/2.0.html
+    info("jack_curses module not found, using normal curses...")
     try:
-        from curses import endwin, A_REVERSE, newwin, newpad, initscr, noecho, cbreak, echo, nocbreak
-        def resizeterm(y, x):
-            pass
+        from curses import endwin, resizeterm, A_REVERSE, newwin, newpad, initscr, noecho, cbreak, echo, nocbreak
+        # def resizeterm(y, x):
+        #     pass
     except ImportError:
         print "curses module not found or too old, please install it (see README)"
 
