--- CMakeLists.txt.org	2015-02-28 19:48:35.421398254 +0300
+++ CMakeLists.txt	2015-02-28 19:49:39.051062881 +0300
@@ -31,6 +31,7 @@
 # Global compiler flags
 if(CMAKE_COMPILER_IS_GNUCC)
    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-multichar -Wall -Wno-unused-but-set-variable -fPIC")
+   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--no-as-needed")
 endif()
 
 add_definitions(-D_REENTRANT)
