diff -urN snortsam/makesnortsam.sh snortsam-patched/makesnortsam.sh
--- snortsam/makesnortsam.sh	2009-11-26 20:57:00.000000000 -0500
+++ snortsam-patched/makesnortsam.sh	2014-07-23 02:16:58.837236410 -0400
@@ -11,7 +11,7 @@
 #        Under Solaris, the OPSEC stuff is linked dynamically.
 #        On other platforms, statically.
 
-# *BSD only: uncomment to build with '-lc_r', default is '-lpthread'
+# *BSD only: uncomment to build with '-lc_r', default is '-pthread'
 #BSDTHREADLIB='-lc_r'
 
 # OpenBSD only: Default is now the new pf2 plugin.
@@ -41,13 +41,13 @@
 
 # OS specific flags
 LINUX_CFLAGS=
-LINUX_LDFLAGS="-lpthread"
+LINUX_LDFLAGS="-pthread"
 OBSD_CFLAGS="-DBSD ${PFPLUGIN}"
-OBSD_LDFLAGS="-lpthread"
+OBSD_LDFLAGS="-pthread"
 BSD_CFLAGS="-DBSD"
-BSD_LDFLAGS=${BSDTHREADLIB:-"-lpthread"}
+BSD_LDFLAGS=${BSDTHREADLIB:-"-pthread"}
 SUNOS_CFLAGS="-DSOLARIS"
-SUNOS_LDFLAGS="-lpthread -lnsl -lsocket -lresolv"
+SUNOS_LDFLAGS="-pthread -lnsl -lsocket -lresolv"
 
 # OPSEC 22 SDK
 OPSEC_CFLAGS="-Iopsec/include/opsec -Iopsec/include -DENABLE_OPSEC"
