--- Configure.orig	2006-02-03 16:15:00.000000000 -0500
+++ Configure	2006-02-03 16:25:04.000000000 -0500
@@ -1256,17 +1256,24 @@ locincpth="$locincpth /opt/gnu/include /
 : no include file wanted by default
 inclwanted=''
 
+: determine libdir name
+: Should be */lib64 on x86-64, sparc64, ppc64
+case `arch` in
+  x86_64) lib='lib64';;
+  *) lib='lib';;
+esac
+  
 groupstype=''
 libnames=''
 : change the next line if compiling for Xenix/286 on Xenix/386
 xlibpth='/usr/lib/386 /lib/386'
 : Possible local library directories to search.
-loclibpth="/usr/local/lib /opt/local/lib /usr/gnu/lib"
-loclibpth="$loclibpth /opt/gnu/lib /usr/GNU/lib /opt/GNU/lib"
+loclibpth="/usr/local/$lib /opt/local/$lib /usr/gnu/$lib"
+loclibpth="$loclibpth /opt/gnu/$lib /usr/GNU/$lib /opt/GNU/$lib"
 
 : general looking path for locating libraries
-glibpth="/lib /usr/lib $xlibpth"
-glibpth="$glibpth /usr/ccs/lib /usr/ucblib /usr/local/lib"
+glibpth="/$lib /usr/$lib $xlibpth"
+glibpth="$glibpth /usr/ccs/$lib /usr/ucblib /usr/local/$lib"
 test -f /usr/shlib/libc.so && glibpth="/usr/shlib $glibpth"
 test -f /shlib/libc.so     && glibpth="/shlib $glibpth"
 
@@ -5951,8 +5958,8 @@ fi
 : Reproduce behavior of 5.005 and earlier, maybe drop that in 5.7.
 case "$installstyle" in
 '')	case "$prefix" in
-		*perl*) dflt='lib';;
-		*) dflt='lib/perl5' ;;
+		*perl*) dflt='lib64';;
+		*) dflt='lib64/perl5' ;;
 	esac
 	;;
 *)	dflt="$installstyle" ;;
@@ -5968,8 +5975,8 @@ installstyle=$dflt
 : /opt/perl/lib/perl5... would be redundant.
 : The default "style" setting is made in installstyle.U
 case "$installstyle" in
-*lib/perl5*) set dflt privlib lib/$package/$version ;;
-*)	 set dflt privlib lib/$version ;;
+*lib64/perl5*) set dflt privlib lib64/$package/$version ;;
+*)	 set dflt privlib lib64/$version ;;
 esac
 eval $prefixit
 $cat <<EOM
@@ -6471,8 +6478,8 @@ siteprefixexp="$ansexp"
 prog=`echo $package | $sed 's/-*[0-9.]*$//'`
 case "$sitelib" in
 '') case "$installstyle" in
-	*lib/perl5*) dflt=$siteprefix/lib/$package/site_$prog/$version ;;
-	*)	 dflt=$siteprefix/lib/site_$prog/$version ;;
+	*lib64/perl5*) dflt=$siteprefix/lib64/$package/site_$prog/$version ;;
+	*)	 dflt=$siteprefix/lib64/site_$prog/$version ;;
 	esac
 	;;
 *)	dflt="$sitelib"
@@ -6598,8 +6605,8 @@ case "$vendorprefix" in
 	'')
 		prog=`echo $package | $sed 's/-*[0-9.]*$//'`
 		case "$installstyle" in
-		*lib/perl5*) dflt=$vendorprefix/lib/$package/vendor_$prog/$version ;;
-		*)	     dflt=$vendorprefix/lib/vendor_$prog/$version ;;
+		*lib64/perl5*) dflt=$vendorprefix/lib64/$package/vendor_$prog/$version ;;
+		*)	     dflt=$vendorprefix/lib64/vendor_$prog/$version ;;
 		esac
 		;;
 	*)	dflt="$vendorlib"
@@ -10556,9 +10563,9 @@ else
 		else
 			for net in net socket
 			do
-				if test -f /usr/lib/lib$net$_a; then
-					( ($nm $nm_opt /usr/lib/lib$net$_a | eval $nm_extract) ||  \
-					$ar t /usr/lib/lib$net$_a) 2>/dev/null >> libc.list
+				if test -f /usr/$lib/lib$net$_a; then
+					( ($nm $nm_opt /usr/$lib/lib$net$_a | eval $nm_extract) ||  \
+					$ar t /usr/$lib/lib$net$_a) 2>/dev/null >> libc.list
 					if $contains socket libc.list >/dev/null 2>&1; then
 						d_socket="$define"
 						socketlib="-l$net"
@@ -21677,6 +21684,7 @@ ldflags='$ldflags'
 ldflags_uselargefiles='$ldflags_uselargefiles'
 ldlibpthname='$ldlibpthname'
 less='$less'
+lib='$lib'
 lib_ext='$lib_ext'
 libc='$libc'
 libperl='$libperl'
