diff -Naur perl-5.8.8-orig/makedepend.SH perl-5.8.8/makedepend.SH
--- perl-5.8.8-orig/makedepend.SH	2006-10-01 20:05:40.000000000 -0600
+++ perl-5.8.8/makedepend.SH	2006-10-01 20:07:03.000000000 -0600
@@ -167,6 +167,7 @@
             -e '/^#.*<builtin>/d' \
             -e '/^#.*<built-in>/d' \
             -e '/^#.*<command line>/d' \
+            -e '/^#.*<command-line>/d' \
 	    -e '/^#.*"-"/d' \
 	    -e '/: file path prefix .* never used$/d' \
 	    -e 's#\.[0-9][0-9]*\.c#'"$file.c#" \
