diff -Nuar menu-cache-1.0.0/menu-cache-gen/menu-compose.c menu-cache-1.0.0-fix/menu-cache-gen/menu-compose.c
--- menu-cache-1.0.0/menu-cache-gen/menu-compose.c	2014-09-28 16:00:07.000000000 +0000
+++ menu-cache-1.0.0-fix/menu-cache-gen/menu-compose.c	2015-01-13 12:42:57.702341824 +0000
@@ -1023,7 +1023,7 @@
     gboolean ok = FALSE;
 
     tmp = (char *)g_getenv("CACHE_GEN_VERSION");
-    if (sscanf(tmp, "%d.%u", &i, &req_version) == 2)
+    if (tmp && sscanf(tmp, "%d.%u", &i, &req_version) == 2)
     {
         if (i != VER_MAJOR) /* unsupported format requested */
             return FALSE;
