From 699d8b803eea2e36d84b5aaa695db47b7718063e Mon Sep 17 00:00:00 2001
From: Gregor Jasny <gjasny@googlemail.com>
Date: Wed, 15 Jun 2016 20:50:27 +0200
Subject: libv4lconvert: Exclude memsrc implementation for libjpeg-turbo

Signed-off-by: Gregor Jasny <gjasny@googlemail.com>

diff --git a/lib/libv4lconvert/jpeg_memsrcdest.c b/lib/libv4lconvert/jpeg_memsrcdest.c
index 323e7af..0540470 100644
--- a/lib/libv4lconvert/jpeg_memsrcdest.c
+++ b/lib/libv4lconvert/jpeg_memsrcdest.c
@@ -31,7 +31,7 @@
 
 /* libjpeg8 and later come with their own (API compatible) memory source
    and dest */
-#if JPEG_LIB_VERSION < 80
+#if JPEG_LIB_VERSION < 80 && !defined(MEM_SRCDST_SUPPORTED)
 
 /* Expanded data source object for memory input */
 
@@ -309,5 +309,5 @@ jpeg_mem_dest (j_compress_ptr cinfo, unsigned char ** outbuffer,
 	dest->pub.free_in_buffer = dest->buf_size;
 }
 
-#endif /* JPEG_LIB_VERSION < 80 */
+#endif /* JPEG_LIB_VERSION < 80 && !defined(MEM_SRCDST_SUPPORTED) */
 #endif /* HAVE_JPEG */
diff --git a/lib/libv4lconvert/jpeg_memsrcdest.h b/lib/libv4lconvert/jpeg_memsrcdest.h
index 28a6477..b13bf3f 100644
--- a/lib/libv4lconvert/jpeg_memsrcdest.h
+++ b/lib/libv4lconvert/jpeg_memsrcdest.h
@@ -1,6 +1,6 @@
 #include <jpeglib.h>
 
-#if JPEG_LIB_VERSION < 80
+#if JPEG_LIB_VERSION < 80 && !defined(MEM_SRCDST_SUPPORTED)
 
 void
 jpeg_mem_src (j_decompress_ptr cinfo, unsigned char * buffer,
-- 
cgit v0.10.2

