diff -Nuar lame-3.99.5/configure.in lame-3.99.5-fix/configure.in
--- lame-3.99.5/configure.in	2012-02-28 18:50:27.000000000 +0000
+++ lame-3.99.5-fix/configure.in	2015-06-28 12:50:41.842852622 +0000
@@ -87,6 +87,16 @@
 
 dnl Checks for header files.
 AC_HEADER_STDC
+
+case $host_cpu in
+i586*|i686*)
+check_xmmintrin=
+ ;;
+*)
+check_xmmintrin="xmmintrin.h"
+ ;;
+esac
+
 AC_CHECK_HEADERS( \
 		 errno.h \
 		 fcntl.h \
@@ -96,7 +106,7 @@
 		 sys/soundcard.h \
 		 sys/time.h \
 		 unistd.h \
-		 xmmintrin.h \
+		 ${check_xmmintrin} \
 		 linux/soundcard.h)
 
 dnl Checks for typedefs, structures, and compiler characteristics.
