--- menu.c
+++ menu.c
@@ -58,7 +58,7 @@
 	    if (option (OPTASCIICHARS))
 	      addch ('`');
 	    else if (Charset_is_utf8)
-	      addstr ("\342\224\224"); /* WACS_LLCORNER */
+	      add_wch (WACS_LLCORNER);
 	    else
 	      addch (ACS_LLCORNER);
 	    break;
@@ -66,7 +66,7 @@
 	    if (option (OPTASCIICHARS))
 	      addch (',');
 	    else if (Charset_is_utf8)
-	      addstr ("\342\224\214"); /* WACS_ULCORNER */
+	      add_wch (WACS_ULCORNER);
 	    else
 	      addch (ACS_ULCORNER);
 	    break;
@@ -74,7 +74,7 @@
 	    if (option (OPTASCIICHARS))
 	      addch ('|');
 	    else if (Charset_is_utf8)
-	      addstr ("\342\224\234"); /* WACS_LTEE */
+	      add_wch (WACS_LTEE);
 	    else
 	      addch (ACS_LTEE);
 	    break;
@@ -82,7 +82,7 @@
 	    if (option (OPTASCIICHARS))
 	      addch ('-');
 	    else if (Charset_is_utf8)
-	      addstr ("\342\224\200"); /* WACS_HLINE */
+	      add_wch (WACS_HLINE);
 	    else
 	      addch (ACS_HLINE);
 	    break;
@@ -90,7 +90,7 @@
 	    if (option (OPTASCIICHARS))
 	      addch ('|');
 	    else if (Charset_is_utf8)
-	      addstr ("\342\224\202"); /* WACS_VLINE */
+	      add_wch (WACS_VLINE);
 	    else
 	      addch (ACS_VLINE);
 	    break;
@@ -98,7 +98,7 @@
 	    if (option (OPTASCIICHARS))
 	      addch ('-');
 	    else if (Charset_is_utf8)
-	      addstr ("\342\224\254"); /* WACS_TTEE */
+	      add_wch (WACS_TTEE);
 	    else
 	      addch (ACS_TTEE);
 	    break;
@@ -106,7 +106,7 @@
 	    if (option (OPTASCIICHARS))
 	      addch ('-');
 	    else if (Charset_is_utf8)
-	      addstr ("\342\224\264"); /* WACS_BTEE */
+	      add_wch (WACS_BTEE);
 	    else
 	      addch (ACS_BTEE);
 	    break;
--- mutt_curses.h
+++ mutt_curses.h
@@ -43,6 +43,7 @@
 #else /* USE_SLANG_CURSES */
 
 #if HAVE_NCURSESW_NCURSES_H
+# define _XOPEN_SOURCE_EXTENDED
 # include <ncursesw/ncurses.h>
 #elif HAVE_NCURSES_NCURSES_H
 # include <ncurses/ncurses.h>
