diff --git a/Makefile.am b/Makefile.am
index 1f36629..20d205e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,14 +2,16 @@
 ## Use aclocal -I m4; automake --foreign
 
 AUTOMAKE_OPTIONS = 1.6 foreign
-EXTRA_PROGRAMS = mutt_dotlock pgpring pgpewrap makedoc
+EXTRA_PROGRAMS = mutt_dotlock pgpring pgpewrap
+
+noinst_PROGRAMS = makedoc
 
 if BUILD_IMAP
 IMAP_SUBDIR = imap
 IMAP_INCLUDES = -I$(top_srcdir)/imap
 endif
 
-SUBDIRS = m4 po intl doc contrib $(IMAP_SUBDIR)
+SUBDIRS = m4 po intl $(IMAP_SUBDIR) . doc contrib
 
 bin_SCRIPTS = muttbug flea @SMIMEAUX_TARGET@
 
@@ -37,10 +39,6 @@ mutt_LDADD = @MUTT_LIB_OBJECTS@ @LIBOBJS@ $(LIBIMAP) $(MUTTLIBS) \
 mutt_DEPENDENCIES = @MUTT_LIB_OBJECTS@ @LIBOBJS@ $(LIBIMAPDEPS) \
 	$(INTLDEPS)
 
-makedoc_SOURCES = makedoc.c
-makedoc_LDADD =
-makedoc_DEPENDENCIES = 
-
 # $(makedoc_OBJECTS): $(makedoc_SOURCES)
 # 	$(HOST_CC) $(DEFS) $(AM_CPPFLAGS) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) -c $<
 
