diff -Nuar -Nuar sandbox-2.6/configure.ac sandbox-2.6-fixed/configure.ac
--- sandbox-2.6/configure.ac	2012-07-03 19:43:15.000000000 +0000
+++ sandbox-2.6-fixed/configure.ac	2013-10-10 12:07:09.955305402 +0000
@@ -108,8 +108,10 @@
 	sys/user.h
 	sys/wait.h
 	asm/ptrace.h
-	linux/ptrace.h
 ]))
+if test "$HAVE_LINUX_PTRACE_H" = ""; then
+	AC_CHECK_HEADER([linux/ptrace.h])
+fi	
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 dnl Do this after all headers have been checked.
