diff -r 784caad93325 xen/common/keyhandler.c
--- a/xen/common/keyhandler.c	Tue Nov 10 15:03:52 2009 +0000
+++ b/xen/common/keyhandler.c	Tue Jan 05 10:47:49 2010 +0000
@@ -106,6 +106,7 @@
     unsigned int cpu;
 
     /* We want to get everything out that we possibly can. */
+    watchdog_disable();
     console_start_sync();
 
     printk("'%c' pressed -> dumping registers\n", key);
@@ -125,6 +126,7 @@
     printk("\n");
 
     console_end_sync();
+    watchdog_enable();
 }
 
 static void dump_dom0_registers(unsigned char key)
