--- linux-2.6.34-sabayon.orig/arch/ia64/include/asm/compat.h	2010-05-16 23:17:36.000000000 +0200
+++ linux-2.6.34-sabayon/arch/ia64/include/asm/compat.h	2010-09-18 17:09:30.728304722 +0200
@@ -199,7 +199,7 @@ ptr_to_compat(void __user *uptr)
 }
 
 static __inline__ void __user *
-compat_alloc_user_space (long len)
+arch_compat_alloc_user_space (long len)
 {
 	struct pt_regs *regs = task_pt_regs(current);
 	return (void __user *) (((regs->r12 & 0xffffffff) & -16) - len);
--- linux-2.6.34-sabayon.orig/arch/mips/include/asm/compat.h	2010-05-16 23:17:36.000000000 +0200
+++ linux-2.6.34-sabayon/arch/mips/include/asm/compat.h	2010-09-18 17:09:30.728822831 +0200
@@ -145,7 +145,7 @@ static inline compat_uptr_t ptr_to_compa
 	return (u32)(unsigned long)uptr;
 }
 
-static inline void __user *compat_alloc_user_space(long len)
+static inline void __user *arch_compat_alloc_user_space(long len)
 {
 	struct pt_regs *regs = (struct pt_regs *)
 		((unsigned long) current_thread_info() + THREAD_SIZE - 32) - 1;
--- linux-2.6.34-sabayon.orig/arch/parisc/include/asm/compat.h	2010-05-16 23:17:36.000000000 +0200
+++ linux-2.6.34-sabayon/arch/parisc/include/asm/compat.h	2010-09-18 17:09:30.728822831 +0200
@@ -147,7 +147,7 @@ static inline compat_uptr_t ptr_to_compa
 	return (u32)(unsigned long)uptr;
 }
 
-static __inline__ void __user *compat_alloc_user_space(long len)
+static __inline__ void __user *arch_compat_alloc_user_space(long len)
 {
 	struct pt_regs *regs = &current->thread.regs;
 	return (void __user *)regs->gr[30];
--- linux-2.6.34-sabayon.orig/arch/powerpc/include/asm/compat.h	2010-05-16 23:17:36.000000000 +0200
+++ linux-2.6.34-sabayon/arch/powerpc/include/asm/compat.h	2010-09-18 17:09:30.728822831 +0200
@@ -134,7 +134,7 @@ static inline compat_uptr_t ptr_to_compa
 	return (u32)(unsigned long)uptr;
 }
 
-static inline void __user *compat_alloc_user_space(long len)
+static inline void __user *arch_compat_alloc_user_space(long len)
 {
 	struct pt_regs *regs = current->thread.regs;
 	unsigned long usp = regs->gpr[1];
--- linux-2.6.34-sabayon.orig/arch/s390/include/asm/compat.h	2010-05-16 23:17:36.000000000 +0200
+++ linux-2.6.34-sabayon/arch/s390/include/asm/compat.h	2010-09-18 17:09:30.728822831 +0200
@@ -181,7 +181,7 @@ static inline int is_compat_task(void)
 
 #endif
 
-static inline void __user *compat_alloc_user_space(long len)
+static inline void __user *arch_compat_alloc_user_space(long len)
 {
 	unsigned long stack;
 
--- linux-2.6.34-sabayon.orig/arch/sparc/include/asm/compat.h	2010-05-16 23:17:36.000000000 +0200
+++ linux-2.6.34-sabayon/arch/sparc/include/asm/compat.h	2010-09-18 17:09:30.729822459 +0200
@@ -167,7 +167,7 @@ static inline compat_uptr_t ptr_to_compa
 	return (u32)(unsigned long)uptr;
 }
 
-static inline void __user *compat_alloc_user_space(long len)
+static inline void __user *arch_compat_alloc_user_space(long len)
 {
 	struct pt_regs *regs = current_thread_info()->kregs;
 	unsigned long usp = regs->u_regs[UREG_I6];
--- linux-2.6.34-sabayon.orig/arch/x86/include/asm/compat.h	2010-05-16 23:17:36.000000000 +0200
+++ linux-2.6.34-sabayon/arch/x86/include/asm/compat.h	2010-09-18 17:09:30.729822459 +0200
@@ -205,7 +205,7 @@ static inline compat_uptr_t ptr_to_compa
 	return (u32)(unsigned long)uptr;
 }
 
-static inline void __user *compat_alloc_user_space(long len)
+static inline void __user *arch_compat_alloc_user_space(long len)
 {
 	struct pt_regs *regs = task_pt_regs(current);
 	return (void __user *)regs->sp - len;
--- linux-2.6.34-sabayon.orig/include/linux/compat.h	2010-05-16 23:17:36.000000000 +0200
+++ linux-2.6.34-sabayon/include/linux/compat.h	2010-09-18 17:09:30.729822459 +0200
@@ -356,5 +356,7 @@ asmlinkage long compat_sys_newfstatat(un
 asmlinkage long compat_sys_openat(unsigned int dfd, const char __user *filename,
 				  int flags, int mode);
 
+extern void __user *compat_alloc_user_space(unsigned long len);
+
 #endif /* CONFIG_COMPAT */
 #endif /* _LINUX_COMPAT_H */
--- linux-2.6.34-sabayon.orig/kernel/compat.c	2010-05-16 23:17:36.000000000 +0200
+++ linux-2.6.34-sabayon/kernel/compat.c	2010-09-18 17:09:58.574836565 +0200
@@ -25,6 +25,7 @@
 #include <linux/posix-timers.h>
 #include <linux/times.h>
 #include <linux/ptrace.h>
+#include <linux/module.h>
 #include <linux/gfp.h>
 
 #include <asm/uaccess.h>
@@ -1140,3 +1141,24 @@ compat_sys_sysinfo(struct compat_sysinfo
 
 	return 0;
 }
+
+/*
+ * Allocate user-space memory for the duration of a single system call,
+ * in order to marshall parameters inside a compat thunk.
+ */
+void __user *compat_alloc_user_space(unsigned long len)
+{
+	void __user *ptr;
+
+	/* If len would occupy more than half of the entire compat space... */
+	if (unlikely(len > (((compat_uptr_t)~0) >> 1)))
+		return NULL;
+
+	ptr = arch_compat_alloc_user_space(len);
+
+	if (unlikely(!access_ok(VERIFY_WRITE, ptr, len)))
+		return NULL;
+
+	return ptr;
+}
+EXPORT_SYMBOL_GPL(compat_alloc_user_space);
