From 6c25bb24a250eaa41df02a0fe0ea696f227e8d18 Mon Sep 17 00:00:00 2001
From: Thomas S Hatch <thatch45@gmail.com>
Date: Sat, 24 Mar 2012 11:51:24 -0600
Subject: [PATCH] fix issue with syndic resolution of master 1p

---
 salt/__init__.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/salt/__init__.py b/salt/__init__.py
index 61b98e3..bce0065 100644
--- a/salt/__init__.py
+++ b/salt/__init__.py
@@ -223,7 +223,7 @@ class Syndic(object):
             # Some of the opts need to be changed to match the needed opts
             # in the minion class.
             opts['master'] = opts['syndic_master']
-            opts['master_ip'] = salt.config.dns_check(opts['master'])
+            opts['master_ip'] = salt.utils.dns_check(opts['master'])
 
             opts['master_uri'] = ('tcp://' + opts['master_ip'] +
                                   ':' + str(opts['master_port']))
-- 
1.7.8.5

