Index: info/Makefile.am
===================================================================
RCS file: /cvsroot/texinfo/texinfo/info/Makefile.am,v
retrieving revision 1.14
retrieving revision 1.15
diff -u -p -r1.14 -r1.15
--- info/Makefile.in	15 Dec 2007 14:32:51 -0000	1.14
+++ info/Makefile.in	28 Mar 2008 23:39:20 -0000	1.15
@@ -70,8 +70,10 @@ cmd_sources = $(srcdir)/session.c $(srcd
   $(srcdir)/m-x.c $(srcdir)/indices.c $(srcdir)/nodemenu.c \
   $(srcdir)/footnotes.c $(srcdir)/variables.c
 
-# The $(EXEEXT) should be added by Automake, but isn't. Fine.
-$(generated_sources): makedoc$(EXEEXT) $(cmd_sources)
+# Make the target only funs.h even though we create all the
+# $(generated_sources) so that parallel makes won't do the rm && makedoc
+# more than once.
+funs.h: makedoc$(EXEEXT) $(cmd_sources)
 	rm -f $(generated_sources)
 	$(top_builddir)/$(native_tools)/info/makedoc $(cmd_sources)
 
