fix from upstream

http://bugs.gentoo.org/198545

--- texinfo/makeinfo/cmds.c	2007/09/26 20:53:39	1.76
+++ texinfo/makeinfo/cmds.c	2007/10/13 23:44:34	1.77
@@ -1631,10 +1631,14 @@
 
       if (!no_headers && !html)
         {
+          /* use add_* instead of insert_* because otherwise the
+             file header ("This is ...") will end up inside the
+             dir section markers.  */
           kill_self_indent (-1); /* make sure there's no indentation */
-          insert_string ("INFO-DIR-SECTION ");
-          insert_string (line);
-          insert ('\n');
+          cm_noindent (); /* make sure again */
+          add_word ("INFO-DIR-SECTION ");
+          add_word (line);
+          add_char ('\n');
         }
 
       free (line);
