http://bugs.gentoo.org/195313

--- texinfo-4.11/util/dvipdf.test
+++ texinfo-4.11/util/dvipdf.test
@@ -16,6 +16,8 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+required='tex'
+
 . ./defs || exit 1
 
 set -e
--- texinfo-4.11/util/texi2dvi.test
+++ texinfo-4.11/util/texi2dvi.test
@@ -16,6 +16,8 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+required='tex'
+
 . ./defs || exit 1
 
 set -e
