--- postgresql-7.4.26.orig/configure.in	2009-10-13 17:52:02.629889931 +0200
+++ postgresql-7.4.26/configure.in	2009-10-13 18:02:40.776550323 +0200
@@ -777,7 +777,7 @@
 
 if test "$with_krb5" = yes ; then
   AC_CHECK_HEADER(krb5.h, [], [AC_MSG_ERROR([header file <krb5.h> is required for Kerberos 5])])
-  AC_CHECK_HEADER(com_err.h, [], [AC_MSG_ERROR([header file <com_err.h> is required for Kerberos 5])])
+  AC_CHECK_HEADER(et/com_err.h, [], [AC_MSG_ERROR([header file <et/com_err.h> is required for Kerberos 5])])
 fi
 
 if test "$with_openssl" = yes ; then
diff -ur postgresql-7.4.26.orig/src/backend/libpq/auth.c postgresql-7.4.26/src/backend/libpq/auth.c
--- postgresql-7.4.26.orig/src/backend/libpq/auth.c	2003-12-20 19:25:02.000000000 +0100
+++ postgresql-7.4.26/src/backend/libpq/auth.c	2009-10-13 18:15:14.693467302 +0200
@@ -150,7 +150,7 @@
  */
 
 #include <krb5.h>
-#include <com_err.h>
+#include <et/com_err.h>
 
 /*
  * pg_an_to_ln -- return the local name corresponding to an authentication
diff -ur postgresql-7.4.26.orig/src/interfaces/libpq/fe-auth.c postgresql-7.4.26/src/interfaces/libpq/fe-auth.c
--- postgresql-7.4.26.orig/src/interfaces/libpq/fe-auth.c	2007-07-23 20:10:13.000000000 +0200
+++ postgresql-7.4.26/src/interfaces/libpq/fe-auth.c	2009-10-13 18:15:14.685467676 +0200
@@ -243,7 +243,7 @@
  */
 
 #include <krb5.h>
-#include <com_err.h>
+#include <et/com_err.h>
 
 /*
  * pg_an_to_ln -- return the local name corresponding to an authentication
