diff --git a/config/programs.m4 b/config/programs.m4
index ae3b91a..6af24d2 100644
--- a/config/programs.m4
+++ b/config/programs.m4
@@ -153,13 +153,13 @@ AC_DEFUN([PGAC_CHECK_GETTEXT],
 dnl FIXME: We should probably check for version >=0.10.36.
   AC_CHECK_PROGS(XGETTEXT, xgettext)
 
-  # Note: share/locale is always the default, independent of $datadir
-  localedir='${prefix}/share/locale'
-  if test x"$prefix" = x"NONE"; then
-    exp_localedir="$ac_default_prefix/share/locale"
-  else
-    exp_localedir="$prefix/share/locale"
-  fi
+  # Note: share/locale *WAS* always the default, independent of $datadir
+  AC_ARG_WITH([locale-dir],
+   AC_HELP_STRING([--with-locale-dir],[Set path to locale files]),
+   [ localedir="${withval}" ],
+   [ localedir='${prefix}/share/locale' ]
+  )
+  exp_localedir="${localedir}"
 
   AC_SUBST(localedir)
   AC_DEFINE_UNQUOTED(LOCALEDIR, ["$exp_localedir"],
diff --git a/configure.in b/configure.in
index f7bdec3..c8b5b7e 100644
--- a/configure.in
+++ b/configure.in
@@ -19,17 +19,8 @@ m4_pattern_forbid(^PGAC_)dnl to catch undefined macros
 dnl
 dnl The GNU folks apparently haven't heard that some people don't use
 dnl Texinfo. Use this sorcery to use "docdir" instead of "infodir".
-m4_define([info], [doc])
-m4_define([infodir], [docdir])
 AC_INIT([PostgreSQL], [7.4.21], [pgsql-bugs@postgresql.org])
-m4_undefine([infodir])
-m4_undefine([info])
-AC_SUBST(docdir)
-
-m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.53], [], [m4_fatal([Autoconf version 2.53 is required.
-Untested combinations of 'autoconf' and PostgreSQL versions are not
-recommended.  You can remove the check from 'configure.in' but it is then
-your responsibility whether the result works or not.])])
+
 AC_COPYRIGHT([Copyright 2004 PostgreSQL Global Development Group])
 AC_CONFIG_SRCDIR([src/backend/access/common/heaptuple.c])
 AC_CONFIG_AUX_DIR(config)
