diff -Naur postgresql-8.5alpha3.orig/contrib/adminpack/Makefile postgresql-8.5alpha3/contrib/adminpack/Makefile
--- postgresql-8.5alpha3.orig/contrib/adminpack/Makefile	2007-11-11 00:59:50.000000000 +0100
+++ postgresql-8.5alpha3/contrib/adminpack/Makefile	2009-07-09 18:10:56.000000000 +0200
@@ -1,7 +1,7 @@
 # $PostgreSQL: pgsql/contrib/adminpack/Makefile,v 1.6 2007/11/10 23:59:50 momjian Exp $
 
 MODULE_big = adminpack
-PG_CPPFLAGS = -I$(libpq_srcdir)
+PG_CPPFLAGS = -I$(libpq_srcdir) -I../../src/include/
 DATA_built = adminpack.sql
 DATA = uninstall_adminpack.sql
 OBJS = adminpack.o
diff -Naur postgresql-8.5alpha3.orig/contrib/dblink/Makefile postgresql-8.5alpha3/contrib/dblink/Makefile
--- postgresql-8.5alpha3.orig/contrib/dblink/Makefile	2007-11-11 00:59:50.000000000 +0100
+++ postgresql-8.5alpha3/contrib/dblink/Makefile	2009-07-09 22:22:02.000000000 +0200
@@ -1,7 +1,7 @@
 # $PostgreSQL: pgsql/contrib/dblink/Makefile,v 1.15 2007/11/10 23:59:50 momjian Exp $
 
 MODULE_big = dblink
-PG_CPPFLAGS = -I$(libpq_srcdir)
+PG_CPPFLAGS = -I$(libpq_srcdir) -I../../src/include/
 OBJS	= dblink.o
 SHLIB_LINK = $(libpq)
 
diff -Naur postgresql-8.5alpha3.orig/contrib/Makefile postgresql-8.5alpha3/contrib/Makefile
--- postgresql-8.5alpha3.orig/contrib/Makefile	2009-03-26 00:20:01.000000000 +0100
+++ postgresql-8.5alpha3/contrib/Makefile	2009-07-09 18:13:23.000000000 +0200
@@ -25,25 +25,18 @@
 		isn		\
 		lo		\
 		ltree		\
-		oid2name	\
 		pageinspect	\
-		passwordcheck	\
 		pg_buffercache	\
 		pg_freespacemap \
 		pg_standby	\
 		pg_stat_statements \
 		pg_trgm		\
-		pg_upgrade	\
-		pg_upgrade_support \
-		pgbench		\
 		pgcrypto	\
 		pgrowlocks	\
 		pgstattuple	\
 		seg		\
 		spi		\
 		tablefunc	\
		test_parser	\
 		tsearch2	\
-		unaccent	\
-		vacuumlo
 
 ifeq ($(with_openssl),yes)
 WANTED_DIRS += sslinfo
diff -Naur postgresql-8.5alpha3.orig/contrib/uuid-ossp/Makefile postgresql-8.5alpha3/contrib/uuid-ossp/Makefile
--- postgresql-8.5alpha3.orig/contrib/uuid-ossp/Makefile	2007-11-13 01:13:19.000000000 +0100
+++ postgresql-8.5alpha3/contrib/uuid-ossp/Makefile	2009-07-09 19:12:44.000000000 +0200
@@ -1,11 +1,13 @@
 # $PostgreSQL: pgsql/contrib/uuid-ossp/Makefile,v 1.4 2007/11/13 00:13:19 tgl Exp $
 
+PG_CPPFLAGS += "-DHAVE_OSSP_UUID_H"
+
 MODULE_big = uuid-ossp
 OBJS = uuid-ossp.o
 DATA_built = uuid-ossp.sql
 DATA = uninstall_uuid-ossp.sql
 
-SHLIB_LINK += $(OSSP_UUID_LIBS)
+SHLIB_LINK += -lossp-uuid
 
 ifdef USE_PGXS
 PG_CONFIG = pg_config
diff -Naur postgresql-8.5alpha3.orig/contrib/xml2/Makefile postgresql-8.5alpha3/contrib/xml2/Makefile
--- postgresql-8.5alpha3.orig/contrib/xml2/Makefile	2008-05-08 18:49:37.000000000 +0200
+++ postgresql-8.5alpha3/contrib/xml2/Makefile	2009-07-09 19:14:32.000000000 +0200
@@ -4,7 +4,8 @@
 
 OBJS = $(if $(filter -lxslt, $(LIBS)), xpath.o xslt_proc.o, xpath.o)
 
-SHLIB_LINK += $(filter -lxslt, $(LIBS)) $(filter -lxml2, $(LIBS))
+PG_CPPFLAGS = $(shell xml2-config --cflags)
+SHLIB_LINK += $(shell xml2-config --libs) $(shell xslt-config --libs)
 
 DATA_built = pgxml.sql
 DATA = uninstall_pgxml.sql
diff -Naur postgresql-8.5alpha3.orig/GNUmakefile.in postgresql-8.5alpha3/GNUmakefile.in
--- postgresql-8.5alpha3.orig/GNUmakefile.in	2009-01-15 02:53:49.000000000 +0100
+++ postgresql-8.5alpha3/GNUmakefile.in	2009-07-09 19:16:19.000000000 +0200
@@ -11,7 +11,6 @@
 all:
	$(MAKE) -C doc all
	$(MAKE) -C src all
-	$(MAKE) -C config all
	@echo "All of PostgreSQL successfully made. Ready to install."

 world:
@@ -27,7 +26,6 @@
 install:
	$(MAKE) -C doc $@
	$(MAKE) -C src $@
-	$(MAKE) -C config $@
	@echo "PostgreSQL installation complete."

 install-world:
@@ -40,7 +38,6 @@
 installdirs uninstall coverage:
	$(MAKE) -C doc $@
	$(MAKE) -C src $@
-	$(MAKE) -C config $@

 distprep:
	$(MAKE) -C doc $@
diff -Naur postgresql-8.5alpha3.orig/src/bin/initdb/Makefile 
postgresql-8.5alpha3/src/bin/initdb/Makefile
--- postgresql-8.5alpha3.orig/src/bin/initdb/Makefile	2009-01-01 18:23:53.000000000 +0100
+++ postgresql-8.5alpha3/src/bin/initdb/Makefile	2009-07-09 19:17:40.000000000 +0200
@@ -14,7 +14,7 @@
 top_builddir = ../../..
 include $(top_builddir)/src/Makefile.global
 
-override CPPFLAGS := -DFRONTEND -I$(libpq_srcdir) $(CPPFLAGS)
+override CPPFLAGS := -DFRONTEND -I$(top_srcdir)/src/interfaces/libpq $(CPPFLAGS)
 
 OBJS=	initdb.o encnames.o pqsignal.o $(WIN32RES)
 
diff -Naur postgresql-8.5alpha3.orig/src/bin/Makefile postgresql-8.5alpha3/src/bin/Makefile
--- postgresql-8.5alpha3.orig/src/bin/Makefile	2009-01-01 18:23:53.000000000 +0100
+++ postgresql-8.5alpha3/src/bin/Makefile	2009-07-09 19:18:21.000000000 +0200
@@ -13,8 +13,8 @@
 top_builddir = ../..
 include $(top_builddir)/src/Makefile.global
 
-SUBDIRS = initdb pg_ctl pg_dump \
-	psql scripts pg_config pg_controldata pg_resetxlog
+SUBDIRS = initdb pg_ctl \
+	pg_controldata pg_resetxlog
 ifeq ($(PORTNAME), win32)
 SUBDIRS+=pgevent
 endif
diff -Naur postgresql-8.5alpha3.orig/src/include/pg_config_manual.h postgresql-8.5alpha3/src/include/pg_config_manual.h
--- postgresql-8.5alpha3.orig/src/include/pg_config_manual.h	2009-06-11 16:49:08.000000000 +0200
+++ postgresql-8.5alpha3/src/include/pg_config_manual.h	2009-07-09 19:19:40.000000000 +0200
@@ -146,7 +146,7 @@
  * here's where to twiddle it.  You can also override this at runtime
  * with the postmaster's -k switch.
  */
-#define DEFAULT_PGSOCKET_DIR  "/tmp"
+#define DEFAULT_PGSOCKET_DIR  "/var/run/postgresql"
 
 /*
  * The random() function is expected to yield values between 0 and
diff -Naur postgresql-8.5alpha3.orig/src/Makefile postgresql-8.5alpha3/src/Makefile
--- postgresql-8.5alpha3.orig/src/Makefile	2009-02-24 11:06:32.000000000 +0100
+++ postgresql-8.5alpha3/src/Makefile	2009-07-09 19:20:44.000000000 +0200
@@ -19,15 +19,11 @@
 	$(MAKE) -C backend $@
 	$(MAKE) -C backend/utils/mb/conversion_procs $@
 	$(MAKE) -C backend/snowball $@
-	$(MAKE) -C include $@
-	$(MAKE) -C interfaces $@
-	$(MAKE) -C backend/replication/libpqwalreceiver $@
 	$(MAKE) -C bin $@
 	$(MAKE) -C pl $@
-	$(MAKE) -C makefiles $@
 	$(MAKE) -C test/regress $@
 
-install: install-local
+install:
 
 install-local: installdirs-local
 	$(INSTALL_DATA) Makefile.global '$(DESTDIR)$(pgxsdir)/$(subdir)/Makefile.global'
diff -Naur postgresql-8.5alpha3.orig/src/Makefile.global.in postgresql-8.5alpha3/src/Makefile.global.in
--- postgresql-8.5alpha3.orig/src/Makefile.global.in	2009-06-23 05:46:00.000000000 +0200
+++ postgresql-8.5alpha3/src/Makefile.global.in	2009-07-09 19:39:46.000000000 +0200
@@ -443,6 +441,8 @@
    LDFLAGS += $(PROFILE)
 endif
 
+CFLAGS += -I${top_srcdir}/src/include
+
 
 ##########################################################################
 #
diff -Naur postgresql-8.5alpha3.orig/src/port/Makefile postgresql-8.5alpha3/src/port/Makefile
--- postgresql-8.5alpha3.orig/src/port/Makefile	2008-11-25 19:19:31.000000000 +0100
+++ postgresql-8.5alpha3/src/port/Makefile	2009-07-09 19:24:34.000000000 +0200
@@ -39,11 +39,10 @@
 # foo_srv.o and foo.o are both built from foo.c, but only foo.o has -DFRONTEND
 OBJS_SRV = $(OBJS:%.o=%_srv.o)
 
-all: libpgport.a libpgport_srv.a
+all: libpgport_srv.a
 
 # libpgport is needed by some contrib
-install: all installdirs
-	$(INSTALL_STLIB) libpgport.a '$(DESTDIR)$(libdir)/libpgport.a'
+install:
 
 installdirs:
 	$(mkinstalldirs) '$(DESTDIR)$(libdir)'
diff -Naur postgresql-8.5alpha3.orig/src/test/regress/GNUmakefile postgresql-8.5alpha3/src/test/regress/GNUmakefile
--- postgresql-8.5alpha3.orig/src/test/regress/GNUmakefile	2009-01-01 18:24:04.000000000 +0100
+++ postgresql-8.5alpha3/src/test/regress/GNUmakefile	2009-07-09 19:28:25.000000000 +0200
@@ -141,7 +141,7 @@
 pg_regress_call = ./pg_regress --inputdir=$(srcdir) --dlpath=. --multibyte=$(MULTIBYTE) --load-language=plpgsql $(NOLOCALE)
 
 check: all
-	$(pg_regress_call) --temp-install=./tmp_check --top-builddir=$(top_builddir) --schedule=$(srcdir)/parallel_schedule $(MAXCONNOPT) $(TEMP_CONF)
+	$(pg_regress_call) --temp-install=./tmp_check --top-builddir=$(top_builddir) --schedule=$(srcdir)/parallel_schedule $(MAXCONNOPT) $(TEMP_CONF) --psqldir=/usr/lib/postgresql-${SLOT}/bin/
 
 installcheck: all
 	$(pg_regress_call) --psqldir=$(PSQLDIR) --schedule=$(srcdir)/serial_schedule
