=== GNUmakefile.in
==================================================================
--- GNUmakefile.in	(revision 14)
+++ GNUmakefile.in	(local)
@@ -11,19 +11,16 @@
 all:
 	$(MAKE) -C doc all
 	$(MAKE) -C src all
-	$(MAKE) -C config all
 	@echo "All of PostgreSQL successfully made. Ready to install."
 
 install:
 	$(MAKE) -C doc $@
 	$(MAKE) -C src $@
-	$(MAKE) -C config $@
 	@echo "PostgreSQL installation complete."
 
 installdirs uninstall:
 	$(MAKE) -C doc $@
 	$(MAKE) -C src $@
-	$(MAKE) -C config $@
 
 distprep:
 	$(MAKE) -C doc $@
=== contrib/Makefile
==================================================================
--- contrib/Makefile	(revision 14)
+++ contrib/Makefile	(local)
@@ -20,22 +20,19 @@
 		isn		\
 		lo		\
 		ltree		\
-		oid2name	\
 		pageinspect	\
 		pg_buffercache	\
 		pg_freespacemap \
 		pg_standby	\
 		pg_trgm		\
-		pgbench		\
 		pgcrypto	\
 		pgrowlocks	\
 		pgstattuple	\
 		seg		\
 		spi		\
 		tablefunc	\
-		test_parser	\
 		tsearch2	\
-		vacuumlo
+		test_parser
 
 ifeq ($(with_openssl),yes)
 WANTED_DIRS += sslinfo
=== contrib/adminpack/Makefile
==================================================================
--- contrib/adminpack/Makefile	(revision 14)
+++ contrib/adminpack/Makefile	(local)
@@ -1,7 +1,7 @@
 # $PostgreSQL: pgsql/contrib/adminpack/Makefile,v 1.6 2007/11/10 23:59:50 momjian Exp $
 
 MODULE_big = adminpack
-PG_CPPFLAGS = -I$(libpq_srcdir)
+PG_CPPFLAGS = -I$(libpq_srcdir) -I../../src/include/
 DATA_built = adminpack.sql
 DATA = uninstall_adminpack.sql
 OBJS = adminpack.o
=== contrib/uuid-ossp/Makefile
==================================================================
--- contrib/uuid-ossp/Makefile	(revision 14)
+++ contrib/uuid-ossp/Makefile	(local)
@@ -1,11 +1,13 @@
 # $PostgreSQL: pgsql/contrib/uuid-ossp/Makefile,v 1.4 2007/11/13 00:13:19 tgl Exp $
 
+PG_CPPFLAGS += "-DHAVE_OSSP_UUID_H"
+
 MODULE_big = uuid-ossp
 OBJS = uuid-ossp.o
 DATA_built = uuid-ossp.sql
 DATA = uninstall_uuid-ossp.sql
 
-SHLIB_LINK += $(OSSP_UUID_LIBS)
+SHLIB_LINK += -lossp-uuid
 
 ifdef USE_PGXS
 PG_CONFIG = pg_config
=== src/Makefile
==================================================================
--- src/Makefile	(revision 14)
+++ src/Makefile	(local)
@@ -19,14 +19,11 @@
 	$(MAKE) -C backend $@
 	$(MAKE) -C backend/utils/mb/conversion_procs $@
 	$(MAKE) -C backend/snowball $@
-	$(MAKE) -C include $@
-	$(MAKE) -C interfaces $@
 	$(MAKE) -C bin $@
 	$(MAKE) -C pl $@
-	$(MAKE) -C makefiles $@
 	$(MAKE) -C test/regress $@
 
-install: install-local
+install:
 
 install-local: installdirs-local
 	$(INSTALL_DATA) Makefile.global '$(DESTDIR)$(pgxsdir)/$(subdir)/Makefile.global'
=== src/Makefile.global.in
==================================================================
--- src/Makefile.global.in	(revision 14)
+++ src/Makefile.global.in	(local)
@@ -373,10 +373,8 @@
 
 
 submake-libpq:
-	$(MAKE) -C $(libpq_builddir) all
 
 submake-libpgport:
-	$(MAKE) -C $(top_builddir)/src/port all
 
 .PHONY: submake-libpq submake-libpgport
 
@@ -419,7 +417,9 @@
    LDFLAGS += $(PROFILE)
 endif
 
+CFLAGS += -I${top_srcdir}/src/include
 
+
 ##########################################################################
 #
 # substitute implementations of C library routines (see src/port/)
=== src/bin/Makefile
==================================================================
--- src/bin/Makefile	(revision 14)
+++ src/bin/Makefile	(local)
@@ -13,8 +13,8 @@
 top_builddir = ../..
 include $(top_builddir)/src/Makefile.global
 
-DIRS = initdb ipcclean pg_ctl pg_dump \
-	psql scripts pg_config pg_controldata pg_resetxlog
+DIRS = initdb ipcclean pg_ctl \
+	pg_controldata pg_resetxlog
 ifeq ($(PORTNAME), win32)
 DIRS+=pgevent
 endif
=== src/bin/initdb/Makefile
==================================================================
--- src/bin/initdb/Makefile	(revision 14)
+++ src/bin/initdb/Makefile	(local)
@@ -14,7 +14,7 @@
 top_builddir = ../../..
 include $(top_builddir)/src/Makefile.global
 
-override CPPFLAGS := -DFRONTEND -I$(libpq_srcdir) $(CPPFLAGS)
+override CPPFLAGS := -DFRONTEND -I$(top_srcdir)/src/interfaces/libpq $(CPPFLAGS)
 
 OBJS=	initdb.o encnames.o pqsignal.o $(WIN32RES)
 
=== src/include/pg_config_manual.h
==================================================================
--- src/include/pg_config_manual.h	(revision 14)
+++ src/include/pg_config_manual.h	(local)
@@ -184,7 +184,7 @@
  * here's where to twiddle it.  You can also override this at runtime
  * with the postmaster's -k switch.
  */
-#define DEFAULT_PGSOCKET_DIR  "/tmp"
+#define DEFAULT_PGSOCKET_DIR  "/var/run/postgresql"
 
 /*
  * The random() function is expected to yield values between 0 and
=== src/port/Makefile
==================================================================
--- src/port/Makefile	(revision 14)
+++ src/port/Makefile	(local)
@@ -35,11 +35,10 @@
 # foo_srv.o and foo.o are both built from foo.c, but only foo.o has -DFRONTEND
 OBJS_SRV = $(OBJS:%.o=%_srv.o)
 
-all: libpgport.a libpgport_srv.a
+all: libpgport_srv.a
 
 # libpgport is needed by some contrib
-install: all installdirs
-	$(INSTALL_STLIB) libpgport.a '$(DESTDIR)$(libdir)/libpgport.a'
+install:
 
 installdirs:
 	$(mkinstalldirs) '$(DESTDIR)$(libdir)'
=== src/test/regress/GNUmakefile
==================================================================
--- src/test/regress/GNUmakefile	(revision 14)
+++ src/test/regress/GNUmakefile	(local)
@@ -141,7 +141,7 @@
 ##
 
 check: all
-	./pg_regress --temp-install=./tmp_check --top-builddir=$(top_builddir) --srcdir=$(abs_srcdir) --temp-port=$(TEMP_PORT) --schedule=$(srcdir)/parallel_schedule --multibyte=$(MULTIBYTE) --load-language=plpgsql $(MAXCONNOPT) $(NOLOCALE) $(TEMP_CONF)
+	./pg_regress --temp-install=./tmp_check --top-builddir=$(top_builddir) --srcdir=$(abs_srcdir) --temp-port=$(TEMP_PORT) --schedule=$(srcdir)/parallel_schedule --multibyte=$(MULTIBYTE) --load-language=plpgsql $(MAXCONNOPT) $(NOLOCALE) $(TEMP_CONF) --psqldir=/usr/lib/postgresql-${SLOT}/bin/
 
 installcheck: all
 	./pg_regress --psqldir=$(PSQLDIR) --schedule=$(srcdir)/serial_schedule --srcdir=$(abs_srcdir) --multibyte=$(MULTIBYTE) --load-language=plpgsql $(NOLOCALE)
=== src/test/regress/pg_regress.c
==================================================================
--- src/test/regress/pg_regress.c	(revision 14)
+++ src/test/regress/pg_regress.c	(local)
@@ -781,9 +781,6 @@
 		sprintf(tmp, "%s/install/%s", temp_install, datadir);
 		datadir = tmp;
 
-		/* psql will be installed into temp-install bindir */
-		psqldir = bindir;
-
 		/*
 		 * Set up shared library paths to include the temp install.
 		 *
@@ -2053,7 +2050,7 @@
 		 */
 		snprintf(buf, sizeof(buf),
 				 SYSTEMQUOTE "\"%s/psql\" -X postgres <%s 2>%s" SYSTEMQUOTE,
-				 bindir, DEVNULL, DEVNULL);
+				 psqldir, DEVNULL, DEVNULL);
 		for (i = 0; i < 60; i++)
 		{
 			/* Done if psql succeeds */
