=== GNUmakefile.in
==================================================================
--- GNUmakefile.in	(revision 12)
+++ GNUmakefile.in	(local)
@@ -23,7 +23,6 @@
 	$(MAKE) -C src $@
 
 install-all-headers:
-	$(MAKE) -C src $@
 
 # clean, distclean, etc should apply to contrib too, even though
 # it's not built by default
=== contrib/Makefile
==================================================================
--- contrib/Makefile	(revision 12)
+++ contrib/Makefile	(local)
@@ -13,7 +13,6 @@
 		dbmirror	\
 		dbsize		\
 		earthdistance	\
-		findoidjoins	\
 		fulltextindex	\
 		fuzzystrmatch	\
 		intagg		\
@@ -23,11 +22,7 @@
 		ltree		\
 		miscutil	\
 		noupdate	\
-		oid2name	\
-		pg_autovacuum	\
-		pg_dumplo	\
 		pg_logger	\
-		pgbench		\
 		pgcrypto	\
 		pgstattuple	\
 		rserv		\
@@ -39,8 +34,7 @@
 		tips		\
 		tsearch		\
 		tsearch2	\
-		userlock	\
-		vacuumlo
+		userlock
 
 # Missing:
 #		array		\ (removed all but the README)
=== src/Makefile
==================================================================
--- src/Makefile	(revision 12)
+++ src/Makefile	(local)
@@ -14,11 +14,8 @@
 
 
 all install installdirs uninstall dep depend distprep:
-	$(MAKE) -C port $@
 	$(MAKE) -C backend $@
 	$(MAKE) -C backend/utils/mb/conversion_procs $@
-	$(MAKE) -C include $@
-	$(MAKE) -C interfaces $@
 	$(MAKE) -C bin $@
 	$(MAKE) -C pl $@
 
=== src/Makefile.global.in
==================================================================
--- src/Makefile.global.in	(revision 12)
+++ src/Makefile.global.in	(local)
@@ -291,10 +291,10 @@
 libpq = -L$(libpq_builddir) -lpq
 
 submake-libpq:
-	$(MAKE) -C $(libpq_builddir) all
+	@true $(MAKE) -C $(libpq_builddir) all
 
 submake-libpgport:
-	$(MAKE) -C $(top_builddir)/src/port all
+	@true $(MAKE) -C $(top_builddir)/src/port all
 
 .PHONY: submake-libpq submake-libpgport
 
=== src/bin/Makefile
==================================================================
--- src/bin/Makefile	(revision 12)
+++ src/bin/Makefile	(local)
@@ -13,9 +13,8 @@
 top_builddir = ../..
 include $(top_builddir)/src/Makefile.global
 
-DIRS := initdb initlocation ipcclean pg_ctl pg_dump pg_id \
-	psql scripts pg_config pg_controldata pg_resetxlog \
-	pg_encoding
+DIRS := initdb initlocation ipcclean pg_ctl pg_id \
+	pg_controldata pg_resetxlog
 
 ifeq ($(with_tcl), yes)
 	DIRS += pgtclsh
=== src/include/pg_config_manual.h
==================================================================
--- src/include/pg_config_manual.h	(revision 12)
+++ src/include/pg_config_manual.h	(local)
@@ -165,7 +165,7 @@
  * here's where to twiddle it.  You can also override this at runtime
  * with the postmaster's -k switch.
  */
-#define DEFAULT_PGSOCKET_DIR  "/tmp"
+#define DEFAULT_PGSOCKET_DIR  "/var/run/postgresql"
 
 /*
  * Defining this will make float4 and float8 operations faster by
