diff -Nuar subversion-1.6.17.orig//Makefile.in subversion-1.6.17/Makefile.in
--- subversion-1.6.17.orig//Makefile.in	2011-11-17 01:23:07.941305947 +0600
+++ subversion-1.6.17/Makefile.in	2011-11-17 01:23:42.348307134 +0600
@@ -678,7 +678,7 @@
 extraclean-swig-headers: clean-swig-headers
 	$(EXTRACLEAN_SWIG_HEADERS)
 
-$(SWIG_PL_DIR)/native/Makefile.PL: $(SWIG_SRC_DIR)/perl/native/Makefile.PL.in
+$(SWIG_PL_DIR)/native/Makefile.PL: $(SWIG_SRC_DIR)/perl/native/Makefile.PL.in libsvn_swig_perl
 	./config.status subversion/bindings/swig/perl/native/Makefile.PL
 
 $(SWIG_PL_DIR)/native/Makefile: $(SWIG_PL_DIR)/native/Makefile.PL
