diff -urN b/samples/powerbtn/powerbtn.sh c/samples/powerbtn/powerbtn.sh
--- b/samples/powerbtn/powerbtn.sh	2014-08-05 01:13:22.458907799 -0400
+++ c/samples/powerbtn/powerbtn.sh	2014-08-05 01:14:08.089910046 -0400
@@ -53,6 +53,7 @@
 # A list of power management system process names.
 PMS="gnome-settings-daemon kpowersave xfce4-power-manager"
 PMS="$PMS guidance-power-manager.py dalston-power-applet"
+PMS="$PMS cinnamon-settings-daemon"
 
 # If one of those is running or any of several others,
 if pidof x $PMS > /dev/null ||
