diff -Nuar dhcpcd-5.5.0/dhcpcd.c dhcpcd-5.5.0-fix/dhcpcd.c
--- dhcpcd-5.5.0/dhcpcd.c	2012-01-24 13:27:55.000000000 +0200
+++ dhcpcd-5.5.0-fix/dhcpcd.c	2012-01-30 06:43:17.000000000 +0200
@@ -1162,7 +1162,7 @@
 	free(iface->state->offer);
 	iface->state->offer = NULL;
 
-	if (ifo->options & DHCPCD_IPV6RS)
+	if (options & DHCPCD_IPV6RS && ifo->options & DHCPCD_IPV6RS)
 		ipv6rs_start(iface);
 
 	if (iface->state->arping_index < ifo->arping_len) {
