diff -Nuar dhcpcd-6.2.0/dhcpcd.conf dhcpcd-6.2.0-fix/dhcpcd.conf
--- dhcpcd-6.2.0/dhcpcd.conf	2014-01-03 21:12:19.000000000 +0000
+++ dhcpcd-6.2.0-fix/dhcpcd.conf	2014-01-20 10:10:39.024221778 +0000
@@ -33,3 +33,7 @@
 # A hook script is provided to lookup the hostname if not set by the DHCP
 # server, but it should not be run by default.
 nohook lookup-hostname
+
+# FL-971: do not run wpa_supplicant hook by default, this will prevent race condition, example issue
+# NetworkManager runs wpa_supplicant on it's own. Also it starts dhcpcd daemon, which also trying to run wpa_supplicant, making 2 instances running at the same time.
+nohook wpa_supplicant
